<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Datasets description | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</title>
  <meta name="description" content="3 Datasets description | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Datasets description | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Datasets description | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  
  
  



<meta name="date" content="2023-06-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Omic.html"/>
<link rel="next" href="exploration-of-genotype-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script>
  function unrolltab(evt, divName, tabName) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementById(divName).getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
      tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementById(divName).getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
      tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(tabName).style.display = "block";
    evt.currentTarget.className += " active";
  }
</script>

<script>
  document.addEventListener("DOMContentLoaded", function(event) { 
    document.getElementById("gwas_results_1").click();
    document.getElementById("gwas_plot_1").click();
    document.getElementById("manhattan-plots-tab").click();
    document.getElementById("locuszoom-plots-tabs").click();
 });
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="" target="blank">OmicSHIELD</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#materials-to-read-beforehand"><i class="fa fa-check"></i><b>1.1</b> Materials to read beforehand</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#what-are-resources-a-very-simple-explanation-without-any-technicalities"><i class="fa fa-check"></i><b>1.2</b> What are “resources”: A very simple explanation without any technicalities</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#capabilities-of-omicshield"><i class="fa fa-check"></i><b>1.3</b> Capabilities of OmicSHIELD</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#opal-servers"><i class="fa fa-check"></i><b>1.4</b> Opal servers</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>2</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html"><i class="fa fa-check"></i><b>3</b> Datasets description</a>
<ul>
<li class="chapter" data-level="3.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAdescription"><i class="fa fa-check"></i><b>3.1</b> CINECA project</a></li>
<li class="chapter" data-level="3.2" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAopal"><i class="fa fa-check"></i><b>3.2</b> Data organization in the Opal server</a></li>
<li class="chapter" data-level="3.3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#TCGALiverdatadescription"><i class="fa fa-check"></i><b>3.3</b> TCGA Liver project</a></li>
<li class="chapter" data-level="3.4" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#GSE66351description"><i class="fa fa-check"></i><b>3.4</b> Epigenetic data from the GSE66351 project</a></li>
<li class="chapter" data-level="3.5" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXdescription"><i class="fa fa-check"></i><b>3.5</b> HELIX project</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXopal"><i class="fa fa-check"></i><b>3.5.1</b> Data organization in the Opal server</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html"><i class="fa fa-check"></i><b>4</b> Exploration of Genotype data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#connection-to-the-opal-server"><i class="fa fa-check"></i><b>4.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="4.2" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#multi-vcf"><i class="fa fa-check"></i><b>4.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="4.3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>4.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="4.4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#merge-vcf-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>4.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="4.5" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-info"><i class="fa fa-check"></i><b>4.5</b> Basic information of the Genotype data</a></li>
<li class="chapter" data-level="4.6" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-statistics-of-the-genotype-data"><i class="fa fa-check"></i><b>4.6</b> Basic statistics of the genotype data</a></li>
<li class="chapter" data-level="4.7" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.7</b> Principal component analysis</a></li>
</ul></li>
<li class="part"><span><b>II Workflows</b></span></li>
<li class="chapter" data-level="5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html"><i class="fa fa-check"></i><b>5</b> Genome wide association study (GWAS) analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#single-cohort"><i class="fa fa-check"></i><b>5.1</b> Single cohort</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-1"><i class="fa fa-check"></i><b>5.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.1.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-vcf-resources"><i class="fa fa-check"></i><b>5.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.1.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes"><i class="fa fa-check"></i><b>5.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.1.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#vcf_merge"><i class="fa fa-check"></i><b>5.1.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.1.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas"><i class="fa fa-check"></i><b>5.1.5</b> Performing the GWAS</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-cohorts"><i class="fa fa-check"></i><b>5.2</b> Multi cohorts</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>5.2.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-vcf2"><i class="fa fa-check"></i><b>5.2.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.2.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-10"><i class="fa fa-check"></i><b>5.2.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.2.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information-1"><i class="fa fa-check"></i><b>5.2.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.2.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1"><i class="fa fa-check"></i><b>5.2.5</b> Performing a meta - GWAS</a></li>
<li class="chapter" data-level="5.2.6" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#gwas-plots"><i class="fa fa-check"></i><b>5.2.6</b> GWAS Plots</a></li>
<li class="chapter" data-level="5.2.7" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control"><i class="fa fa-check"></i><b>5.2.7</b> Meta-level quality control</a></li>
<li class="chapter" data-level="5.2.8" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-analysis"><i class="fa fa-check"></i><b>5.2.8</b> Meta-analysis</a></li>
<li class="chapter" data-level="5.2.9" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas"><i class="fa fa-check"></i><b>5.2.9</b> Ultra fast pooled GWAS analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#post-omics"><i class="fa fa-check"></i><b>5.3</b> Post-omic analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#enrichment-analysis"><i class="fa fa-check"></i><b>5.3.1</b> Enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html"><i class="fa fa-check"></i><b>6</b> Polygenic risk scores</a>
<ul>
<li class="chapter" data-level="6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#definition"><i class="fa fa-check"></i><b>6.1</b> Definition</a></li>
<li class="chapter" data-level="6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-configuration"><i class="fa fa-check"></i><b>6.2</b> Study configuration</a></li>
<li class="chapter" data-level="6.3" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#source-of-polygenic-scores"><i class="fa fa-check"></i><b>6.3</b> Source of polygenic scores</a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>6.4</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="6.5" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#assign-the-vcf-resources-1"><i class="fa fa-check"></i><b>6.5</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="6.6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#calculate-the-prs"><i class="fa fa-check"></i><b>6.6</b> Calculate the PRS</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#pgs-catalog-example"><i class="fa fa-check"></i><b>6.6.1</b> PGS Catalog Example</a></li>
<li class="chapter" data-level="6.6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#custom-prs_table-example"><i class="fa fa-check"></i><b>6.6.2</b> Custom <code>prs_table</code> example</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-of-the-prs-results"><i class="fa fa-check"></i><b>6.7</b> Study of the PRS results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plink.html"><a href="plink.html"><i class="fa fa-check"></i><b>7</b> PLINK</a>
<ul>
<li class="chapter" data-level="7.1" data-path="plink.html"><a href="plink.html#connection-to-the-opal-server-3"><i class="fa fa-check"></i><b>7.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="7.2" data-path="plink.html"><a href="plink.html#build-the-plink-call"><i class="fa fa-check"></i><b>7.2</b> Build the PLINK call</a></li>
<li class="chapter" data-level="7.3" data-path="plink.html"><a href="plink.html#call-the-analysis"><i class="fa fa-check"></i><b>7.3</b> Call the analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="snptest.html"><a href="snptest.html"><i class="fa fa-check"></i><b>8</b> SNPTEST</a>
<ul>
<li class="chapter" data-level="8.1" data-path="snptest.html"><a href="snptest.html#connection-to-the-opal-server-4"><i class="fa fa-check"></i><b>8.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="8.2" data-path="snptest.html"><a href="snptest.html#build-the-snptest-call"><i class="fa fa-check"></i><b>8.2</b> Build the SNPTEST call</a></li>
<li class="chapter" data-level="8.3" data-path="snptest.html"><a href="snptest.html#call-the-analysis-1"><i class="fa fa-check"></i><b>8.3</b> Call the analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>9</b> Differential gene expression (DGE) analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#connection-to-the-opal-server-5"><i class="fa fa-check"></i><b>9.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="9.2" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#assign-the-rse-resource"><i class="fa fa-check"></i><b>9.2</b> Assign the RSE resource</a></li>
<li class="chapter" data-level="9.3" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#inspect-the-rse"><i class="fa fa-check"></i><b>9.3</b> Inspect the RSE</a></li>
<li class="chapter" data-level="9.4" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#pre-processing-for-rnaseq-data"><i class="fa fa-check"></i><b>9.4</b> Pre-processing for RNAseq data</a></li>
<li class="chapter" data-level="9.5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#dge-analysis"><i class="fa fa-check"></i><b>9.5</b> DGE analysis</a></li>
<li class="chapter" data-level="9.6" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>9.6</b> Surrogate variable analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>10</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#initial-steps-for-all-use-cases"><i class="fa fa-check"></i><b>10.1</b> Initial steps for all use cases</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#connection-to-the-opal-server-6"><i class="fa fa-check"></i><b>10.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="10.1.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#assign-the-expressionset-resource"><i class="fa fa-check"></i><b>10.1.2</b> Assign the <code>ExpressionSet</code> resource</a></li>
<li class="chapter" data-level="10.1.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#ewas-inspect"><i class="fa fa-check"></i><b>10.1.3</b> Inspect the <code>ExpressionSet</code></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-pooled-analysis"><i class="fa fa-check"></i><b>10.2</b> Single CpG pooled analysis</a></li>
<li class="chapter" data-level="10.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpgs-pooled-analysis"><i class="fa fa-check"></i><b>10.3</b> Multiple CpGs pooled analysis</a></li>
<li class="chapter" data-level="10.4" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#full-genome-meta-analysis"><i class="fa fa-check"></i><b>10.4</b> Full genome meta-analysis</a></li>
<li class="chapter" data-level="10.5" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>10.5</b> Adjusting for surrogate variables</a></li>
</ul></li>
<li class="part"><span><b>III Showcase on a real dataset</b></span></li>
<li class="chapter" data-level="11" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html"><i class="fa fa-check"></i><b>11</b> Transcriptomic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXone"><i class="fa fa-check"></i><b>11.1</b> Getting started.</a></li>
<li class="chapter" data-level="11.2" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXdatamanip"><i class="fa fa-check"></i><b>11.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="11.3" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXDGE"><i class="fa fa-check"></i><b>11.3</b> Differential gene expression analysis.</a></li>
<li class="chapter" data-level="11.4" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXsurrogateAnnot"><i class="fa fa-check"></i><b>11.4</b> Surrogate variable analysis.</a></li>
<li class="chapter" data-level="11.5" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXenrichment"><i class="fa fa-check"></i><b>11.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html"><i class="fa fa-check"></i><b>12</b> Epigenetic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="12.1" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_gettingstarted"><i class="fa fa-check"></i><b>12.1</b> Getting started.</a></li>
<li class="chapter" data-level="12.2" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_load"><i class="fa fa-check"></i><b>12.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="12.3" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis3"><i class="fa fa-check"></i><b>12.3</b> Full genome meta-analysis.</a></li>
<li class="chapter" data-level="12.4" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis4"><i class="fa fa-check"></i><b>12.4</b> Adjusting for surrogate variables.</a></li>
<li class="chapter" data-level="12.5" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIXenrichment"><i class="fa fa-check"></i><b>12.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="part"><span><b>IV Technical details</b></span></li>
<li class="chapter" data-level="13" data-path="fast-gwas.html"><a href="fast-gwas.html"><i class="fa fa-check"></i><b>13</b> Fast GWAS</a></li>
<li class="chapter" data-level="14" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>14</b> PCA</a></li>
<li class="chapter" data-level="15" data-path="differential-privacy.html"><a href="differential-privacy.html"><i class="fa fa-check"></i><b>15</b> Differential privacy</a></li>
<li class="chapter" data-level="16" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html"><i class="fa fa-check"></i><b>16</b> Internal structure of the PRS functionality</a>
<ul>
<li class="chapter" data-level="16.1" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#client"><i class="fa fa-check"></i><b>16.1</b> Client</a></li>
<li class="chapter" data-level="16.2" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-resolver"><i class="fa fa-check"></i><b>16.2</b> Server resolver</a></li>
<li class="chapter" data-level="16.3" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-analysis"><i class="fa fa-check"></i><b>16.3</b> Server analysis</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="compress.html"><a href="compress.html"><i class="fa fa-check"></i><b>17</b> Compression of GDS files vs. performance</a></li>
<li class="chapter" data-level="18" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html"><i class="fa fa-check"></i><b>18</b> Recommended R server specs</a>
<ul>
<li class="chapter" data-level="18.1" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#limma-voom"><i class="fa fa-check"></i><b>18.1</b> Limma + voom</a></li>
<li class="chapter" data-level="18.2" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#gwas-pca"><i class="fa fa-check"></i><b>18.2</b> GWAS + PCA</a></li>
<li class="chapter" data-level="18.3" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#general-remarks"><i class="fa fa-check"></i><b>18.3</b> General remarks</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="disclosure-filter-configurations.html"><a href="disclosure-filter-configurations.html"><i class="fa fa-check"></i><b>19</b> Disclosure filter configurations</a></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="20" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>20</b> Session info</a></li>
<li class="chapter" data-level="21" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>21</b> Contributors</a></li>
<li class="part"><span><b>VI References</b></span></li>
<li class="chapter" data-level="22" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>22</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="DatasetsDescription" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Datasets description<a href="DatasetsDescription.html#DatasetsDescription" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Here, we will <strong>describe the data employed for the analyses presented in the bookdown</strong>, giving details about <strong>experimental designs</strong>, the type of <strong>features available</strong>, and the <strong>structure and organization</strong> of data in the <a href="https://opaldoc.obiba.org/en/latest/"><strong>Opal infrastructure</strong></a>. Employed data constitute both <strong>public data</strong> downloaded from public repositories, and <strong>real datasets</strong> deriving from an European H2020 research project on exposome and health (<a href="https://www.projecthelix.eu">HELIX Project</a>). The public datasets have been uploaded to the <a href="https://opal-demo.obiba.org/">Opal demo site</a> and can be easily accessed without additional permissions. On the other hand, the <strong>HELIX</strong> dataset is available in the <a href="https://datashield.isglobal.org/repo">Opal BRGE site</a> hosted by the <a href="https://brge.isglobal.org/">Bioinformatic Research Group in Epidemiology</a> of ISGlobal. In both cases, data are organized in the Opal simulating a <strong>single-site DataSHIELD infrastructure</strong>. Details for <strong>accessing</strong> the server will be found in each section, at the beginning of each pipeline.</p>
<div id="CINECAdescription" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> CINECA project<a href="DatasetsDescription.html#CINECAdescription" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Data from the public CINECA project were employed in this bookdown for demonstrating the capabilities of dsOmicSHIELD for performing <strong>GWAS and other genetic analyses</strong>. Data correspond to synthetic data set generated by the <a href="https://www.cineca-project.eu/">CINECA Project</a> made freely available under the Creative Commons Licence (CC-BY) (funding: EC H2020 grant 825775). In <strong>Figure <a href="DatasetsDescription.html#fig:figCINECA">3.1</a></strong>, an overall description of the datasets and cohorts belonging to the <strong>CINECA project</strong> is presented.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:figCINECA"></span>
<img src="https://images.squarespace-cdn.com/content/v1/5cc2c7b2d86cc9589d33dfcc/1573825882582-AS6SBW8PMXNGKSJWS79H/cohort_figure_updated_Nov2019.png?format=750w" alt="The figure depicts CINECA project cohorts representing &gt;1.4 Million individuals and spanning three continents."  />
<p class="caption">
Figure 3.1: The figure depicts CINECA project cohorts representing &gt;1.4 Million individuals and spanning three continents.
</p>
</div>
</div>
<div id="CINECAopal" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Data organization in the Opal server<a href="DatasetsDescription.html#CINECAopal" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Data were initially accessible from a public EGA repository (ID: <a href="https://ega-archive.org/studies/EGAS00001002472/">EGAD00001006673</a>), being a deliverable of the <a href="https://veis.bsc.es/en/">VEIS project</a>. Previous to analysis, data were <strong>pruned</strong>, and the individuals were separated into <strong>three synthetic cohorts</strong>. Information on individuals and SNP count of each cohort can be found on the Table <a href="DatasetsDescription.html#tab:tab1">3.1</a>.</p>
<table>
<caption><span id="tab:tab1">Table 3.1: </span> Number of SNPs and individuals by cohort</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">Cohort 1</th>
<th align="right">Cohort 2</th>
<th align="right">Cohort 3</th>
<th align="right">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Number of SNPs</td>
<td align="right">865,240</td>
<td align="right">865,240</td>
<td align="right">865,240</td>
<td align="right">865,240</td>
</tr>
<tr class="even">
<td>Number of individuals</td>
<td align="right">817</td>
<td align="right">1,073</td>
<td align="right">614</td>
<td align="right">2,504</td>
</tr>
</tbody>
</table>
<p>The three datasets were uploaded to the <a href="https://opal-demo.obiba.org/">Opal demo site</a> organized as (1) an array of <strong>resource</strong> objects, corresponding to unitary VCF files with variant information (one per chromosome and cohort), and (2) <strong>table</strong> objects corresponding to the phenotype data (one per cohort). Data are in Opal simulating a <strong>single-site DataSHIELD architecture</strong>, as in <strong>Figure <a href="DatasetsDescription.html#fig:FigCINECAopal">3.2</a></strong>. In this figure, the data analyst corresponds to the “RStudio” session, which through the <strong>DataSHIELD Interface (DSI)</strong> connects with the Opal server located at the cohort network.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:FigCINECAopal"></span>
<img src="fig/single_cohort.png" alt="Proposed infrastructure to perform single-cohort GWAS studies." width="90%" />
<p class="caption">
Figure 3.2: Proposed infrastructure to perform single-cohort GWAS studies.
</p>
</div>
<p>For more information about <strong>DataSHIELD data formats</strong>, please, refers to the bookdown <a href="https://isglobal-brge.github.io/resource_bookdown/index.html"><em>“Orchestrating privacy-protected non-disclosive big data analyses of data from different resources with R and DataSHIELD”</em></a>.</p>
</div>
<div id="TCGALiverdatadescription" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> TCGA Liver project<a href="DatasetsDescription.html#TCGALiverdatadescription" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The public TCGA Liver project was selected for illustrating how to perform <strong>transcriptomic data analysis</strong> in DataSHIELD. These data derived from the famous <a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga">TCGA project</a>. We have uploaded to the demo Opal server a resource called <code>tcga_liver</code> whose URL is <a href="http://duffel.rail.bio/recount/TCGA/rse_gene_liver.Rdata" class="uri">http://duffel.rail.bio/recount/TCGA/rse_gene_liver.Rdata</a> available through the <a href="https://jhubiostatistics.shinyapps.io/recount/">recount project</a>.</p>
<p>Some summaries of the dataset are the following:</p>
<table>
<thead>
<tr class="header">
<th></th>
<th>TCGA Liver data</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Number of individuals</strong></td>
<td>424</td>
</tr>
<tr class="even">
<td><strong>Number of genes</strong></td>
<td>58,037</td>
</tr>
<tr class="odd">
<td><strong>Number of covariate fields</strong></td>
<td>864</td>
</tr>
</tbody>
</table>
<p>The <code>RangedSummarizedExperiment</code> (RSE) objects are part of the <a href="https://www.bioconductor.org/">BioConductor project</a>, as a brief summary, these types of objects are designed to contain different multi-omic data, they can also contain the individuals phenotypes and additional metadata associated with a particular experiment. For further information about what are <code>RangedSummarizedExperiment</code> please take a look at the <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html">official documentation</a></p>
<p>The resource used on this project contains a <code>RangedSummarizedExperiment</code> with the RNAseq profiling of liver cancer data from TCGA. The resource is located inside the <code>OMICS</code> project. The structure used is illustrated on the following figure.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:dgeProposal1"></span>
<img src="fig/dge_structure.png" alt="Proposed infrastructure to perform DGE studies." width="698" />
<p class="caption">
Figure 3.3: Proposed infrastructure to perform DGE studies.
</p>
</div>
<p>The data analyst corresponds to the “RStudio” session, which through DataSHIELD Interface (DSI) connects with the Opal server. This Opal server contains a resource that corresponds to the TCGA liver RSE.</p>
</div>
<div id="GSE66351description" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Epigenetic data from the GSE66351 project<a href="DatasetsDescription.html#GSE66351description" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The public data from the GSE66351 project were used in this section for illustrating how to perform <strong>EWAS</strong> in DataSHIELD. They has been downloaded from <a href="https://www.ncbi.nlm.nih.gov/geo/">GEO</a> (accession number GSE66351) and consist on DNA methylation reads from the Illumina 450K array. Data corresponds to CpGs beta values measured in the superior temporal gyrus and prefrontal cortex brain regions of patients with Alzheimer’s.</p>
<p>This kind of data is encapsulated on a type of R object called <code>ExpressionSet</code>, this objects are part of the <a href="https://www.bioconductor.org/">BioConductor project</a> and are meant to contain different sources of genomic data, alongside the genomic data they can also contain the phenotypes and metadata associated to a study. Researchers who are not familiar with <code>ExpressionSet</code> can find further information <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf">here</a>. Notice that genomic data is encoded as beta-values that ensure data harmonization across studies.</p>
<p>In order to illustrate how to perform data analyses using federated data, we have split the data into two synthetic cohorts (split by individuals). We have created two resources on the demo Opal server called <code>GSE66351_1</code> and <code>GSE66351_2</code> respectively. They can be found inside the <code>OMICS</code> project. The structure used is illustrated on the following figure.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:ewasProposal1"></span>
<img src="fig/ewas_structure.png" alt="Proposed infrastructure to perform EWAS studies." width="730" />
<p class="caption">
Figure 3.4: Proposed infrastructure to perform EWAS studies.
</p>
</div>
<p>Some summaries of the datasets are the following:</p>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Cohort 1</th>
<th align="right">Cohort 2</th>
<th align="right">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Number of CpGs</td>
<td align="right">481,868</td>
<td align="right">481,868</td>
<td align="right">481,868</td>
</tr>
<tr class="even">
<td>Number of individuals</td>
<td align="right">100</td>
<td align="right">90</td>
<td align="right">190</td>
</tr>
<tr class="odd">
<td>Number of covariates</td>
<td align="right">49</td>
<td align="right">49</td>
<td align="right">49</td>
</tr>
<tr class="even">
<td>Number of annotation features</td>
<td align="right">37</td>
<td align="right">37</td>
<td align="right">37</td>
</tr>
</tbody>
</table>
</div>
<div id="HELIXdescription" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> HELIX project<a href="DatasetsDescription.html#HELIXdescription" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The real-world research problem presented in this book comes from the <a href="https://www.projecthelix.eu">HELIX Project</a>. The <strong>HELIX (Human Early-Life Exposome) project</strong> gathers data from <strong>6 longitudinal-based European birth cohorts</strong>, with the aim of evaluating the effect of <strong>environmental risk factors</strong> on <strong>mothers and children health</strong>, with a special focus in the effects on <strong>molecular health profiles (omic data)</strong>. <strong>HELIX cohorts</strong> include the <em>BIB</em> (Born in Bradford) (United Kingdom), <em>EDEN</em> (Étude des Déterminants pré et postnatals du développement et de la santé de l’ENfant) (France), <em>INMA</em> (INfancia y Medio Ambiente) (Spain), <em>KANC</em> (Kaunus Cohort) (Lithuania), <em>MoBa</em> (Norwegian Mother and Child Cohort Study) (Norway), and <em>Rhea</em> (Mother-Child Cohort in Crete) (Greece). General details of the <strong>study design</strong> of <strong>HELIX</strong> can be found in the <strong>Figure <a href="DatasetsDescription.html#fig:FigHELIX">3.5</a></strong>. The whole <strong>HELIX dataset</strong> includes a total of 31,472 mother-child pairs. Among them, a <strong>subcohort of 1,301 children</strong> (approximately 200 children in each cohort) was selected <strong>here</strong> according to the following criteria of eligibility: 1) age 6 to 11 years; 2) omic data available (including gene expression and EWAS data); 3) complete address history; and 4) no serious health problems that may affect the clinical testing or the child safety.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:FigHELIX"></span>
<img src="http://www.projecthelix.eu/images/estructura/helix_study.gif" alt="General overview of the HELIX research project." width="90%" />
<p class="caption">
Figure 3.5: General overview of the HELIX research project.
</p>
</div>
<p>In the <strong>1,301 children</strong>, a wide range of <strong>environmental exposures</strong> were evaluated to define the early-life exposome during <strong>two time periods</strong>: the <strong>prenatal pregnancy period</strong> and <strong>childhood</strong> (age 6 to 11 years). Collected exposures can be organized into three main parts of the <strong>exposome</strong>: <strong>outdoor exposures, chemical exposures measured through biomarkers, and lifestyle factors</strong>. All variables incorporated in the dataset have been appropriately <strong>pre-processed</strong> previous to analysis (outliers removal, normalization, and missing values imputation). Regarding <strong>phenotype data</strong>, a wide range of <strong>health outcomes</strong> are also available for the <strong>childhood</strong> period including phenotypes related to (1) obesity and cardiometabolic health, (2) respiratory health, and (3) cognition and mental health. Finally, <strong>omic data</strong> (mainly <strong>transcriptomic</strong> and <strong>genome-wide DNA methylation data</strong>) are also available in the <strong>childhood</strong> period. The <strong>objective</strong> of this dataset was illustrating a real case study of:</p>
<ul>
<li>Differential <strong>gene expression</strong> analysis.</li>
<li>Differential <strong>DNA methylation</strong> analysis.</li>
</ul>
<div id="HELIXopal" class="section level3 hasAnchor" number="3.5.1">
<h3><span class="header-section-number">3.5.1</span> Data organization in the Opal server<a href="DatasetsDescription.html#HELIXopal" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Omic</strong> data from the <strong>HELIX project</strong> were uploaded into to the <a href="https://datashield.isglobal.org/brge/ui/index.html">Opal BRGE site</a> simulating a <strong>single-site DataSHIELD architecture</strong>, as illustrated in <strong>Figure <a href="DatasetsDescription.html#fig:FigHELIXopal">3.6</a></strong>. Data were stored under the form of <strong>resources</strong> (one per cohort and data type). As illustrated in the figure, transcriptomic data) were stored under the form of <em>expressionSets</em> while EWAS data were stored as <em>GenomicRatioSet</em>. A summary of the structure and organization of this data in the <strong>Opal</strong> for the <strong>HELIX project</strong> is illustrated in <strong>Figure <a href="DatasetsDescription.html#fig:FigHELIXopal">3.6</a></strong>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:FigHELIXopal"></span>
<img src="fig/HELIX_OPAL_STRUCTURE.png" alt="Input HELIX datasets organized by cohort in the Opal server." width="90%" />
<p class="caption">
Figure 3.6: Input HELIX datasets organized by cohort in the Opal server.
</p>
</div>
<p>In the next subsection, we present <strong>extra details</strong> for each of these datasets.</p>
<div id="TranscriptomicsHELIXDatadescript" class="section level4 hasAnchor" number="3.5.1.1">
<h4><span class="header-section-number">3.5.1.1</span> Gene expression data<a href="DatasetsDescription.html#TranscriptomicsHELIXDatadescript" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <strong>gene expression</strong> data available for the <strong>HELIX cohorts</strong> correspond to <strong>microarray data</strong> deriving from the <em>Human Transcriptome Array 2.0</em> platform of Affymetrix. These are hosted in the the <a href="https://datashield.isglobal.org/brge/ui/index.html">Opal BRGE site</a> as <em>ExpressionSet</em> objects. Particularly, one <em>ExpressionSet</em> file per cohort is hosted in the <a href="https://datashield.isglobal.org/brge/ui/index.html">Opal BRGE site</a> in the form of a <strong>resource</strong>, as presented in the <strong>Figure <a href="DatasetsDescription.html#fig:FigHELIXresourcesOpaltranscrip">3.7</a></strong>. As it can be seen, data are organized within the Opal server in a <strong>project</strong> called <strong>HELIX_omics</strong>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:FigHELIXresourcesOpaltranscrip"></span>
<img src="fig/geneexpression_HELIX.png" alt="Input ExposomeSet data organized by cohort in the Opal server as resources." width="3000px" height="500px" />
<p class="caption">
Figure 3.7: Input ExposomeSet data organized by cohort in the Opal server as resources.
</p>
</div>
</div>
<div id="EWASHELIXDatadescript" class="section level4 hasAnchor" number="3.5.1.2">
<h4><span class="header-section-number">3.5.1.2</span> Epigenetic data<a href="DatasetsDescription.html#EWASHELIXDatadescript" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <strong>epigenetic</strong> data available for the <strong>HELIX cohorts</strong> correspond to genome-wide <strong>DNA methylation data</strong> obtained by microarray experiments using the <em>Infinium HumanMethylation450k</em> platform from Illumina. The data format in which these data are hosted in the Opal is the type <strong>GenomicRatioSet</strong>. <strong>GenomicRatioSet</strong> is a extend of the class <strong>SummarizedExperiment</strong> especially designed for DNA methylation data obtained with the <em>Infinium HumanMethylation450k</em> platform, that will be analysed with the associated R package <a href="http://bioconductor.org/packages/release/bioc/html/minfi.html">minfi</a>. This type of data usually contain <strong>pre-processed DNA methylation values</strong> at a genome-wide level, in the form of <strong>M or/and Beta values</strong>, together with the associated <strong>genomic coordinates</strong>. As in the case of <strong>ExpressionSet</strong> types, <strong>GenomicRatioSet</strong> can also incorporate <strong>phenotype</strong> and <strong>metadata</strong> information.</p>
<p>Particularly, one <strong>GenomicRatioSet</strong> file per cohort is hosted in the <a href="https://datashield.isglobal.org/brge/ui/index.html">Opal BRGE site</a> in the form of a <strong>resource</strong>, as presented in the <strong>Figure <a href="DatasetsDescription.html#fig:FigHELIXresourcesOpalDNAm">3.8</a></strong>. As it can be seen, data are organized within the Opal server in a <strong>project</strong> called <strong>HELIX_omics</strong>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:FigHELIXresourcesOpalDNAm"></span>
<img src="fig/dnaM_HELIX.png" alt="Input GenomicRatioSet data organized by cohort in the Opal server as resources." width="3000px" height="500px" />
<p class="caption">
Figure 3.8: Input GenomicRatioSet data organized by cohort in the Opal server as resources.
</p>
</div>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Omic.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="exploration-of-genotype-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OmicSHIELD_bookdown.pdf", "OmicSHIELD_bookdown.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
