<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</title>
  <meta name="description" content="3 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  
  
  



<meta name="date" content="2022-03-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Omic.html"/>
<link rel="next" href="DatasetsDescription.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script>
  function unrolltab(evt, divName, tabName) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementById(divName).getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
      tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementById(divName).getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
      tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(tabName).style.display = "block";
    evt.currentTarget.className += " active";
  }
</script>

<script>
  document.addEventListener("DOMContentLoaded", function(event) { 
    document.getElementById("gwas_results_1").click();
    document.getElementById("gwas_plot_1").click();
    document.getElementById("manhattan-plots-tab").click();
    document.getElementById("locuszoom-plots-tabs").click();
 });
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="" target="blank">OmicSHIELD</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#materials-to-read-beforehand"><i class="fa fa-check"></i><b>1.1</b> Materials to read beforehand</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#what-are-resources-a-very-simple-explanation-without-any-technicalities"><i class="fa fa-check"></i><b>1.2</b> What are “resources”: A very simple explanation without any technicalities</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#capabilities-of-omicshield"><i class="fa fa-check"></i><b>1.3</b> Capabilities of OmicSHIELD</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#opal-servers"><i class="fa fa-check"></i><b>1.4</b> Opal servers</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>2</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html"><i class="fa fa-check"></i><b>3</b> Exploration of Genotype data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#setting-up"><i class="fa fa-check"></i><b>3.1</b> Setting up</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#connection-to-the-opal-server"><i class="fa fa-check"></i><b>3.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="3.1.2" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#assign-the-vcf-resources"><i class="fa fa-check"></i><b>3.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="3.1.3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#assign-the-phenotypes"><i class="fa fa-check"></i><b>3.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="3.1.4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#merge-gds-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>3.1.4</b> Merge GDS (genotype) and phenotype information</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-info"><i class="fa fa-check"></i><b>3.2</b> Basic information of the Genotype data</a></li>
<li class="chapter" data-level="3.3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-statistics-of-the-genotype-data"><i class="fa fa-check"></i><b>3.3</b> Basic statistics of the genotype data</a></li>
<li class="chapter" data-level="3.4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#principal-component-analysis"><i class="fa fa-check"></i><b>3.4</b> Principal component analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html"><i class="fa fa-check"></i><b>4</b> Datasets description.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAdescription"><i class="fa fa-check"></i><b>4.1</b> CINECA project.</a></li>
<li class="chapter" data-level="4.2" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAopal"><i class="fa fa-check"></i><b>4.2</b> Data organization in the Opal server.</a></li>
<li class="chapter" data-level="4.3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#TCGALiverdatadescription"><i class="fa fa-check"></i><b>4.3</b> TCGA Liver project.</a></li>
<li class="chapter" data-level="4.4" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#GSE66351description"><i class="fa fa-check"></i><b>4.4</b> Epigenetic data from the GSE66351 project</a></li>
<li class="chapter" data-level="4.5" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXdescription"><i class="fa fa-check"></i><b>4.5</b> HELIX project.</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXopal"><i class="fa fa-check"></i><b>4.5.1</b> Data organization in the Opal server.</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Workflows</b></span></li>
<li class="chapter" data-level="5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html"><i class="fa fa-check"></i><b>5</b> Genome wide association study (GWAS) analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#single-cohort"><i class="fa fa-check"></i><b>5.1</b> Single cohort</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-1"><i class="fa fa-check"></i><b>5.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.1.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-vcf-resources-1"><i class="fa fa-check"></i><b>5.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.1.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>5.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.1.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#vcf_merge"><i class="fa fa-check"></i><b>5.1.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.1.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas"><i class="fa fa-check"></i><b>5.1.5</b> Performing the GWAS</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-cohorts"><i class="fa fa-check"></i><b>5.2</b> Multi cohorts</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>5.2.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-vcf"><i class="fa fa-check"></i><b>5.2.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.2.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>5.2.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.2.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>5.2.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.2.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1"><i class="fa fa-check"></i><b>5.2.5</b> Performing a meta - GWAS</a></li>
<li class="chapter" data-level="5.2.6" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#gwas-plots"><i class="fa fa-check"></i><b>5.2.6</b> GWAS Plots</a></li>
<li class="chapter" data-level="5.2.7" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control"><i class="fa fa-check"></i><b>5.2.7</b> Meta-level quality control</a></li>
<li class="chapter" data-level="5.2.8" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-analysis"><i class="fa fa-check"></i><b>5.2.8</b> Meta-analysis</a></li>
<li class="chapter" data-level="5.2.9" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas"><i class="fa fa-check"></i><b>5.2.9</b> Ultra fast pooled GWAS analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#post-omics"><i class="fa fa-check"></i><b>5.3</b> Post-omic analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#enrichment-analysis"><i class="fa fa-check"></i><b>5.3.1</b> Enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html"><i class="fa fa-check"></i><b>6</b> Polygenic risk scores</a>
<ul>
<li class="chapter" data-level="6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#definition"><i class="fa fa-check"></i><b>6.1</b> Definition</a></li>
<li class="chapter" data-level="6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-configuration"><i class="fa fa-check"></i><b>6.2</b> Study configuration</a></li>
<li class="chapter" data-level="6.3" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#source-of-polygenic-scores"><i class="fa fa-check"></i><b>6.3</b> Source of polygenic scores</a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>6.4</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="6.5" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#assign-the-vcf-resources-2"><i class="fa fa-check"></i><b>6.5</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="6.6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#calculate-the-prs"><i class="fa fa-check"></i><b>6.6</b> Calculate the PRS</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#pgs-catalog-example"><i class="fa fa-check"></i><b>6.6.1</b> PGS Catalog Example</a></li>
<li class="chapter" data-level="6.6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#custom-prs_table-example"><i class="fa fa-check"></i><b>6.6.2</b> Custom <code>prs_table</code> example</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-of-the-prs-results"><i class="fa fa-check"></i><b>6.7</b> Study of the PRS results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>7</b> Differential gene expression (DGE) analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#connection-to-the-opal-server-3"><i class="fa fa-check"></i><b>7.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="7.2" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#assign-the-rse-resource"><i class="fa fa-check"></i><b>7.2</b> Assign the RSE resource</a></li>
<li class="chapter" data-level="7.3" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#inspect-the-rse"><i class="fa fa-check"></i><b>7.3</b> Inspect the RSE</a></li>
<li class="chapter" data-level="7.4" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#pre-processing-for-rnaseq-data"><i class="fa fa-check"></i><b>7.4</b> Pre-processing for RNAseq data</a></li>
<li class="chapter" data-level="7.5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#dge-analysis"><i class="fa fa-check"></i><b>7.5</b> DGE analysis</a></li>
<li class="chapter" data-level="7.6" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>7.6</b> Surrogate variable analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>8</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#initial-steps-for-all-use-cases"><i class="fa fa-check"></i><b>8.1</b> Initial steps for all use cases</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#connection-to-the-opal-server-4"><i class="fa fa-check"></i><b>8.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="8.1.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#assign-the-expressionset-resource"><i class="fa fa-check"></i><b>8.1.2</b> Assign the <code>ExpressionSet</code> resource</a></li>
<li class="chapter" data-level="8.1.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#ewas-inspect"><i class="fa fa-check"></i><b>8.1.3</b> Inspect the <code>ExpressionSet</code></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-pooled-analysis"><i class="fa fa-check"></i><b>8.2</b> Single CpG pooled analysis</a></li>
<li class="chapter" data-level="8.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpgs-pooled-analysis"><i class="fa fa-check"></i><b>8.3</b> Multiple CpGs pooled analysis</a></li>
<li class="chapter" data-level="8.4" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#full-genome-meta-analysis"><i class="fa fa-check"></i><b>8.4</b> Full genome meta-analysis</a></li>
<li class="chapter" data-level="8.5" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>8.5</b> Adjusting for surrogate variables</a></li>
</ul></li>
<li class="part"><span><b>III Showcase on a real dataset</b></span></li>
<li class="chapter" data-level="9" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html"><i class="fa fa-check"></i><b>9</b> Transcriptomic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXone"><i class="fa fa-check"></i><b>9.1</b> Getting started.</a></li>
<li class="chapter" data-level="9.2" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXdatamanip"><i class="fa fa-check"></i><b>9.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="9.3" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXDGE"><i class="fa fa-check"></i><b>9.3</b> Differential gene expression analysis.</a></li>
<li class="chapter" data-level="9.4" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXsurrogateAnnot"><i class="fa fa-check"></i><b>9.4</b> Surrogate variable analysis.</a></li>
<li class="chapter" data-level="9.5" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXenrichment"><i class="fa fa-check"></i><b>9.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html"><i class="fa fa-check"></i><b>10</b> Epigenetic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_gettingstarted"><i class="fa fa-check"></i><b>10.1</b> Getting started.</a></li>
<li class="chapter" data-level="10.2" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_load"><i class="fa fa-check"></i><b>10.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="10.3" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis3"><i class="fa fa-check"></i><b>10.3</b> Full genome meta-analysis.</a></li>
<li class="chapter" data-level="10.4" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis4"><i class="fa fa-check"></i><b>10.4</b> Adjusting for surrogate variables.</a></li>
<li class="chapter" data-level="10.5" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIXenrichment"><i class="fa fa-check"></i><b>10.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="references..html"><a href="references..html"><i class="fa fa-check"></i><b>11</b> References.</a></li>
<li class="part"><span><b>IV Technical details</b></span></li>
<li class="chapter" data-level="12" data-path="gwas.html"><a href="gwas.html"><i class="fa fa-check"></i><b>12</b> GWAS</a></li>
<li class="chapter" data-level="13" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>13</b> PCA</a></li>
<li class="chapter" data-level="14" data-path="differential-privacy.html"><a href="differential-privacy.html"><i class="fa fa-check"></i><b>14</b> Differential privacy</a></li>
<li class="chapter" data-level="15" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html"><i class="fa fa-check"></i><b>15</b> Internal structure of the PRS functionality</a>
<ul>
<li class="chapter" data-level="15.1" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#client"><i class="fa fa-check"></i><b>15.1</b> Client</a></li>
<li class="chapter" data-level="15.2" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-resolver"><i class="fa fa-check"></i><b>15.2</b> Server resolver</a></li>
<li class="chapter" data-level="15.3" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-analysis"><i class="fa fa-check"></i><b>15.3</b> Server analysis</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="compress.html"><a href="compress.html"><i class="fa fa-check"></i><b>16</b> Compression of GDS files vs. performance</a></li>
<li class="chapter" data-level="17" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html"><i class="fa fa-check"></i><b>17</b> Recommended R server specs</a>
<ul>
<li class="chapter" data-level="17.1" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#limma-voom"><i class="fa fa-check"></i><b>17.1</b> Limma + voom</a></li>
<li class="chapter" data-level="17.2" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#gwas-pca"><i class="fa fa-check"></i><b>17.2</b> GWAS + PCA</a></li>
<li class="chapter" data-level="17.3" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#general-remarks"><i class="fa fa-check"></i><b>17.3</b> General remarks</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="disclosure-filter-configurations.html"><a href="disclosure-filter-configurations.html"><i class="fa fa-check"></i><b>18</b> Disclosure filter configurations</a></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a></li>
<li class="chapter" data-level="20" data-path="contributos.html"><a href="contributos.html"><i class="fa fa-check"></i><b>20</b> Contributos</a></li>
<li class="part"><span><b>VI References</b></span></li>
<li class="chapter" data-level="21" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>21</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exploration-of-genotype-data" class="section level1" number="3">
<h1><span class="header-section-number">3</span> Exploration of Genotype data</h1>
<p>Theww different use cases will be illustrated:</p>
<ul>
<li>Basic information retrieval.</li>
<li>Basic statistics calculation.</li>
<li>Principal component analysis</li>
</ul>
<p>For all the illustrated cases we will use dataset that contains all the information in a single file of GDS format, for information about this format please refer to the (official documentation)[<a href="https://bioconductor.org/packages/release/bioc/vignettes/gdsfmt/inst/doc/gdsfmt.html" class="uri">https://bioconductor.org/packages/release/bioc/vignettes/gdsfmt/inst/doc/gdsfmt.html</a>].</p>
<p>The phenotype information used throughout this section is not contained inside the GDS files mentioned previously, it is contained using traditional CSV/Excel files. This is to replicate the typical scenario where an investigator receives the genomic and phenotype data separated and has to merge it to study a specific relationship between the gene expression / variants and a certain phenotype.</p>
<p>The data used along this section is a synthetic data set generated by the CINECA project made freely available under the Creative Commons Licence (CC-BY) (funding: EC H2020 grant 825775). It has been pruned to provide a small dataset for easy and quick demnostration. Information on the dataset can be found on the Table <a href="exploration-of-genotype-data.html#tab:tab1">3.1</a>.</p>
<table>
<caption><span id="tab:tab1">Table 3.1: </span> Numer of SNPs and individuals by cohort</caption>
<thead>
<tr class="header">
<th></th>
<th align="right">Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Number of SNPs</td>
<td align="right">651,808</td>
</tr>
<tr class="even">
<td>Number of individuals</td>
<td align="right">2,504</td>
</tr>
</tbody>
</table>
<table>
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">
<p>
⚠️ RESOURCES USED ALONG THIS SECTION
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">
<p>
From <a href="https://opal-demo.obiba.org/">https://opal-demo.obiba.org/</a> :
</p>
</td>
</tr>
<tr class="even">
<td style="text-align: left;">
<table>
<thead>
<tr class="header">
<th>
<p>
STUDY
</p>
</th>
<th>
<p>
TABLE
</p>
</th>
<th>
<p>
PROFILE
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
OMICS.gds
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr 
</tbody></table></td></tr>
</tbody>
</table>
<div id="setting-up" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Setting up</h2>
<p>To begin, we will describe how to setup the work environment of the researcher to follow the use cases, this setup includes the following steps:</p>
<ul>
<li>Connecting to the Opal server</li>
<li>Selecting the resources we need (genotype and phenotype)</li>
<li>Loading the data of the resources so we can begin working with it</li>
</ul>
<div id="connection-to-the-opal-server" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Connection to the Opal server</h3>
<p>We have to create an Opal connection object to the cohort server. We do that using the following functions.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="exploration-of-genotype-data.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># require(&#39;DSI&#39;)</span></span>
<span id="cb3-2"><a href="exploration-of-genotype-data.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># require(&#39;DSOpal&#39;)</span></span>
<span id="cb3-3"><a href="exploration-of-genotype-data.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># require(&#39;dsBaseClient&#39;)</span></span>
<span id="cb3-4"><a href="exploration-of-genotype-data.html#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># require(&#39;dsOmicsClient&#39;)</span></span>
<span id="cb3-5"><a href="exploration-of-genotype-data.html#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb3-6"><a href="exploration-of-genotype-data.html#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># builder &lt;- DSI::newDSLoginBuilder()</span></span>
<span id="cb3-7"><a href="exploration-of-genotype-data.html#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># builder$append(server = &quot;cohort1&quot;, url = &quot;https://opal-demo.obiba.org/&quot;,</span></span>
<span id="cb3-8"><a href="exploration-of-genotype-data.html#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                user =  &quot;dsuser&quot;, password = &quot;P@ssw0rd&quot;,</span></span>
<span id="cb3-9"><a href="exploration-of-genotype-data.html#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                driver = &quot;OpalDriver&quot;, profile = &quot;omics&quot;)</span></span>
<span id="cb3-10"><a href="exploration-of-genotype-data.html#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co"># builder$append(server = &quot;cohort2&quot;, url = &quot;https://opal-demo.obiba.org/&quot;,</span></span>
<span id="cb3-11"><a href="exploration-of-genotype-data.html#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                user =  &quot;dsuser&quot;, password = &quot;P@ssw0rd&quot;,</span></span>
<span id="cb3-12"><a href="exploration-of-genotype-data.html#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                driver = &quot;OpalDriver&quot;, profile = &quot;omics&quot;)</span></span>
<span id="cb3-13"><a href="exploration-of-genotype-data.html#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co"># builder$append(server = &quot;cohort3&quot;, url = &quot;https://opal-demo.obiba.org/&quot;,</span></span>
<span id="cb3-14"><a href="exploration-of-genotype-data.html#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                user =  &quot;dsuser&quot;, password = &quot;P@ssw0rd&quot;,</span></span>
<span id="cb3-15"><a href="exploration-of-genotype-data.html#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co">#                driver = &quot;OpalDriver&quot;, profile = &quot;omics&quot;)</span></span>
<span id="cb3-16"><a href="exploration-of-genotype-data.html#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co"># logindata &lt;- builder$build()</span></span>
<span id="cb3-17"><a href="exploration-of-genotype-data.html#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="co"># conns &lt;- DSI::datashield.login(logins = logindata)</span></span></code></pre></div>
</div>
<div id="assign-the-vcf-resources" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Assign the VCF resources</h3>
<p>Now that we have created a connection object to the Opal, we have started a new R session on the server, our analysis will take place in this remote session, so we have to load the data into it.</p>
<p>In this use case we will use a resource from the <code>OMICS</code> project hosted on the <a href="https://opal-demo.obiba.org/">demo Opal server</a>. This resource corresponds to a GDS file with Genotype information. The names of the resource is <code>gds</code>. Following the Opal syntax, we will refer to it using the string <code>OMICS.gds</code>.</p>
<p>To load the resource we will use the <code>DSI::datashield.assign.resource()</code> function.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="exploration-of-genotype-data.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # Cohort 1 resources</span></span>
<span id="cb4-2"><a href="exploration-of-genotype-data.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># lapply(1:21, function(x){</span></span>
<span id="cb4-3"><a href="exploration-of-genotype-data.html#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   DSI::datashield.assign.resource(conns[1], paste0(&quot;chr&quot;, x), paste0(&quot;GWAS.chr&quot;, x,&quot;A&quot;))</span></span>
<span id="cb4-4"><a href="exploration-of-genotype-data.html#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   })</span></span>
<span id="cb4-5"><a href="exploration-of-genotype-data.html#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb4-6"><a href="exploration-of-genotype-data.html#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># # Cohort 2 resources</span></span>
<span id="cb4-7"><a href="exploration-of-genotype-data.html#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="co"># lapply(1:21, function(x){</span></span>
<span id="cb4-8"><a href="exploration-of-genotype-data.html#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   DSI::datashield.assign.resource(conns[2], paste0(&quot;chr&quot;, x), paste0(&quot;GWAS.chr&quot;, x,&quot;B&quot;))</span></span>
<span id="cb4-9"><a href="exploration-of-genotype-data.html#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   })</span></span>
<span id="cb4-10"><a href="exploration-of-genotype-data.html#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb4-11"><a href="exploration-of-genotype-data.html#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co"># # Cohort 3 resources</span></span>
<span id="cb4-12"><a href="exploration-of-genotype-data.html#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co"># lapply(1:21, function(x){</span></span>
<span id="cb4-13"><a href="exploration-of-genotype-data.html#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   DSI::datashield.assign.resource(conns[3], paste0(&quot;chr&quot;, x), paste0(&quot;GWAS.chr&quot;, x,&quot;C&quot;))</span></span>
<span id="cb4-14"><a href="exploration-of-genotype-data.html#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   })</span></span></code></pre></div>
<p>Now we have assigned the resource named <code>OMICS.gds</code> into our remote R session into a variable called <code>geno</code>. To verify this step has been performed correctly, we could use the <code>ds.class</code> function to check for their class and that it exists on the remote session.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="exploration-of-genotype-data.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.class(&quot;chr1&quot;)</span></span></code></pre></div>
<p>To resolve the resource and retrieve the data in the remote session we will use the <code>DSI::datashield.assign.expr()</code> function. This function runs a line of code on the remote session,<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> and in particular we want to run the function <code>as.resource.object()</code>, which is the DataSHIELD function in charge of resolving the resource.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="exploration-of-genotype-data.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lapply(1:21, function(x){</span></span>
<span id="cb6-2"><a href="exploration-of-genotype-data.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   DSI::datashield.assign.expr(conns = conns, symbol = paste0(&quot;gds&quot;, x, &quot;_object&quot;),</span></span>
<span id="cb6-3"><a href="exploration-of-genotype-data.html#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                             expr = as.symbol(paste0(&quot;as.resource.object(chr&quot;, x, &quot;)&quot;)))</span></span>
<span id="cb6-4"><a href="exploration-of-genotype-data.html#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co"># })</span></span></code></pre></div>
<p>Now we have resolved the resource named <code>geno</code> into our remote R session. The object retrieved have been assigned into a variable named <code>gds</code>. We can check the process was successful as we did before.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="exploration-of-genotype-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.class(&quot;gds&quot;)</span></span></code></pre></div>
</div>
<div id="assign-the-phenotypes" class="section level3" number="3.1.3">
<h3><span class="header-section-number">3.1.3</span> Assign the phenotypes</h3>
<p>The object we have loaded into our remote session is a GDS file that containS genomic information of the individuals. Now we will include phenotype information and merge it together. The phenotypes information is a table that contains the individuals as rows and phenotypes as columns. In this use case, we will use a resource (as with the GDS file) to load the phenotypes table into the remote session.</p>
<table>
<colgroup>
<col width="74%" />
<col width="25%" />
</colgroup>
<thead>
<tr class="header">
<th>⚠️</th>
<th align="right">IMPORTANT INFORMATION</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td align="right">On this section, the phenotype information is used only on one occasion: when plotting the results of the PCA with a grouping variable. All the other functions can be used with what we have up to this point, you can skip to #basic-info and use <code>"gds"</code> instead of <code>"gds.Data"</code> if you are not interested on the PCA plot with groupings.</td>
</tr>
</tbody>
</table>
<p>The procedure is practically the same as before with some minor tweaks. Instead of using the function <code>as.resource.object</code>, we will use <code>as.resource.data.frame</code>, this is because before we were loading a special object (GDS file) and now we are loading a plain table, so the internal treatment on the remote session has to be different.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="exploration-of-genotype-data.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># # Cohort 1 phenotypes table</span></span>
<span id="cb8-2"><a href="exploration-of-genotype-data.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># DSI::datashield.assign.resource(conns[1], &quot;pheno&quot;, &quot;GWAS.ega_phenotypes_1&quot;)</span></span>
<span id="cb8-3"><a href="exploration-of-genotype-data.html#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb8-4"><a href="exploration-of-genotype-data.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co"># # Cohort 2 phenotypes table</span></span>
<span id="cb8-5"><a href="exploration-of-genotype-data.html#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># DSI::datashield.assign.resource(conns[2], &quot;pheno&quot;, &quot;GWAS.ega_phenotypes_2&quot;)</span></span>
<span id="cb8-6"><a href="exploration-of-genotype-data.html#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb8-7"><a href="exploration-of-genotype-data.html#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co"># # Cohort 3 phenotypes table</span></span>
<span id="cb8-8"><a href="exploration-of-genotype-data.html#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co"># DSI::datashield.assign.resource(conns[3], &quot;pheno&quot;, &quot;GWAS.ega_phenotypes_3&quot;)</span></span>
<span id="cb8-9"><a href="exploration-of-genotype-data.html#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb8-10"><a href="exploration-of-genotype-data.html#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co"># # Resolve phenotypes table</span></span>
<span id="cb8-11"><a href="exploration-of-genotype-data.html#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co"># DSI::datashield.assign.expr(conns = conns, symbol = &quot;pheno_object&quot;,</span></span>
<span id="cb8-12"><a href="exploration-of-genotype-data.html#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                             expr = quote(as.resource.data.frame(pheno)))</span></span></code></pre></div>
<p>We can follow the same analogy as before to know that we have assigned the phenotypes table to a variable called <code>pheno_object</code> on the remote R session.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="exploration-of-genotype-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.class(&quot;pheno_object&quot;)</span></span></code></pre></div>
<p>We can also check the column names to see which information is present on the table.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="exploration-of-genotype-data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.colnames(&quot;pheno_object&quot;)[[1]]</span></span></code></pre></div>
</div>
<div id="merge-gds-genotype-and-phenotype-information" class="section level3" number="3.1.4">
<h3><span class="header-section-number">3.1.4</span> Merge GDS (genotype) and phenotype information</h3>
<p>Arrived at this point, we have a GDS object and a phenotypes table on our remote session. The next step is to merge them. Before doing that however, we have to gather some information from the phenotypes table. The information to gather is summarized on the Table <a href="exploration-of-genotype-data.html#tab:tab2">3.2</a>.</p>
<table>
<caption><span id="tab:tab2">Table 3.2: </span> Information to gather from the phenotypes table.</caption>
<colgroup>
<col width="20%" />
<col width="79%" />
</colgroup>
<thead>
<tr class="header">
<th>Information</th>
<th>Details</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Which column has the samples identifier?</td>
<td>Column name that contains the IDs of the samples. Those are the IDs that will be matched to the ones present on the VCF objects.</td>
</tr>
<tr class="even">
<td>Which is the sex column on the covariates file?</td>
<td>Column name that contains the sex information.</td>
</tr>
<tr class="odd">
<td>How are males and how are females encoded into this column?</td>
<td>There is not a standard way to encode the sex information, some people use 0/1; male/female; etc. Our approach uses a library that requires a specific encoding, for that reason we need to know the original encoding to perform the translation.</td>
</tr>
</tbody>
</table>
<p>If we are not sure about the exact column names, we can use the function <code>ds.colnames</code> as we did before. Also, we can use the function <code>ds.table1D</code> to check the level names of the categorical variables.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="exploration-of-genotype-data.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.table1D(&quot;pheno_object$sex&quot;)$counts</span></span></code></pre></div>
<p>From the different functions we used, we can extract our answers (Summarized on the Table <a href="exploration-of-genotype-data.html#tab:tab3">3.3</a>)</p>
<table>
<caption><span id="tab:tab3">Table 3.3: </span> Summary of options from the example phenotypes table.</caption>
<colgroup>
<col width="68%" />
<col width="31%" />
</colgroup>
<thead>
<tr class="header">
<th>Question</th>
<th>Answer</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Which column has the samples identifier?</td>
<td>“subject_id”</td>
</tr>
<tr class="even">
<td>Which is the sex column on the covariates file?</td>
<td>“sex”</td>
</tr>
<tr class="odd">
<td>How are males and how are females encoded into this column?</td>
<td>“male”/“female”</td>
</tr>
</tbody>
</table>
<p>With all this information we can now merge the phenotypes and GDS object into a type of object named GenotypeData. We will use the <code>ds.GenotypeData</code> function.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="exploration-of-genotype-data.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lapply(1:21, function(x){</span></span>
<span id="cb12-2"><a href="exploration-of-genotype-data.html#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   ds.GenotypeData(x=paste0(&#39;gds&#39;, x,&#39;_object&#39;), covars = &#39;pheno_object&#39;, </span></span>
<span id="cb12-3"><a href="exploration-of-genotype-data.html#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                   columnId = &quot;subject_id&quot;, sexId = &quot;sex&quot;,</span></span>
<span id="cb12-4"><a href="exploration-of-genotype-data.html#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                   male_encoding = &quot;male&quot;, female_encoding = &quot;female&quot;,</span></span>
<span id="cb12-5"><a href="exploration-of-genotype-data.html#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                   case_control_column = &quot;diabetes_diagnosed_doctor&quot;, </span></span>
<span id="cb12-6"><a href="exploration-of-genotype-data.html#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                   case = &quot;Yes&quot;, control = &quot;No&quot;,</span></span>
<span id="cb12-7"><a href="exploration-of-genotype-data.html#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                   newobj.name = paste0(&#39;gds.Data&#39;, x), datasources = conns)</span></span>
<span id="cb12-8"><a href="exploration-of-genotype-data.html#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co"># })</span></span></code></pre></div>
<p>The object we just created is named <code>gds.Data</code> on the remote session.</p>
</div>
</div>
<div id="basic-info" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Basic information of the Genotype data</h2>
<p>Once we have everything loaded in the remote R sessions, we can begin a simple exploratory analysis by checking basic information. We can check:</p>
<ul>
<li>The number of variants and individuals</li>
<li>The chromosomes that are present on the data</li>
<li>The available phenotypes (only applies if we have merged phenotypes information)</li>
<li>Get the SNP rs IDs</li>
</ul>
<p>To get the number of variants and individuals:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="exploration-of-genotype-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># dimensions &lt;- do.call(rbind, lapply(1:21, function(x){</span></span>
<span id="cb13-2"><a href="exploration-of-genotype-data.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   data.frame(ds.genoDimensions(paste0(&#39;gds.Data&#39;, x)))</span></span>
<span id="cb13-3"><a href="exploration-of-genotype-data.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># }))</span></span>
<span id="cb13-4"><a href="exploration-of-genotype-data.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb13-5"><a href="exploration-of-genotype-data.html#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># data.frame(dimensions)</span></span></code></pre></div>
<p>To get the chromosomes that are present on the data:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="exploration-of-genotype-data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.getChromosomeNames(&#39;gds.Data1&#39;)</span></span></code></pre></div>
<p>To get the available phenotypes:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="exploration-of-genotype-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.varLabels(&#39;gds.Data1&#39;)</span></span></code></pre></div>
<p>To get the SNP rs IDs:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="exploration-of-genotype-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># head(ds.getSNPs(&#39;gds.Data1&#39;))</span></span></code></pre></div>
</div>
<div id="basic-statistics-of-the-genotype-data" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Basic statistics of the genotype data</h2>
<p>We have also developed functions to compute statistics of the genotype data, the statistics implemented on <code>dsOmicsClient</code> are:</p>
<ul>
<li>Allele frequencies. Brief definition (by Nature)[<a href="https://www.nature.com/scitable/definition/allele-frequency-298/#" class="uri">https://www.nature.com/scitable/definition/allele-frequency-298/#</a>:~:text=An%20allele%20frequency%20is%20calculated,a%20percentage%2C%20or%20a%20fraction.].</li>
<li>Hardy-Weinberg Equilibrium testing. Information on the test can be found (here)[<a href="https://www.sciencedirect.com/science/article/pii/S0002929707607356https://www.sciencedirect.com/science/article/pii/S0002929707607356" class="uri">https://www.sciencedirect.com/science/article/pii/S0002929707607356https://www.sciencedirect.com/science/article/pii/S0002929707607356</a>].</li>
</ul>
<p>To get the allele frequencies:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="exploration-of-genotype-data.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ds.alleleFrequency(paste0(&#39;gds.Data&#39;, 1:21), type = &quot;split&quot;)</span></span></code></pre></div>
<p>To do the Hardy-Weinberg Equilibrium testing:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="exploration-of-genotype-data.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># do.call(rbind, lapply(1:21, function(x){</span></span>
<span id="cb18-2"><a href="exploration-of-genotype-data.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="co">#   data.frame(ds.exactHWE(paste0(&#39;gds.Data&#39;, x)))</span></span>
<span id="cb18-3"><a href="exploration-of-genotype-data.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># }))</span></span></code></pre></div>
</div>
<div id="principal-component-analysis" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Principal component analysis</h2>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="exploration-of-genotype-data.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># pca_res &lt;- ds.PCA(paste0(&#39;gds.Data&#39;, 1:21), standardize = F, snpBlock = 10)</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="exploration-of-genotype-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plotPCA(pca_res, group = &quot;sex&quot;)</span></span></code></pre></div>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>The same methodology and code can be used with unitary VCF resources that contain the variant information of all chromosomes.<a href="exploration-of-genotype-data.html#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Omic.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="DatasetsDescription.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OmicSHIELD_bookdown.pdf", "OmicSHIELD_bookdown.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
