<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</title>
  <meta name="description" content="4 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Exploration of Genotype data | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  
  
  



<meta name="date" content="2023-06-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="DatasetsDescription.html"/>
<link rel="next" href="genome-wide-association-study-gwas-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script>
  function unrolltab(evt, divName, tabName) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementById(divName).getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
      tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementById(divName).getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
      tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(tabName).style.display = "block";
    evt.currentTarget.className += " active";
  }
</script>

<script>
  document.addEventListener("DOMContentLoaded", function(event) { 
    document.getElementById("gwas_results_1").click();
    document.getElementById("gwas_plot_1").click();
    document.getElementById("manhattan-plots-tab").click();
    document.getElementById("locuszoom-plots-tabs").click();
 });
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="" target="blank">OmicSHIELD</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#materials-to-read-beforehand"><i class="fa fa-check"></i><b>1.1</b> Materials to read beforehand</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#what-are-resources-a-very-simple-explanation-without-any-technicalities"><i class="fa fa-check"></i><b>1.2</b> What are “resources”: A very simple explanation without any technicalities</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#capabilities-of-omicshield"><i class="fa fa-check"></i><b>1.3</b> Capabilities of OmicSHIELD</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#opal-servers"><i class="fa fa-check"></i><b>1.4</b> Opal servers</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>2</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html"><i class="fa fa-check"></i><b>3</b> Datasets description</a>
<ul>
<li class="chapter" data-level="3.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAdescription"><i class="fa fa-check"></i><b>3.1</b> CINECA project</a></li>
<li class="chapter" data-level="3.2" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAopal"><i class="fa fa-check"></i><b>3.2</b> Data organization in the Opal server</a></li>
<li class="chapter" data-level="3.3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#TCGALiverdatadescription"><i class="fa fa-check"></i><b>3.3</b> TCGA Liver project</a></li>
<li class="chapter" data-level="3.4" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#GSE66351description"><i class="fa fa-check"></i><b>3.4</b> Epigenetic data from the GSE66351 project</a></li>
<li class="chapter" data-level="3.5" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXdescription"><i class="fa fa-check"></i><b>3.5</b> HELIX project</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXopal"><i class="fa fa-check"></i><b>3.5.1</b> Data organization in the Opal server</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html"><i class="fa fa-check"></i><b>4</b> Exploration of Genotype data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#connection-to-the-opal-server"><i class="fa fa-check"></i><b>4.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="4.2" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#multi-vcf"><i class="fa fa-check"></i><b>4.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="4.3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>4.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="4.4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#merge-vcf-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>4.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="4.5" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-info"><i class="fa fa-check"></i><b>4.5</b> Basic information of the Genotype data</a></li>
<li class="chapter" data-level="4.6" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-statistics-of-the-genotype-data"><i class="fa fa-check"></i><b>4.6</b> Basic statistics of the genotype data</a></li>
<li class="chapter" data-level="4.7" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.7</b> Principal component analysis</a></li>
</ul></li>
<li class="part"><span><b>II Workflows</b></span></li>
<li class="chapter" data-level="5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html"><i class="fa fa-check"></i><b>5</b> Genome wide association study (GWAS) analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#single-cohort"><i class="fa fa-check"></i><b>5.1</b> Single cohort</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-1"><i class="fa fa-check"></i><b>5.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.1.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-vcf-resources"><i class="fa fa-check"></i><b>5.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.1.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes"><i class="fa fa-check"></i><b>5.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.1.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#vcf_merge"><i class="fa fa-check"></i><b>5.1.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.1.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas"><i class="fa fa-check"></i><b>5.1.5</b> Performing the GWAS</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-cohorts"><i class="fa fa-check"></i><b>5.2</b> Multi cohorts</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>5.2.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-vcf2"><i class="fa fa-check"></i><b>5.2.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.2.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-10"><i class="fa fa-check"></i><b>5.2.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.2.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information-1"><i class="fa fa-check"></i><b>5.2.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.2.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1"><i class="fa fa-check"></i><b>5.2.5</b> Performing a meta - GWAS</a></li>
<li class="chapter" data-level="5.2.6" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#gwas-plots"><i class="fa fa-check"></i><b>5.2.6</b> GWAS Plots</a></li>
<li class="chapter" data-level="5.2.7" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control"><i class="fa fa-check"></i><b>5.2.7</b> Meta-level quality control</a></li>
<li class="chapter" data-level="5.2.8" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-analysis"><i class="fa fa-check"></i><b>5.2.8</b> Meta-analysis</a></li>
<li class="chapter" data-level="5.2.9" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas"><i class="fa fa-check"></i><b>5.2.9</b> Ultra fast pooled GWAS analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#post-omics"><i class="fa fa-check"></i><b>5.3</b> Post-omic analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#enrichment-analysis"><i class="fa fa-check"></i><b>5.3.1</b> Enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html"><i class="fa fa-check"></i><b>6</b> Polygenic risk scores</a>
<ul>
<li class="chapter" data-level="6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#definition"><i class="fa fa-check"></i><b>6.1</b> Definition</a></li>
<li class="chapter" data-level="6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-configuration"><i class="fa fa-check"></i><b>6.2</b> Study configuration</a></li>
<li class="chapter" data-level="6.3" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#source-of-polygenic-scores"><i class="fa fa-check"></i><b>6.3</b> Source of polygenic scores</a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>6.4</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="6.5" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#assign-the-vcf-resources-1"><i class="fa fa-check"></i><b>6.5</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="6.6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#calculate-the-prs"><i class="fa fa-check"></i><b>6.6</b> Calculate the PRS</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#pgs-catalog-example"><i class="fa fa-check"></i><b>6.6.1</b> PGS Catalog Example</a></li>
<li class="chapter" data-level="6.6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#custom-prs_table-example"><i class="fa fa-check"></i><b>6.6.2</b> Custom <code>prs_table</code> example</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-of-the-prs-results"><i class="fa fa-check"></i><b>6.7</b> Study of the PRS results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plink.html"><a href="plink.html"><i class="fa fa-check"></i><b>7</b> PLINK</a>
<ul>
<li class="chapter" data-level="7.1" data-path="plink.html"><a href="plink.html#connection-to-the-opal-server-3"><i class="fa fa-check"></i><b>7.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="7.2" data-path="plink.html"><a href="plink.html#build-the-plink-call"><i class="fa fa-check"></i><b>7.2</b> Build the PLINK call</a></li>
<li class="chapter" data-level="7.3" data-path="plink.html"><a href="plink.html#call-the-analysis"><i class="fa fa-check"></i><b>7.3</b> Call the analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="snptest.html"><a href="snptest.html"><i class="fa fa-check"></i><b>8</b> SNPTEST</a>
<ul>
<li class="chapter" data-level="8.1" data-path="snptest.html"><a href="snptest.html#connection-to-the-opal-server-4"><i class="fa fa-check"></i><b>8.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="8.2" data-path="snptest.html"><a href="snptest.html#build-the-snptest-call"><i class="fa fa-check"></i><b>8.2</b> Build the SNPTEST call</a></li>
<li class="chapter" data-level="8.3" data-path="snptest.html"><a href="snptest.html#call-the-analysis-1"><i class="fa fa-check"></i><b>8.3</b> Call the analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>9</b> Differential gene expression (DGE) analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#connection-to-the-opal-server-5"><i class="fa fa-check"></i><b>9.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="9.2" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#assign-the-rse-resource"><i class="fa fa-check"></i><b>9.2</b> Assign the RSE resource</a></li>
<li class="chapter" data-level="9.3" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#inspect-the-rse"><i class="fa fa-check"></i><b>9.3</b> Inspect the RSE</a></li>
<li class="chapter" data-level="9.4" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#pre-processing-for-rnaseq-data"><i class="fa fa-check"></i><b>9.4</b> Pre-processing for RNAseq data</a></li>
<li class="chapter" data-level="9.5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#dge-analysis"><i class="fa fa-check"></i><b>9.5</b> DGE analysis</a></li>
<li class="chapter" data-level="9.6" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>9.6</b> Surrogate variable analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>10</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#initial-steps-for-all-use-cases"><i class="fa fa-check"></i><b>10.1</b> Initial steps for all use cases</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#connection-to-the-opal-server-6"><i class="fa fa-check"></i><b>10.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="10.1.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#assign-the-expressionset-resource"><i class="fa fa-check"></i><b>10.1.2</b> Assign the <code>ExpressionSet</code> resource</a></li>
<li class="chapter" data-level="10.1.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#ewas-inspect"><i class="fa fa-check"></i><b>10.1.3</b> Inspect the <code>ExpressionSet</code></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-pooled-analysis"><i class="fa fa-check"></i><b>10.2</b> Single CpG pooled analysis</a></li>
<li class="chapter" data-level="10.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpgs-pooled-analysis"><i class="fa fa-check"></i><b>10.3</b> Multiple CpGs pooled analysis</a></li>
<li class="chapter" data-level="10.4" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#full-genome-meta-analysis"><i class="fa fa-check"></i><b>10.4</b> Full genome meta-analysis</a></li>
<li class="chapter" data-level="10.5" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>10.5</b> Adjusting for surrogate variables</a></li>
</ul></li>
<li class="part"><span><b>III Showcase on a real dataset</b></span></li>
<li class="chapter" data-level="11" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html"><i class="fa fa-check"></i><b>11</b> Transcriptomic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXone"><i class="fa fa-check"></i><b>11.1</b> Getting started.</a></li>
<li class="chapter" data-level="11.2" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXdatamanip"><i class="fa fa-check"></i><b>11.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="11.3" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXDGE"><i class="fa fa-check"></i><b>11.3</b> Differential gene expression analysis.</a></li>
<li class="chapter" data-level="11.4" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXsurrogateAnnot"><i class="fa fa-check"></i><b>11.4</b> Surrogate variable analysis.</a></li>
<li class="chapter" data-level="11.5" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXenrichment"><i class="fa fa-check"></i><b>11.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html"><i class="fa fa-check"></i><b>12</b> Epigenetic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="12.1" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_gettingstarted"><i class="fa fa-check"></i><b>12.1</b> Getting started.</a></li>
<li class="chapter" data-level="12.2" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_load"><i class="fa fa-check"></i><b>12.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="12.3" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis3"><i class="fa fa-check"></i><b>12.3</b> Full genome meta-analysis.</a></li>
<li class="chapter" data-level="12.4" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis4"><i class="fa fa-check"></i><b>12.4</b> Adjusting for surrogate variables.</a></li>
<li class="chapter" data-level="12.5" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIXenrichment"><i class="fa fa-check"></i><b>12.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="part"><span><b>IV Technical details</b></span></li>
<li class="chapter" data-level="13" data-path="fast-gwas.html"><a href="fast-gwas.html"><i class="fa fa-check"></i><b>13</b> Fast GWAS</a></li>
<li class="chapter" data-level="14" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>14</b> PCA</a></li>
<li class="chapter" data-level="15" data-path="differential-privacy.html"><a href="differential-privacy.html"><i class="fa fa-check"></i><b>15</b> Differential privacy</a></li>
<li class="chapter" data-level="16" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html"><i class="fa fa-check"></i><b>16</b> Internal structure of the PRS functionality</a>
<ul>
<li class="chapter" data-level="16.1" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#client"><i class="fa fa-check"></i><b>16.1</b> Client</a></li>
<li class="chapter" data-level="16.2" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-resolver"><i class="fa fa-check"></i><b>16.2</b> Server resolver</a></li>
<li class="chapter" data-level="16.3" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-analysis"><i class="fa fa-check"></i><b>16.3</b> Server analysis</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="compress.html"><a href="compress.html"><i class="fa fa-check"></i><b>17</b> Compression of GDS files vs. performance</a></li>
<li class="chapter" data-level="18" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html"><i class="fa fa-check"></i><b>18</b> Recommended R server specs</a>
<ul>
<li class="chapter" data-level="18.1" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#limma-voom"><i class="fa fa-check"></i><b>18.1</b> Limma + voom</a></li>
<li class="chapter" data-level="18.2" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#gwas-pca"><i class="fa fa-check"></i><b>18.2</b> GWAS + PCA</a></li>
<li class="chapter" data-level="18.3" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#general-remarks"><i class="fa fa-check"></i><b>18.3</b> General remarks</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="disclosure-filter-configurations.html"><a href="disclosure-filter-configurations.html"><i class="fa fa-check"></i><b>19</b> Disclosure filter configurations</a></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="20" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>20</b> Session info</a></li>
<li class="chapter" data-level="21" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>21</b> Contributors</a></li>
<li class="part"><span><b>VI References</b></span></li>
<li class="chapter" data-level="22" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>22</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="exploration-of-genotype-data" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Exploration of Genotype data<a href="exploration-of-genotype-data.html#exploration-of-genotype-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<table>
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">
<p>
⚠️ RESOURCES USED ALONG THIS SECTION
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">
<p>
From <a href="https://opal-demo.obiba.org/">https://opal-demo.obiba.org/</a> :
</p>
</td>
</tr>
<tr class="even">
<td style="text-align: left;">
<table>
<thead>
<tr class="header">
<th>
<p>
STUDY
</p>
</th>
<th>
<p>
TABLE
</p>
</th>
<th>
<p>
PROFILE
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.chr{1..21}A
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort2
</p>
</td>
<td>
<p>
GWAS.chr{1..21}B
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort3
</p>
</td>
<td>
<p>
GWAS.chr{1..21}C
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_1
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort2
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_2
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort3
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_3
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr 
</tbody></table></td></tr>
</tbody>
</table>
<p>Along this section we will illustrate how to get some information about the loaded genotype data and how to perform some basic exploration analysis. Three different aspects will be illustrated:</p>
<ul>
<li>Basic information retrieval</li>
<li>Basic statistics calculation</li>
<li>Principal component analysis</li>
</ul>
<p>Information on the data used can be found on the Section (#CINECAopal). We will be working on a virtual scenario with three different cohorts to visualize a real scenario. The same functions can be used in the case of a single cohort.</p>
<p>The study configuration for this section is the following:</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:multiCohortImage2"></span>
<img src="fig/multi_cohort.png" alt="Proposed infrastructure." width="736" />
<p class="caption">
Figure 4.1: Proposed infrastructure.
</p>
</div>
<div id="connection-to-the-opal-server" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Connection to the Opal server<a href="exploration-of-genotype-data.html#connection-to-the-opal-server" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have to create an Opal connection object to the different cohorts server. We do that using the following functions.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="exploration-of-genotype-data.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSI&#39;</span>)</span>
<span id="cb3-2"><a href="exploration-of-genotype-data.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSOpal&#39;</span>)</span>
<span id="cb3-3"><a href="exploration-of-genotype-data.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsBaseClient&#39;</span>)</span>
<span id="cb3-4"><a href="exploration-of-genotype-data.html#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsOmicsClient&#39;</span>)</span>
<span id="cb3-5"><a href="exploration-of-genotype-data.html#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="exploration-of-genotype-data.html#cb3-6" aria-hidden="true" tabindex="-1"></a>builder <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">newDSLoginBuilder</span>()</span>
<span id="cb3-7"><a href="exploration-of-genotype-data.html#cb3-7" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort1&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb3-8"><a href="exploration-of-genotype-data.html#cb3-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb3-9"><a href="exploration-of-genotype-data.html#cb3-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb3-10"><a href="exploration-of-genotype-data.html#cb3-10" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort2&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb3-11"><a href="exploration-of-genotype-data.html#cb3-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb3-12"><a href="exploration-of-genotype-data.html#cb3-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb3-13"><a href="exploration-of-genotype-data.html#cb3-13" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort3&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb3-14"><a href="exploration-of-genotype-data.html#cb3-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb3-15"><a href="exploration-of-genotype-data.html#cb3-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb3-16"><a href="exploration-of-genotype-data.html#cb3-16" aria-hidden="true" tabindex="-1"></a>logindata <span class="ot">&lt;-</span> builder<span class="sc">$</span><span class="fu">build</span>()</span>
<span id="cb3-17"><a href="exploration-of-genotype-data.html#cb3-17" aria-hidden="true" tabindex="-1"></a>conns <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">datashield.login</span>(<span class="at">logins =</span> logindata)</span></code></pre></div>
<p>It is important to note that in this use case, we are only using one server that contains all the resources (<a href="https://opal-demo.obiba.org/" class="uri">https://opal-demo.obiba.org/</a>), on this server there are all the resources that correspond to the different cohorts. On a more real scenario each one of the <code>builder$append</code> instructions would be connecting to a different server.</p>
</div>
<div id="multi-vcf" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Assign the VCF resources<a href="exploration-of-genotype-data.html#multi-vcf" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have created a connection object to the different Opals, we have started three R session, our analysis will take place on those remote sessions, so we have to load the data into them.</p>
<p>In this use case we will use 63 different resources from the <code>GWAS</code> project hosted on the <a href="https://opal-demo.obiba.org/">demo Opal server</a>. This resources correspond to VCF files with information on individual chromosomes (21 chromosomes per three cohorts). The names of the resources are <code>chrXXY</code> (where XX is the chromosome number and Y is the cohort A/B/C). Following the Opal syntax, we will refer to them using the string <code>GWAS.chrXXY</code>.</p>
<p>We have to refer specifically to each different server by using <code>conns[X]</code>, this allows us to communicate with the server of interest to indicate to it the resources that it has to load.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="exploration-of-genotype-data.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 1 resources</span></span>
<span id="cb4-2"><a href="exploration-of-genotype-data.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb4-3"><a href="exploration-of-genotype-data.html#cb4-3" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">1</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;A&quot;</span>))</span>
<span id="cb4-4"><a href="exploration-of-genotype-data.html#cb4-4" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb4-5"><a href="exploration-of-genotype-data.html#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="exploration-of-genotype-data.html#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 2 resources</span></span>
<span id="cb4-7"><a href="exploration-of-genotype-data.html#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb4-8"><a href="exploration-of-genotype-data.html#cb4-8" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">2</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;B&quot;</span>))</span>
<span id="cb4-9"><a href="exploration-of-genotype-data.html#cb4-9" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb4-10"><a href="exploration-of-genotype-data.html#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="exploration-of-genotype-data.html#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 3 resources</span></span>
<span id="cb4-12"><a href="exploration-of-genotype-data.html#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb4-13"><a href="exploration-of-genotype-data.html#cb4-13" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">3</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;C&quot;</span>))</span>
<span id="cb4-14"><a href="exploration-of-genotype-data.html#cb4-14" aria-hidden="true" tabindex="-1"></a>  })</span></code></pre></div>
<p>Now we have assigned all the resources named <code>GWAS.chrXXY</code> into our remote R session. We have assigned them to the variables called <code>chrXX</code>. To verify this step has been performed correctly, we could use the <code>ds.class</code> function to check for their class and that they exist on the remote sessions.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="exploration-of-genotype-data.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;chr1&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;GDSFileResourceClient&quot; &quot;FileResourceClient&quot;    &quot;ResourceClient&quot;       
[4] &quot;R6&quot;                   

$cohort2
[1] &quot;GDSFileResourceClient&quot; &quot;FileResourceClient&quot;    &quot;ResourceClient&quot;       
[4] &quot;R6&quot;                   

$cohort3
[1] &quot;GDSFileResourceClient&quot; &quot;FileResourceClient&quot;    &quot;ResourceClient&quot;       
[4] &quot;R6&quot;                   </code></pre>
<p>We can see that the object <code>chr1</code> exists in all the three servers.</p>
<p>Finally the resources are resolved to retrieve the data in the remote sessions.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="exploration-of-genotype-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb7-2"><a href="exploration-of-genotype-data.html#cb7-2" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.expr</span>(<span class="at">conns =</span> conns, <span class="at">symbol =</span> <span class="fu">paste0</span>(<span class="st">&quot;gds&quot;</span>, x, <span class="st">&quot;_object&quot;</span>),</span>
<span id="cb7-3"><a href="exploration-of-genotype-data.html#cb7-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">expr =</span> <span class="fu">as.symbol</span>(<span class="fu">paste0</span>(<span class="st">&quot;as.resource.object(chr&quot;</span>, x, <span class="st">&quot;)&quot;</span>)))</span>
<span id="cb7-4"><a href="exploration-of-genotype-data.html#cb7-4" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>Now we have resolved the resources named <code>chrXX</code> into our remote R sessions. The objects retrieved have been assigned into variables named <code>gdsXX_object</code>. We can check the process was successful as we did before.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="exploration-of-genotype-data.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;gds1_object&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;GdsGenotypeReader&quot;
attr(,&quot;package&quot;)
[1] &quot;GWASTools&quot;

$cohort2
[1] &quot;GdsGenotypeReader&quot;
attr(,&quot;package&quot;)
[1] &quot;GWASTools&quot;

$cohort3
[1] &quot;GdsGenotypeReader&quot;
attr(,&quot;package&quot;)
[1] &quot;GWASTools&quot;</code></pre>
</div>
<div id="assign-the-phenotypes-1" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Assign the phenotypes<a href="exploration-of-genotype-data.html#assign-the-phenotypes-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The objects we have loaded into our remote sessions are VCF files that contain genomic information of the individuals. To perform a GWAS this information has to be related to some phenotypes to extract relationships. Therefore, we need to load the phenotypes into the remote sessions. The phenotypes information is a table that contains the individuals as rows and phenotypes as columns. In this use case, we will use a resource (as with the VCF files) to load the phenotypes table into the remote sessions.</p>
<p>As with the VCF resources, here we load a different resource to each cohort, as it is expected that each cohort has only the phenotypes information regarding their individuals, therefore different tables. Note that we are assigning the resource to the same variable at each server (<code>pheno</code>), so we only need one function call to resolve all the resources together.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="exploration-of-genotype-data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 1 phenotypes table</span></span>
<span id="cb10-2"><a href="exploration-of-genotype-data.html#cb10-2" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">1</span>], <span class="st">&quot;pheno&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_1&quot;</span>)</span>
<span id="cb10-3"><a href="exploration-of-genotype-data.html#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="exploration-of-genotype-data.html#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 2 phenotypes table</span></span>
<span id="cb10-5"><a href="exploration-of-genotype-data.html#cb10-5" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">2</span>], <span class="st">&quot;pheno&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_2&quot;</span>)</span>
<span id="cb10-6"><a href="exploration-of-genotype-data.html#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="exploration-of-genotype-data.html#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 3 phenotypes table</span></span>
<span id="cb10-8"><a href="exploration-of-genotype-data.html#cb10-8" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">3</span>], <span class="st">&quot;pheno&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_3&quot;</span>)</span>
<span id="cb10-9"><a href="exploration-of-genotype-data.html#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="exploration-of-genotype-data.html#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Resolve phenotypes table</span></span>
<span id="cb10-11"><a href="exploration-of-genotype-data.html#cb10-11" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.expr</span>(<span class="at">conns =</span> conns, <span class="at">symbol =</span> <span class="st">&quot;pheno_object&quot;</span>,</span>
<span id="cb10-12"><a href="exploration-of-genotype-data.html#cb10-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">expr =</span> <span class="fu">quote</span>(<span class="fu">as.resource.data.frame</span>(pheno)))</span></code></pre></div>
<p>We can follow the same analogy as before to know that we have assigned the phenotypes table to a variable called <code>pheno_object</code> on the remote R session.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="exploration-of-genotype-data.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;pheno_object&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; 

$cohort2
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; 

$cohort3
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; </code></pre>
<p>We can also check the column names to see which information is present on the table.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="exploration-of-genotype-data.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.colnames</span>(<span class="st">&quot;pheno_object&quot;</span>)[[<span class="dv">1</span>]]</span></code></pre></div>
<pre><code> [1] &quot;age_cancer_diagnosis&quot;                &quot;age_recruitment&quot;                    
 [3] &quot;alcohol_day&quot;                         &quot;basophill_count&quot;                    
 [5] &quot;behaviour_tumour&quot;                    &quot;bmi&quot;                                
 [7] &quot;c_reactive_protein_reportability&quot;    &quot;record_origin&quot;                      
 [9] &quot;report_format&quot;                       &quot;cholesterol&quot;                        
[11] &quot;birth_country&quot;                       &quot;actual_tobacco_smoker&quot;              
[13] &quot;date_diagnosis&quot;                      &quot;diabetes_diagnosed_doctor&quot;          
[15] &quot;diastolic_blood_pressure&quot;            &quot;duration_moderate_activity&quot;         
[17] &quot;duration_vigorous_activity&quot;          &quot;energy&quot;                             
[19] &quot;eosinophill_count&quot;                   &quot;ethnicity&quot;                          
[21] &quot;ever_smoked&quot;                         &quot;fasting_time&quot;                       
[23] &quot;alcohol_frequency&quot;                   &quot;hdl_cholesterol&quot;                    
[25] &quot;hip_circumference&quot;                   &quot;histology_tumour&quot;                   
[27] &quot;home_e_coordinate&quot;                   &quot;home_n_coordinate&quot;                  
[29] &quot;e_house_score&quot;                       &quot;s_house_score&quot;                      
[31] &quot;w_house_score&quot;                       &quot;e_income_score&quot;                     
[33] &quot;w_income_score&quot;                      &quot;ldl_direct&quot;                         
[35] &quot;lymphocyte_count&quot;                    &quot;monocyte_count&quot;                     
[37] &quot;neutrophill_count&quot;                   &quot;n_accidental_death_close_gen_family&quot;
[39] &quot;household_count&quot;                     &quot;children_count&quot;                     
[41] &quot;operations_count&quot;                    &quot;operative_procedures&quot;               
[43] &quot;past_tobacco&quot;                        &quot;platelet_count&quot;                     
[45] &quot;pulse_rate&quot;                          &quot;qualification&quot;                      
[47] &quot;cancer_occurrences&quot;                  &quot;reticulocuyte_count&quot;                
[49] &quot;sleep_duration&quot;                      &quot;height&quot;                             
[51] &quot;subject_id&quot;                          &quot;triglycerides&quot;                      
[53] &quot;cancer_type_idc10&quot;                   &quot;cancer_type_idc9&quot;                   
[55] &quot;weight&quot;                              &quot;leukocyte_count&quot;                    
[57] &quot;sex&quot;                                 &quot;phenotype&quot;                          
[59] &quot;age_death&quot;                           &quot;age_diagnosis_diabetes&quot;             
[61] &quot;date_k85_report&quot;                     &quot;date_k86_report&quot;                    
[63] &quot;date_death&quot;                          &quot;enm_diseases&quot;                       
[65] &quot;source_k85_report&quot;                   &quot;source_k86_report&quot;                  
[67] &quot;age_hbp_diagnosis&quot;                   &quot;mental_disorders&quot;                   
[69] &quot;respiratory_disorders&quot;               &quot;cancer_year&quot;                        
[71] &quot;circulatory_disorders&quot;               &quot;digestive_disorders&quot;                
[73] &quot;nervous_disorders&quot;                   &quot;immune_disorders&quot;                   </code></pre>
</div>
<div id="merge-vcf-genotype-and-phenotype-information" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Merge VCF (genotype) and phenotype information<a href="exploration-of-genotype-data.html#merge-vcf-genotype-and-phenotype-information" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Arrived at this point, we have 21 VCF objects at each cohort R session (each one corresponds to a chromosome) and a phenotypes table. The next step is merging each of the VCF objects with the phenotypes table. The same procedure as the <a href="genome-wide-association-study-gwas-analysis.html#vcf_merge">single cohort</a> can be applied to extract the arguments required.</p>
<p>With all this information we can now merge the phenotypes and VCF objects into a type of object named GenotypeData. We will use the <code>ds.GenotypeData</code> function.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="exploration-of-genotype-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb15-2"><a href="exploration-of-genotype-data.html#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ds.GenotypeData</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&#39;gds&#39;</span>, x,<span class="st">&#39;_object&#39;</span>), <span class="at">covars =</span> <span class="st">&#39;pheno_object&#39;</span>, </span>
<span id="cb15-3"><a href="exploration-of-genotype-data.html#cb15-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">columnId =</span> <span class="st">&quot;subject_id&quot;</span>, <span class="at">sexId =</span> <span class="st">&quot;sex&quot;</span>,</span>
<span id="cb15-4"><a href="exploration-of-genotype-data.html#cb15-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">male_encoding =</span> <span class="st">&quot;male&quot;</span>, <span class="at">female_encoding =</span> <span class="st">&quot;female&quot;</span>,</span>
<span id="cb15-5"><a href="exploration-of-genotype-data.html#cb15-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">case_control_column =</span> <span class="st">&quot;diabetes_diagnosed_doctor&quot;</span>, </span>
<span id="cb15-6"><a href="exploration-of-genotype-data.html#cb15-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">case =</span> <span class="st">&quot;Yes&quot;</span>, <span class="at">control =</span> <span class="st">&quot;No&quot;</span>,</span>
<span id="cb15-7"><a href="exploration-of-genotype-data.html#cb15-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">newobj.name =</span> <span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, x), <span class="at">datasources =</span> conns)</span>
<span id="cb15-8"><a href="exploration-of-genotype-data.html#cb15-8" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>The objects we just created are named <code>gds.DataXX</code> on the remote session. Now we are ready to perform the exploration analysis on the data we just loaded on the remote sessions.</p>
</div>
<div id="basic-info" class="section level2 hasAnchor" number="4.5">
<h2><span class="header-section-number">4.5</span> Basic information of the Genotype data<a href="exploration-of-genotype-data.html#basic-info" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once we have everything loaded in the remote R sessions, we can begin a simple exploratory analysis by checking basic information. We can check the following:</p>
<ul>
<li>The number of variants, chromosome and individuals</li>
<li>The available phenotypes (only applies if we have merged phenotypes information)</li>
<li>Get the available SNP rs IDs</li>
</ul>
<p>To get the number of variants and individuals:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="exploration-of-genotype-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a>dimensions <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb16-2"><a href="exploration-of-genotype-data.html#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="fu">ds.genoDimensions</span>(<span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, x)))</span>
<span id="cb16-3"><a href="exploration-of-genotype-data.html#cb16-3" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb16-4"><a href="exploration-of-genotype-data.html#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="exploration-of-genotype-data.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(dimensions)</span></code></pre></div>
<pre><code>   cohort1.snp_number cohort1.scan_number cohort1.chromosomes
1               69806                 817                   1
2               74985                 817                   2
3               53164                 817                   3
4               56259                 817                   4
5               51985                 817                   5
6               54835                 817                   6
7               53037                 817                   7
8               50372                 817                   8
9               40920                 817                   9
10              47075                 817                  10
11              44979                 817                  11
12              43477                 817                  12
13              33411                 817                  13
14              29742                 817                  14
15              26905                 817                  15
16              29183                 817                  16
17              24546                 817                  17
18              26353                 817                  18
19              20408                 817                  19
20              20771                 817                  20
21              13079                 817                  21
   cohort2.snp_number cohort2.scan_number cohort2.chromosomes
1               69806                1073                   1
2               74985                1073                   2
3               53164                1073                   3
4               56259                1073                   4
5               51985                1073                   5
6               54835                1073                   6
7               53037                1073                   7
8               50372                1073                   8
9               40920                1073                   9
10              47075                1073                  10
11              44979                1073                  11
12              43477                1073                  12
13              33411                1073                  13
14              29742                1073                  14
15              26905                1073                  15
16              29183                1073                  16
17              24546                1073                  17
18              26353                1073                  18
19              20408                1073                  19
20              20771                1073                  20
21              13079                1073                  21
   cohort3.snp_number cohort3.scan_number cohort3.chromosomes
1               69806                 557                   1
2               74985                 557                   2
3               53164                 557                   3
4               56259                 557                   4
5               51985                 557                   5
6               54835                 557                   6
7               53037                 557                   7
8               50372                 557                   8
9               40920                 557                   9
10              47075                 557                  10
11              44979                 557                  11
12              43477                 557                  12
13              33411                 557                  13
14              29742                 557                  14
15              26905                 557                  15
16              29183                 557                  16
17              24546                 557                  17
18              26353                 557                  18
19              20408                 557                  19
20              20771                 557                  20
21              13079                 557                  21</code></pre>
<p>To get the available phenotypes:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="exploration-of-genotype-data.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.varLabels</span>(<span class="st">&#39;gds.Data1&#39;</span>)</span></code></pre></div>
<pre><code>$cohort1
 [1] &quot;age_cancer_diagnosis&quot;                &quot;age_recruitment&quot;                    
 [3] &quot;alcohol_day&quot;                         &quot;basophill_count&quot;                    
 [5] &quot;behaviour_tumour&quot;                    &quot;bmi&quot;                                
 [7] &quot;c_reactive_protein_reportability&quot;    &quot;record_origin&quot;                      
 [9] &quot;report_format&quot;                       &quot;cholesterol&quot;                        
[11] &quot;birth_country&quot;                       &quot;actual_tobacco_smoker&quot;              
[13] &quot;date_diagnosis&quot;                      &quot;diabetes_diagnosed_doctor&quot;          
[15] &quot;diastolic_blood_pressure&quot;            &quot;duration_moderate_activity&quot;         
[17] &quot;duration_vigorous_activity&quot;          &quot;energy&quot;                             
[19] &quot;eosinophill_count&quot;                   &quot;ethnicity&quot;                          
[21] &quot;ever_smoked&quot;                         &quot;fasting_time&quot;                       
[23] &quot;alcohol_frequency&quot;                   &quot;hdl_cholesterol&quot;                    
[25] &quot;hip_circumference&quot;                   &quot;histology_tumour&quot;                   
[27] &quot;home_e_coordinate&quot;                   &quot;home_n_coordinate&quot;                  
[29] &quot;e_house_score&quot;                       &quot;s_house_score&quot;                      
[31] &quot;w_house_score&quot;                       &quot;e_income_score&quot;                     
[33] &quot;w_income_score&quot;                      &quot;ldl_direct&quot;                         
[35] &quot;lymphocyte_count&quot;                    &quot;monocyte_count&quot;                     
[37] &quot;neutrophill_count&quot;                   &quot;n_accidental_death_close_gen_family&quot;
[39] &quot;household_count&quot;                     &quot;children_count&quot;                     
[41] &quot;operations_count&quot;                    &quot;operative_procedures&quot;               
[43] &quot;past_tobacco&quot;                        &quot;platelet_count&quot;                     
[45] &quot;pulse_rate&quot;                          &quot;qualification&quot;                      
[47] &quot;cancer_occurrences&quot;                  &quot;reticulocuyte_count&quot;                
[49] &quot;sleep_duration&quot;                      &quot;height&quot;                             
[51] &quot;scanID&quot;                              &quot;triglycerides&quot;                      
[53] &quot;cancer_type_idc10&quot;                   &quot;cancer_type_idc9&quot;                   
[55] &quot;weight&quot;                              &quot;leukocyte_count&quot;                    
[57] &quot;sex&quot;                                 &quot;phenotype&quot;                          
[59] &quot;age_death&quot;                           &quot;age_diagnosis_diabetes&quot;             
[61] &quot;date_k85_report&quot;                     &quot;date_k86_report&quot;                    
[63] &quot;date_death&quot;                          &quot;enm_diseases&quot;                       
[65] &quot;source_k85_report&quot;                   &quot;source_k86_report&quot;                  
[67] &quot;age_hbp_diagnosis&quot;                   &quot;mental_disorders&quot;                   
[69] &quot;respiratory_disorders&quot;               &quot;cancer_year&quot;                        
[71] &quot;circulatory_disorders&quot;               &quot;digestive_disorders&quot;                
[73] &quot;nervous_disorders&quot;                   &quot;immune_disorders&quot;                   

$cohort2
 [1] &quot;age_cancer_diagnosis&quot;                &quot;age_recruitment&quot;                    
 [3] &quot;alcohol_day&quot;                         &quot;basophill_count&quot;                    
 [5] &quot;behaviour_tumour&quot;                    &quot;bmi&quot;                                
 [7] &quot;c_reactive_protein_reportability&quot;    &quot;record_origin&quot;                      
 [9] &quot;report_format&quot;                       &quot;cholesterol&quot;                        
[11] &quot;birth_country&quot;                       &quot;actual_tobacco_smoker&quot;              
[13] &quot;date_diagnosis&quot;                      &quot;diabetes_diagnosed_doctor&quot;          
[15] &quot;diastolic_blood_pressure&quot;            &quot;duration_moderate_activity&quot;         
[17] &quot;duration_vigorous_activity&quot;          &quot;energy&quot;                             
[19] &quot;eosinophill_count&quot;                   &quot;ethnicity&quot;                          
[21] &quot;ever_smoked&quot;                         &quot;fasting_time&quot;                       
[23] &quot;alcohol_frequency&quot;                   &quot;hdl_cholesterol&quot;                    
[25] &quot;hip_circumference&quot;                   &quot;histology_tumour&quot;                   
[27] &quot;home_e_coordinate&quot;                   &quot;home_n_coordinate&quot;                  
[29] &quot;e_house_score&quot;                       &quot;s_house_score&quot;                      
[31] &quot;w_house_score&quot;                       &quot;e_income_score&quot;                     
[33] &quot;w_income_score&quot;                      &quot;ldl_direct&quot;                         
[35] &quot;lymphocyte_count&quot;                    &quot;monocyte_count&quot;                     
[37] &quot;neutrophill_count&quot;                   &quot;n_accidental_death_close_gen_family&quot;
[39] &quot;household_count&quot;                     &quot;children_count&quot;                     
[41] &quot;operations_count&quot;                    &quot;operative_procedures&quot;               
[43] &quot;past_tobacco&quot;                        &quot;platelet_count&quot;                     
[45] &quot;pulse_rate&quot;                          &quot;qualification&quot;                      
[47] &quot;cancer_occurrences&quot;                  &quot;reticulocuyte_count&quot;                
[49] &quot;sleep_duration&quot;                      &quot;height&quot;                             
[51] &quot;scanID&quot;                              &quot;triglycerides&quot;                      
[53] &quot;cancer_type_idc10&quot;                   &quot;cancer_type_idc9&quot;                   
[55] &quot;weight&quot;                              &quot;leukocyte_count&quot;                    
[57] &quot;sex&quot;                                 &quot;phenotype&quot;                          
[59] &quot;age_death&quot;                           &quot;age_diagnosis_diabetes&quot;             
[61] &quot;date_k85_report&quot;                     &quot;date_k86_report&quot;                    
[63] &quot;date_death&quot;                          &quot;enm_diseases&quot;                       
[65] &quot;source_k85_report&quot;                   &quot;source_k86_report&quot;                  
[67] &quot;age_hbp_diagnosis&quot;                   &quot;mental_disorders&quot;                   
[69] &quot;respiratory_disorders&quot;               &quot;cancer_year&quot;                        
[71] &quot;circulatory_disorders&quot;               &quot;digestive_disorders&quot;                
[73] &quot;nervous_disorders&quot;                   &quot;immune_disorders&quot;                   

$cohort3
 [1] &quot;age_cancer_diagnosis&quot;                &quot;age_recruitment&quot;                    
 [3] &quot;alcohol_day&quot;                         &quot;basophill_count&quot;                    
 [5] &quot;behaviour_tumour&quot;                    &quot;bmi&quot;                                
 [7] &quot;c_reactive_protein_reportability&quot;    &quot;record_origin&quot;                      
 [9] &quot;report_format&quot;                       &quot;cholesterol&quot;                        
[11] &quot;birth_country&quot;                       &quot;actual_tobacco_smoker&quot;              
[13] &quot;date_diagnosis&quot;                      &quot;diabetes_diagnosed_doctor&quot;          
[15] &quot;diastolic_blood_pressure&quot;            &quot;duration_moderate_activity&quot;         
[17] &quot;duration_vigorous_activity&quot;          &quot;energy&quot;                             
[19] &quot;eosinophill_count&quot;                   &quot;ethnicity&quot;                          
[21] &quot;ever_smoked&quot;                         &quot;fasting_time&quot;                       
[23] &quot;alcohol_frequency&quot;                   &quot;hdl_cholesterol&quot;                    
[25] &quot;hip_circumference&quot;                   &quot;histology_tumour&quot;                   
[27] &quot;home_e_coordinate&quot;                   &quot;home_n_coordinate&quot;                  
[29] &quot;e_house_score&quot;                       &quot;s_house_score&quot;                      
[31] &quot;w_house_score&quot;                       &quot;e_income_score&quot;                     
[33] &quot;w_income_score&quot;                      &quot;ldl_direct&quot;                         
[35] &quot;lymphocyte_count&quot;                    &quot;monocyte_count&quot;                     
[37] &quot;neutrophill_count&quot;                   &quot;n_accidental_death_close_gen_family&quot;
[39] &quot;household_count&quot;                     &quot;children_count&quot;                     
[41] &quot;operations_count&quot;                    &quot;operative_procedures&quot;               
[43] &quot;past_tobacco&quot;                        &quot;platelet_count&quot;                     
[45] &quot;pulse_rate&quot;                          &quot;qualification&quot;                      
[47] &quot;cancer_occurrences&quot;                  &quot;reticulocuyte_count&quot;                
[49] &quot;sleep_duration&quot;                      &quot;height&quot;                             
[51] &quot;scanID&quot;                              &quot;triglycerides&quot;                      
[53] &quot;cancer_type_idc10&quot;                   &quot;cancer_type_idc9&quot;                   
[55] &quot;weight&quot;                              &quot;leukocyte_count&quot;                    
[57] &quot;sex&quot;                                 &quot;phenotype&quot;                          
[59] &quot;age_death&quot;                           &quot;age_diagnosis_diabetes&quot;             
[61] &quot;date_k85_report&quot;                     &quot;date_k86_report&quot;                    
[63] &quot;date_death&quot;                          &quot;enm_diseases&quot;                       
[65] &quot;source_k85_report&quot;                   &quot;source_k86_report&quot;                  
[67] &quot;age_hbp_diagnosis&quot;                   &quot;mental_disorders&quot;                   
[69] &quot;respiratory_disorders&quot;               &quot;cancer_year&quot;                        
[71] &quot;circulatory_disorders&quot;               &quot;digestive_disorders&quot;                
[73] &quot;nervous_disorders&quot;                   &quot;immune_disorders&quot;                   

attr(,&quot;class&quot;)
[1] &quot;dsvarLabels&quot; &quot;list&quot;       </code></pre>
<p>To get the SNP rs IDs:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="exploration-of-genotype-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">ds.getSNPs</span>(<span class="st">&#39;gds.Data1&#39;</span>))</span></code></pre></div>
<pre><code>[1] &quot;rs575272151&quot; &quot;rs75454623&quot;  &quot;rs2691315&quot;   &quot;rs2691277&quot;   &quot;rs114420996&quot;
[6] &quot;rs62637817&quot; </code></pre>
</div>
<div id="basic-statistics-of-the-genotype-data" class="section level2 hasAnchor" number="4.6">
<h2><span class="header-section-number">4.6</span> Basic statistics of the genotype data<a href="exploration-of-genotype-data.html#basic-statistics-of-the-genotype-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have also developed functions to compute statistics of the genotype data, the statistics implemented on <code>dsOmicsClient</code> are:</p>
<ul>
<li>Allele frequencies. Brief definition (by Nature)[<a href="https://www.nature.com/scitable/definition/allele-frequency-298/#" class="uri">https://www.nature.com/scitable/definition/allele-frequency-298/#</a>:~:text=An%20allele%20frequency%20is%20calculated,a%20percentage%2C%20or%20a%20fraction.].</li>
<li>Hardy-Weinberg Equilibrium testing. Information on the test can be found (here)[<a href="https://www.sciencedirect.com/science/article/pii/S0002929707607356https://www.sciencedirect.com/science/article/pii/S0002929707607356" class="uri">https://www.sciencedirect.com/science/article/pii/S0002929707607356https://www.sciencedirect.com/science/article/pii/S0002929707607356</a>].</li>
</ul>
<p>To get the allele frequencies we have two different methods, to obtain the pooled frequencies or to obtain them by study server:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="exploration-of-genotype-data.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pooled allele frequencies</span></span>
<span id="cb22-2"><a href="exploration-of-genotype-data.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.alleleFrequency</span>(<span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>))</span></code></pre></div>
<pre><code># A tibble: 762,900 x 3
   rs              n pooled_MAF
   &lt;chr&gt;       &lt;dbl&gt;      &lt;dbl&gt;
 1 rs575272151  2435     0.0863
 2 rs75454623   2428     0.482 
 3 rs2691315    2429     0.409 
 4 rs75478250   2432     0.355 
 5 rs940550     2429     0.297 
 6 rs144169752  2431     0.120 
 7 rs368130516  2429     0.0593
 8 rs202206030  2433     0.0724
 9 rs2853819    2428     0.202 
10 rs202025951  2429     0.184 
# ... with 762,890 more rows</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="exploration-of-genotype-data.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># By-study allele frquencies</span></span>
<span id="cb24-2"><a href="exploration-of-genotype-data.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.alleleFrequency</span>(<span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>), <span class="at">type =</span> <span class="st">&quot;split&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
# A tibble: 853,933 x 3
   rs              n    MAF
 * &lt;chr&gt;       &lt;dbl&gt;  &lt;dbl&gt;
 1 rs575272151   813 0.0795
 2 rs75454623    812 0.476 
 3 rs2691315     812 0.432 
 4 rs2691277     813 0.0861
 5 rs114420996   812 0.131 
 6 rs62637817    812 0.0686
 7 rs75478250    815 0.348 
 8 rs940550      812 0.314 
 9 rs377100675   814 0.259 
10 rs62642131    812 0.137 
# ... with 853,923 more rows

$cohort2
# A tibble: 855,153 x 3
   rs              n    MAF
 * &lt;chr&gt;       &lt;dbl&gt;  &lt;dbl&gt;
 1 rs575272151  1069 0.0965
 2 rs75454623   1064 0.491 
 3 rs2691315    1065 0.433 
 4 rs2691277    1066 0.0982
 5 rs114420996  1064 0.120 
 6 rs62637817   1064 0.0543
 7 rs75478250   1064 0.364 
 8 rs940550     1065 0.336 
 9 rs377100675  1066 0.273 
10 rs62642131   1063 0.129 
# ... with 855,143 more rows

$cohort3
# A tibble: 778,123 x 3
   rs              n    MAF
 * &lt;chr&gt;       &lt;dbl&gt;  &lt;dbl&gt;
 1 rs575272151   553 0.0753
 2 rs75454623    552 0.477 
 3 rs2691315     552 0.331 
 4 rs75478250    553 0.349 
 5 rs940550      552 0.199 
 6 rs144169752   553 0.132 
 7 rs368130516   553 0.0628
 8 rs202206030   553 0.0558
 9 rs2853819     553 0.236 
10 rs202025951   554 0.117 
# ... with 778,113 more rows

attr(,&quot;class&quot;)
[1] &quot;dsalleleFrequency&quot; &quot;list&quot;             </code></pre>
<p>To do the Hardy-Weinberg Equilibrium testing is not implemented to get the pooled results as the allele frequency, we can only get the results from each server independently.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="exploration-of-genotype-data.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb26-2"><a href="exploration-of-genotype-data.html#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="fu">ds.exactHWE</span>(<span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, x)))</span>
<span id="cb26-3"><a href="exploration-of-genotype-data.html#cb26-3" aria-hidden="true" tabindex="-1"></a>}))</span></code></pre></div>
<pre><code># A tibble: 865,292 x 3
   cohort1$rs  $chr      $pval cohort2$rs  $chr      $pval cohort3$rs  $chr 
   &lt;chr&gt;       &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt;
 1 rs575272151 1     8.10e-  1 rs575272151 1     2.89e-  1 rs575272151 1    
 2 rs75454623  1     5.90e-115 rs75454623  1     1.22e-160 rs75454623  1    
 3 rs2691315   1     8.30e-  1 rs2691315   1     3.82e-  1 rs2691315   1    
 4 rs2691277   1     4.02e-  2 rs2691277   1     2.67e-  3 rs2691277   1    
 5 rs114420996 1     8.78e-  1 rs114420996 1     1.00e+  0 rs114420996 1    
 6 rs62637817  1     2.85e-  5 rs62637817  1     2.94e-  2 rs62637817  1    
 7 rs75478250  1     4.45e-  2 rs75478250  1     4.27e-  1 rs75478250  1    
 8 rs940550    1     4.36e-  4 rs940550    1     2.32e-  2 rs940550    1    
 9 rs377100675 1     5.79e- 43 rs377100675 1     3.34e- 57 rs377100675 1    
10 rs62642131  1     4.16e-  5 rs62642131  1     2.01e-  9 rs62642131  1    
# ... with 865,282 more rows</code></pre>
</div>
<div id="principal-component-analysis" class="section level2 hasAnchor" number="4.7">
<h2><span class="header-section-number">4.7</span> Principal component analysis<a href="exploration-of-genotype-data.html#principal-component-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have implemented a pooled component analysis method into OmicSHIELD. This function will create new phenotypes on the GenotypeData objects with the principal components results, so we can later adjust our analysis.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="exploration-of-genotype-data.html#cb28-1" aria-hidden="true" tabindex="-1"></a>pca_res <span class="ot">&lt;-</span> <span class="fu">ds.PCA</span>(<span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>), <span class="at">standardize =</span> F)</span></code></pre></div>
<p>The principal component results can also be visualized using the available grouping variables (categorical) present on the GenotypeData objects used on the <code>ds.PCA</code> function.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="exploration-of-genotype-data.html#cb29-1" aria-hidden="true" tabindex="-1"></a>dsOmicsClient<span class="sc">::</span><span class="fu">plotPCA</span>(pca_res, <span class="at">group =</span> <span class="st">&quot;sex&quot;</span>)</span></code></pre></div>
<p>Or we can visualize the results with no groupings.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="exploration-of-genotype-data.html#cb30-1" aria-hidden="true" tabindex="-1"></a>dsOmicsClient<span class="sc">::</span><span class="fu">plotPCA</span>(pca_res)</span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="exploration-of-genotype-data.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datashield.logout</span>(conns)</span></code></pre></div>
</div>
</div>



</div>
            </section>

          </div>
        </div>
      </div>
<a href="DatasetsDescription.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genome-wide-association-study-gwas-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OmicSHIELD_bookdown.pdf", "OmicSHIELD_bookdown.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
