<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Genome wide association study (GWAS) analysis | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</title>
  <meta name="description" content="5 Genome wide association study (GWAS) analysis | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Genome wide association study (GWAS) analysis | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Genome wide association study (GWAS) analysis | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  
  
  



<meta name="date" content="2022-03-18" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="DatasetsDescription.html"/>
<link rel="next" href="polygenic-risk-scores.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script>
  function unrolltab(evt, divName, tabName) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementById(divName).getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
      tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementById(divName).getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
      tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(tabName).style.display = "block";
    evt.currentTarget.className += " active";
  }
</script>

<script>
  document.addEventListener("DOMContentLoaded", function(event) { 
    document.getElementById("gwas_results_1").click();
    document.getElementById("gwas_plot_1").click();
    document.getElementById("manhattan-plots-tab").click();
    document.getElementById("locuszoom-plots-tabs").click();
 });
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="" target="blank">OmicSHIELD</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#materials-to-read-beforehand"><i class="fa fa-check"></i><b>1.1</b> Materials to read beforehand</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#what-are-resources-a-very-simple-explanation-without-any-technicalities"><i class="fa fa-check"></i><b>1.2</b> What are “resources”: A very simple explanation without any technicalities</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#capabilities-of-omicshield"><i class="fa fa-check"></i><b>1.3</b> Capabilities of OmicSHIELD</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#opal-servers"><i class="fa fa-check"></i><b>1.4</b> Opal servers</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>2</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html"><i class="fa fa-check"></i><b>3</b> Exploration of Genotype data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#setting-up"><i class="fa fa-check"></i><b>3.1</b> Setting up</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#connection-to-the-opal-server"><i class="fa fa-check"></i><b>3.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="3.1.2" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#assign-the-vcf-resources"><i class="fa fa-check"></i><b>3.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="3.1.3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#assign-the-phenotypes"><i class="fa fa-check"></i><b>3.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="3.1.4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#merge-gds-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>3.1.4</b> Merge GDS (genotype) and phenotype information</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-info"><i class="fa fa-check"></i><b>3.2</b> Basic information of the Genotype data</a></li>
<li class="chapter" data-level="3.3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-statistics-of-the-genotype-data"><i class="fa fa-check"></i><b>3.3</b> Basic statistics of the genotype data</a></li>
<li class="chapter" data-level="3.4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#principal-component-analysis"><i class="fa fa-check"></i><b>3.4</b> Principal component analysis</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html"><i class="fa fa-check"></i><b>4</b> Datasets description.</a>
<ul>
<li class="chapter" data-level="4.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAdescription"><i class="fa fa-check"></i><b>4.1</b> CINECA project.</a></li>
<li class="chapter" data-level="4.2" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAopal"><i class="fa fa-check"></i><b>4.2</b> Data organization in the Opal server.</a></li>
<li class="chapter" data-level="4.3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#TCGALiverdatadescription"><i class="fa fa-check"></i><b>4.3</b> TCGA Liver project.</a></li>
<li class="chapter" data-level="4.4" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#GSE66351description"><i class="fa fa-check"></i><b>4.4</b> Epigenetic data from the GSE66351 project</a></li>
<li class="chapter" data-level="4.5" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXdescription"><i class="fa fa-check"></i><b>4.5</b> HELIX project.</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXopal"><i class="fa fa-check"></i><b>4.5.1</b> Data organization in the Opal server.</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Workflows</b></span></li>
<li class="chapter" data-level="5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html"><i class="fa fa-check"></i><b>5</b> Genome wide association study (GWAS) analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#single-cohort"><i class="fa fa-check"></i><b>5.1</b> Single cohort</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-1"><i class="fa fa-check"></i><b>5.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.1.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-vcf-resources-1"><i class="fa fa-check"></i><b>5.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.1.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>5.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.1.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#vcf_merge"><i class="fa fa-check"></i><b>5.1.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.1.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas"><i class="fa fa-check"></i><b>5.1.5</b> Performing the GWAS</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-cohorts"><i class="fa fa-check"></i><b>5.2</b> Multi cohorts</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>5.2.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-vcf"><i class="fa fa-check"></i><b>5.2.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.2.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>5.2.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.2.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>5.2.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.2.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1"><i class="fa fa-check"></i><b>5.2.5</b> Performing a meta - GWAS</a></li>
<li class="chapter" data-level="5.2.6" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#gwas-plots"><i class="fa fa-check"></i><b>5.2.6</b> GWAS Plots</a></li>
<li class="chapter" data-level="5.2.7" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control"><i class="fa fa-check"></i><b>5.2.7</b> Meta-level quality control</a></li>
<li class="chapter" data-level="5.2.8" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-analysis"><i class="fa fa-check"></i><b>5.2.8</b> Meta-analysis</a></li>
<li class="chapter" data-level="5.2.9" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas"><i class="fa fa-check"></i><b>5.2.9</b> Ultra fast pooled GWAS analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#post-omics"><i class="fa fa-check"></i><b>5.3</b> Post-omic analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#enrichment-analysis"><i class="fa fa-check"></i><b>5.3.1</b> Enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html"><i class="fa fa-check"></i><b>6</b> Polygenic risk scores</a>
<ul>
<li class="chapter" data-level="6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#definition"><i class="fa fa-check"></i><b>6.1</b> Definition</a></li>
<li class="chapter" data-level="6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-configuration"><i class="fa fa-check"></i><b>6.2</b> Study configuration</a></li>
<li class="chapter" data-level="6.3" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#source-of-polygenic-scores"><i class="fa fa-check"></i><b>6.3</b> Source of polygenic scores</a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>6.4</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="6.5" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#assign-the-vcf-resources-2"><i class="fa fa-check"></i><b>6.5</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="6.6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#calculate-the-prs"><i class="fa fa-check"></i><b>6.6</b> Calculate the PRS</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#pgs-catalog-example"><i class="fa fa-check"></i><b>6.6.1</b> PGS Catalog Example</a></li>
<li class="chapter" data-level="6.6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#custom-prs_table-example"><i class="fa fa-check"></i><b>6.6.2</b> Custom <code>prs_table</code> example</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-of-the-prs-results"><i class="fa fa-check"></i><b>6.7</b> Study of the PRS results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>7</b> Differential gene expression (DGE) analysis</a>
<ul>
<li class="chapter" data-level="7.1" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#connection-to-the-opal-server-3"><i class="fa fa-check"></i><b>7.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="7.2" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#assign-the-rse-resource"><i class="fa fa-check"></i><b>7.2</b> Assign the RSE resource</a></li>
<li class="chapter" data-level="7.3" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#inspect-the-rse"><i class="fa fa-check"></i><b>7.3</b> Inspect the RSE</a></li>
<li class="chapter" data-level="7.4" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#pre-processing-for-rnaseq-data"><i class="fa fa-check"></i><b>7.4</b> Pre-processing for RNAseq data</a></li>
<li class="chapter" data-level="7.5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#dge-analysis"><i class="fa fa-check"></i><b>7.5</b> DGE analysis</a></li>
<li class="chapter" data-level="7.6" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>7.6</b> Surrogate variable analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>8</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="8.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#initial-steps-for-all-use-cases"><i class="fa fa-check"></i><b>8.1</b> Initial steps for all use cases</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#connection-to-the-opal-server-4"><i class="fa fa-check"></i><b>8.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="8.1.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#assign-the-expressionset-resource"><i class="fa fa-check"></i><b>8.1.2</b> Assign the <code>ExpressionSet</code> resource</a></li>
<li class="chapter" data-level="8.1.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#ewas-inspect"><i class="fa fa-check"></i><b>8.1.3</b> Inspect the <code>ExpressionSet</code></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-pooled-analysis"><i class="fa fa-check"></i><b>8.2</b> Single CpG pooled analysis</a></li>
<li class="chapter" data-level="8.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpgs-pooled-analysis"><i class="fa fa-check"></i><b>8.3</b> Multiple CpGs pooled analysis</a></li>
<li class="chapter" data-level="8.4" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#full-genome-meta-analysis"><i class="fa fa-check"></i><b>8.4</b> Full genome meta-analysis</a></li>
<li class="chapter" data-level="8.5" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>8.5</b> Adjusting for surrogate variables</a></li>
</ul></li>
<li class="part"><span><b>III Showcase on a real dataset</b></span></li>
<li class="chapter" data-level="9" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html"><i class="fa fa-check"></i><b>9</b> Transcriptomic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="9.1" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXone"><i class="fa fa-check"></i><b>9.1</b> Getting started.</a></li>
<li class="chapter" data-level="9.2" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXdatamanip"><i class="fa fa-check"></i><b>9.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="9.3" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXDGE"><i class="fa fa-check"></i><b>9.3</b> Differential gene expression analysis.</a></li>
<li class="chapter" data-level="9.4" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXsurrogateAnnot"><i class="fa fa-check"></i><b>9.4</b> Surrogate variable analysis.</a></li>
<li class="chapter" data-level="9.5" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXenrichment"><i class="fa fa-check"></i><b>9.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html"><i class="fa fa-check"></i><b>10</b> Epigenetic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="10.1" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_gettingstarted"><i class="fa fa-check"></i><b>10.1</b> Getting started.</a></li>
<li class="chapter" data-level="10.2" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_load"><i class="fa fa-check"></i><b>10.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="10.3" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis3"><i class="fa fa-check"></i><b>10.3</b> Full genome meta-analysis.</a></li>
<li class="chapter" data-level="10.4" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis4"><i class="fa fa-check"></i><b>10.4</b> Adjusting for surrogate variables.</a></li>
<li class="chapter" data-level="10.5" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIXenrichment"><i class="fa fa-check"></i><b>10.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="references..html"><a href="references..html"><i class="fa fa-check"></i><b>11</b> References.</a></li>
<li class="part"><span><b>IV Technical details</b></span></li>
<li class="chapter" data-level="12" data-path="gwas.html"><a href="gwas.html"><i class="fa fa-check"></i><b>12</b> GWAS</a></li>
<li class="chapter" data-level="13" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>13</b> PCA</a></li>
<li class="chapter" data-level="14" data-path="differential-privacy.html"><a href="differential-privacy.html"><i class="fa fa-check"></i><b>14</b> Differential privacy</a></li>
<li class="chapter" data-level="15" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html"><i class="fa fa-check"></i><b>15</b> Internal structure of the PRS functionality</a>
<ul>
<li class="chapter" data-level="15.1" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#client"><i class="fa fa-check"></i><b>15.1</b> Client</a></li>
<li class="chapter" data-level="15.2" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-resolver"><i class="fa fa-check"></i><b>15.2</b> Server resolver</a></li>
<li class="chapter" data-level="15.3" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-analysis"><i class="fa fa-check"></i><b>15.3</b> Server analysis</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="compress.html"><a href="compress.html"><i class="fa fa-check"></i><b>16</b> Compression of GDS files vs. performance</a></li>
<li class="chapter" data-level="17" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html"><i class="fa fa-check"></i><b>17</b> Recommended R server specs</a>
<ul>
<li class="chapter" data-level="17.1" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#limma-voom"><i class="fa fa-check"></i><b>17.1</b> Limma + voom</a></li>
<li class="chapter" data-level="17.2" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#gwas-pca"><i class="fa fa-check"></i><b>17.2</b> GWAS + PCA</a></li>
<li class="chapter" data-level="17.3" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#general-remarks"><i class="fa fa-check"></i><b>17.3</b> General remarks</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="disclosure-filter-configurations.html"><a href="disclosure-filter-configurations.html"><i class="fa fa-check"></i><b>18</b> Disclosure filter configurations</a></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="19" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>19</b> Session info</a></li>
<li class="chapter" data-level="20" data-path="contributos.html"><a href="contributos.html"><i class="fa fa-check"></i><b>20</b> Contributos</a></li>
<li class="part"><span><b>VI References</b></span></li>
<li class="chapter" data-level="21" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>21</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genome-wide-association-study-gwas-analysis" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Genome wide association study (GWAS) analysis</h1>
<p>Three different use cases will be illustrated in this section:</p>
<ul>
<li>Single cohort GWAS. Using data from all the individuals.</li>
<li>Multi-cohort GWAS. Using the same individuals as the single cohort separated into three synthetic cohorts.</li>
<li>Polygenic risk score. Using data from all the individuals.</li>
</ul>
<p>For this, we will be using data from the CINECA project presented in the section <a href="DatasetsDescription.html#CINECAdescription">4.1</a>. For all the illustrated cases we will use data divided by chromosome (VCF files). The procedure is the same if all the variants are on a single VCF. The phenotype information used throughout this section is not contained inside the VCF files mentioned previously, it is contained using traditional CSV/Excel files. This is to replicate the typical scenario where an investigator receives the genomic and phenotype data separated and has to merge it to study a specific relationship between the gene expression / variants and a certain phenotype.</p>
<div id="single-cohort" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Single cohort</h2>
<table>
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">
<p>
⚠️ RESOURCES USED ALONG THIS SECTION
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">
<p>
From <a href="https://opal-demo.obiba.org/">https://opal-demo.obiba.org/</a> :
</p>
</td>
</tr>
<tr class="even">
<td style="text-align: left;">
<table>
<thead>
<tr class="header">
<th>
<p>
STUDY
</p>
</th>
<th>
<p>
TABLE
</p>
</th>
<th>
<p>
PROFILE
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.chr{1..21}
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr 
</tbody></table></td></tr>
</tbody>
</table>
<p>The single cohort analysis is a way of performing a GWAS study guaranteeing GDPR data confidentiality. The structure followed is illustrated on the following figure.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:singleCohortImage"></span>
<img src="fig/single_cohort.png" alt="Proposed infrastructure to perform single-cohort GWAS studies." width="935" />
<p class="caption">
Figure 5.1: Proposed infrastructure to perform single-cohort GWAS studies.
</p>
</div>
<p>The data analyst corresponds to the “RStudio” session, which through DataSHIELD Interface (DSI) connects with the Opal server located at the cohort network. This Opal server contains an array of resources that correspond to the different VCF files (sliced by chromosome)<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a> and a resource that corresponds to the phenotypes table of the studied individuals.</p>
<div id="connection-to-the-opal-server-1" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Connection to the Opal server</h3>
<p>We have to create an Opal connection object to the cohort server. We do that using the following functions.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="genome-wide-association-study-gwas-analysis.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSI&#39;</span>)</span>
<span id="cb21-2"><a href="genome-wide-association-study-gwas-analysis.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSOpal&#39;</span>)</span>
<span id="cb21-3"><a href="genome-wide-association-study-gwas-analysis.html#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsBaseClient&#39;</span>)</span>
<span id="cb21-4"><a href="genome-wide-association-study-gwas-analysis.html#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsOmicsClient&#39;</span>)</span>
<span id="cb21-5"><a href="genome-wide-association-study-gwas-analysis.html#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="genome-wide-association-study-gwas-analysis.html#cb21-6" aria-hidden="true" tabindex="-1"></a>builder <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">newDSLoginBuilder</span>()</span>
<span id="cb21-7"><a href="genome-wide-association-study-gwas-analysis.html#cb21-7" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort1&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb21-8"><a href="genome-wide-association-study-gwas-analysis.html#cb21-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb21-9"><a href="genome-wide-association-study-gwas-analysis.html#cb21-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb21-10"><a href="genome-wide-association-study-gwas-analysis.html#cb21-10" aria-hidden="true" tabindex="-1"></a>logindata <span class="ot">&lt;-</span> builder<span class="sc">$</span><span class="fu">build</span>()</span>
<span id="cb21-11"><a href="genome-wide-association-study-gwas-analysis.html#cb21-11" aria-hidden="true" tabindex="-1"></a>conns <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">datashield.login</span>(<span class="at">logins =</span> logindata)</span></code></pre></div>
</div>
<div id="assign-the-vcf-resources-1" class="section level3" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Assign the VCF resources</h3>
<p>Now that we have created a connection object to the Opal, we have started a new R session on the server, our analysis will take place in this remote session, so we have to load the data into it.</p>
<p>In this use case we will use 21 different resources from the <code>GWAS</code> project hosted on the <a href="https://opal-demo.obiba.org/">demo Opal server</a>. This resources correspond to VCF files with information on individual chromosomes. The names of the resources are <code>chrXX</code> (where XX is the chromosome number). Following the Opal syntax, we will refer to them using the string <code>GWAS.chrXX</code>.</p>
<p>To load the resources we will use the <code>DSI::datashield.assign.resource()</code> function. Note that along the use case we use the <code>lapply</code> function to simplify our code since we have to perform repetitive operations. <code>lapply</code> is a way of creating loops in R, for more information visit <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/lapply">the documentation</a>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="genome-wide-association-study-gwas-analysis.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb22-2"><a href="genome-wide-association-study-gwas-analysis.html#cb22-2" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns, <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x))</span>
<span id="cb22-3"><a href="genome-wide-association-study-gwas-analysis.html#cb22-3" aria-hidden="true" tabindex="-1"></a>  })</span></code></pre></div>
<p>Now we have assigned all the resources named <code>GWAS.chrXX</code> into our remote R session. We have assigned them to the variables called <code>chrXX</code>. To verify this step has been performed correctly, we could use the <code>ds.class</code> function to check for their class and that they exist on the remote session.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="genome-wide-association-study-gwas-analysis.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;chr1&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;GDSFileResourceClient&quot; &quot;FileResourceClient&quot;    &quot;ResourceClient&quot;       
[4] &quot;R6&quot;                   </code></pre>
<p>To resolve the resources and retrieve the data in the remote session we will use the <code>DSI::datashield.assign.expr()</code> function. This function runs a line of code on the remote session,<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a> and in particular we want to run the function <code>as.resource.object()</code>, which is the DataSHIELD function in charge of resolving the resources.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="genome-wide-association-study-gwas-analysis.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb25-2"><a href="genome-wide-association-study-gwas-analysis.html#cb25-2" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.expr</span>(<span class="at">conns =</span> conns, <span class="at">symbol =</span> <span class="fu">paste0</span>(<span class="st">&quot;gds&quot;</span>, x, <span class="st">&quot;_object&quot;</span>),</span>
<span id="cb25-3"><a href="genome-wide-association-study-gwas-analysis.html#cb25-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">expr =</span> <span class="fu">as.symbol</span>(<span class="fu">paste0</span>(<span class="st">&quot;as.resource.object(chr&quot;</span>, x, <span class="st">&quot;)&quot;</span>)))</span>
<span id="cb25-4"><a href="genome-wide-association-study-gwas-analysis.html#cb25-4" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>Now we have resolved the resources named <code>chrXX</code> into our remote R session. The objects retrieved have been assigned into variables named <code>gdsXX_object</code>. We can check the process was successful as we did before.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="genome-wide-association-study-gwas-analysis.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;gds1_object&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;GdsGenotypeReader&quot;
attr(,&quot;package&quot;)
[1] &quot;GWASTools&quot;</code></pre>
</div>
<div id="assign-the-phenotypes-1" class="section level3" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Assign the phenotypes</h3>
<p>The objects we have loaded into our remote session are VCF files that contain genomic information of the individuals. To perform a GWAS this information has to be related to some phenotypes to extract relationships. Therefore, we need to load the phenotypes into the remote session. The phenotypes information is a table that contains the individuals as rows and phenotypes as columns. In this use case, we will use a resource (as with the VCF files) to load the phenotypes table into the remote session.</p>
<p>The procedure is practically the same as before with some minor tweaks. To begin, we have the phenotypes information in a single table (hence, a single resource). Then, instead of using the function <code>as.resource.object</code>, we will use <code>as.resource.data.frame</code>, this is because before we were loading a special object (VCF file) and now we are loading a plain table, so the internal treatment on the remote session has to be different.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="genome-wide-association-study-gwas-analysis.html#cb28-1" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns, <span class="st">&quot;pheno&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes&quot;</span>)</span>
<span id="cb28-2"><a href="genome-wide-association-study-gwas-analysis.html#cb28-2" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.expr</span>(<span class="at">conns =</span> conns, <span class="at">symbol =</span> <span class="st">&quot;pheno_object&quot;</span>,</span>
<span id="cb28-3"><a href="genome-wide-association-study-gwas-analysis.html#cb28-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">expr =</span> <span class="fu">quote</span>(<span class="fu">as.resource.data.frame</span>(pheno)))</span></code></pre></div>
<p>We can follow the same analogy as before to know that we have assigned the phenotypes table to a variable called <code>pheno_object</code> on the remote R session.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="genome-wide-association-study-gwas-analysis.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;pheno_object&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; </code></pre>
<p>We can also check the column names to see which information is present on the table.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="genome-wide-association-study-gwas-analysis.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.colnames</span>(<span class="st">&quot;pheno_object&quot;</span>)[[<span class="dv">1</span>]]</span></code></pre></div>
<pre><code> [1] &quot;age_cancer_diagnosis&quot;                &quot;age_recruitment&quot;                    
 [3] &quot;alcohol_day&quot;                         &quot;basophill_count&quot;                    
 [5] &quot;behaviour_tumour&quot;                    &quot;bmi&quot;                                
 [7] &quot;c_reactive_protein_reportability&quot;    &quot;record_origin&quot;                      
 [9] &quot;report_format&quot;                       &quot;cholesterol&quot;                        
[11] &quot;birth_country&quot;                       &quot;actual_tobacco_smoker&quot;              
[13] &quot;date_diagnosis&quot;                      &quot;diabetes_diagnosed_doctor&quot;          
[15] &quot;diastolic_blood_pressure&quot;            &quot;duration_moderate_activity&quot;         
[17] &quot;duration_vigorous_activity&quot;          &quot;energy&quot;                             
[19] &quot;eosinophill_count&quot;                   &quot;ethnicity&quot;                          
[21] &quot;ever_smoked&quot;                         &quot;fasting_time&quot;                       
[23] &quot;alcohol_frequency&quot;                   &quot;hdl_cholesterol&quot;                    
[25] &quot;hip_circumference&quot;                   &quot;histology_tumour&quot;                   
[27] &quot;home_e_coordinate&quot;                   &quot;home_n_coordinate&quot;                  
[29] &quot;e_house_score&quot;                       &quot;s_house_score&quot;                      
[31] &quot;w_house_score&quot;                       &quot;e_income_score&quot;                     
[33] &quot;w_income_score&quot;                      &quot;ldl_direct&quot;                         
[35] &quot;lymphocyte_count&quot;                    &quot;monocyte_count&quot;                     
[37] &quot;neutrophill_count&quot;                   &quot;n_accidental_death_close_gen_family&quot;
[39] &quot;household_count&quot;                     &quot;children_count&quot;                     
[41] &quot;operations_count&quot;                    &quot;operative_procedures&quot;               
[43] &quot;past_tobacco&quot;                        &quot;platelet_count&quot;                     
[45] &quot;pulse_rate&quot;                          &quot;qualification&quot;                      
[47] &quot;cancer_occurrences&quot;                  &quot;reticulocuyte_count&quot;                
[49] &quot;sleep_duration&quot;                      &quot;height&quot;                             
[51] &quot;subject_id&quot;                          &quot;triglycerides&quot;                      
[53] &quot;cancer_type_idc10&quot;                   &quot;cancer_type_idc9&quot;                   
[55] &quot;weight&quot;                              &quot;leukocyte_count&quot;                    
[57] &quot;sex&quot;                                 &quot;phenotype&quot;                          
[59] &quot;age_death&quot;                           &quot;age_diagnosis_diabetes&quot;             
[61] &quot;date_k85_report&quot;                     &quot;date_k86_report&quot;                    
[63] &quot;date_death&quot;                          &quot;enm_diseases&quot;                       
[65] &quot;source_k85_report&quot;                   &quot;source_k86_report&quot;                  
[67] &quot;age_hbp_diagnosis&quot;                   &quot;mental_disorders&quot;                   
[69] &quot;respiratory_disorders&quot;               &quot;cancer_year&quot;                        
[71] &quot;circulatory_disorders&quot;               &quot;digestive_disorders&quot;                
[73] &quot;nervous_disorders&quot;                   &quot;immune_disorders&quot;                   </code></pre>
</div>
<div id="vcf_merge" class="section level3" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> Merge VCF (genotype) and phenotype information</h3>
<p>Arrived at this point, we have 21 VCF objects (each one corresponds to a chromosome) and a phenotypes table on our remote session. The next step is merging each of the VCF objects with the phenotypes table. Before doing that however, we have to gather some information from the phenotypes table. The information to gather is summarized on the Table <a href="exploration-of-genotype-data.html#tab:tab2">3.2</a>.</p>
<table>
<caption><span id="tab:tab2">Table 3.2: </span> Information to gather from the phenotypes table.</caption>
<colgroup>
<col width="20%" />
<col width="79%" />
</colgroup>
<thead>
<tr class="header">
<th>Information</th>
<th>Details</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Which column has the samples identifier?</td>
<td>Column name that contains the IDs of the samples. Those are the IDs that will be matched to the ones present on the VCF objects.</td>
</tr>
<tr class="even">
<td>Which is the sex column on the covariates file?</td>
<td>Column name that contains the sex information.</td>
</tr>
<tr class="odd">
<td>How are males and how are females encoded into this column?</td>
<td>There is not a standard way to encode the sex information, some people use 0/1; male/female; etc. Our approach uses a library that requires a specific encoding, for that reason we need to know the original encoding to perform the translation.</td>
</tr>
<tr class="even">
<td>Which is the case-control column?</td>
<td>Column name that contains the case/control to be studied.</td>
</tr>
<tr class="odd">
<td>What is the encoding for case and for control on this column?</td>
<td>Similar as the sex column, the case/control will be translated to the particular standard of the software we have used to develop our functionalities<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a> .</td>
</tr>
</tbody>
</table>
<p>If we are not sure about the exact column names, we can use the function <code>ds.colnames</code> as we did before. Also, we can use the function <code>ds.table1D</code> to check the level names of the categorical variables.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="genome-wide-association-study-gwas-analysis.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.table1D</span>(<span class="st">&quot;pheno_object$sex&quot;</span>)<span class="sc">$</span>counts</span></code></pre></div>
<pre><code>       pheno_object$sex
female             1271
male               1233
Total              2504</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="genome-wide-association-study-gwas-analysis.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.table1D</span>(<span class="st">&quot;pheno_object$diabetes_diagnosed_doctor&quot;</span>)<span class="sc">$</span>counts</span></code></pre></div>
<pre><code>                     pheno_object$diabetes_diagnosed_doctor
Do_not_know                                             612
No                                                      632
Prefer_not_to_answer                                    590
Yes                                                     668
Total                                                  2502</code></pre>
<p>From the different functions we used, we can extract our answers (Summarized on the Table <a href="exploration-of-genotype-data.html#tab:tab3">3.3</a>)</p>
<table>
<caption><span id="tab:tab3">Table 3.3: </span> Summary of options from the example phenotypes table.</caption>
<colgroup>
<col width="68%" />
<col width="31%" />
</colgroup>
<thead>
<tr class="header">
<th>Question</th>
<th>Answer</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Which column has the samples identifier?</td>
<td>“subject_id”</td>
</tr>
<tr class="even">
<td>Which is the sex column on the covariates file?</td>
<td>“sex”</td>
</tr>
<tr class="odd">
<td>How are males and how are females encoded into this column?</td>
<td>“male”/“female”</td>
</tr>
<tr class="even">
<td>Which is the case-control column of interest of the covariates?</td>
<td>“diabetes_diagnosed_doctor”</td>
</tr>
<tr class="odd">
<td>What is the encoding for case and for control on this column?</td>
<td>“Yes”/“No”</td>
</tr>
</tbody>
</table>
<p>With all this information we can now merge the phenotypes and VCF objects into a type of object named GenotypeData. We will use the <code>ds.GenotypeData</code> function.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="genome-wide-association-study-gwas-analysis.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb37-2"><a href="genome-wide-association-study-gwas-analysis.html#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ds.GenotypeData</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&#39;gds&#39;</span>, x,<span class="st">&#39;_object&#39;</span>), <span class="at">covars =</span> <span class="st">&#39;pheno_object&#39;</span>, </span>
<span id="cb37-3"><a href="genome-wide-association-study-gwas-analysis.html#cb37-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">columnId =</span> <span class="st">&quot;subject_id&quot;</span>, <span class="at">sexId =</span> <span class="st">&quot;sex&quot;</span>, </span>
<span id="cb37-4"><a href="genome-wide-association-study-gwas-analysis.html#cb37-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">male_encoding =</span> <span class="st">&quot;male&quot;</span>, <span class="at">female_encoding =</span> <span class="st">&quot;female&quot;</span>,</span>
<span id="cb37-5"><a href="genome-wide-association-study-gwas-analysis.html#cb37-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">case_control_column =</span> <span class="st">&quot;diabetes_diagnosed_doctor&quot;</span>, </span>
<span id="cb37-6"><a href="genome-wide-association-study-gwas-analysis.html#cb37-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">case =</span> <span class="st">&quot;Yes&quot;</span>, <span class="at">control =</span> <span class="st">&quot;No&quot;</span>, </span>
<span id="cb37-7"><a href="genome-wide-association-study-gwas-analysis.html#cb37-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">newobj.name =</span> <span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, x), <span class="at">datasources =</span> conns)</span>
<span id="cb37-8"><a href="genome-wide-association-study-gwas-analysis.html#cb37-8" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>The objects we just created are named <code>gds.DataXX</code> on the remote session. Now we are ready to perform a GWAS analysis. To perform a GWAS we have to supply this “Genotype Data” objects and some sort of formula in which we will specify the type of association we are interested on studying. The R language has it’s own way of writing formulas, a simple example would be the following (relate the <code>condition</code> to the <code>smoke</code> variable adjusted by <code>sex</code>):</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="genome-wide-association-study-gwas-analysis.html#cb38-1" aria-hidden="true" tabindex="-1"></a>condition <span class="sc">~</span> smoke <span class="sc">+</span> sex</span></code></pre></div>
<p>For this use case we will use the following formula.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="genome-wide-association-study-gwas-analysis.html#cb39-1" aria-hidden="true" tabindex="-1"></a>diabetes_diagnosed_doctor <span class="sc">~</span> sex <span class="sc">+</span> hdl_cholesterol</span></code></pre></div>
</div>
<div id="performing-the-gwas" class="section level3" number="5.1.5">
<h3><span class="header-section-number">5.1.5</span> Performing the GWAS</h3>
<p>We are finally ready to achieve our ultimate goal, performing a GWAS analysis. We have our data (genotype + phenotypes) organized into the correspondent objects (GenotypeData) and our association formula ready. The function <code>ds.metaGWAS</code> is in charge of doing the analysis, we have to supply it with the names of the GenotypeData objects of the remote R session and the formula. Since we have 21 different objects, the <code>paste0</code> function is used to simplify the call.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="genome-wide-association-study-gwas-analysis.html#cb40-1" aria-hidden="true" tabindex="-1"></a>results_single <span class="ot">&lt;-</span> <span class="fu">ds.metaGWAS</span>(<span class="at">genoData =</span> <span class="fu">paste0</span>(<span class="st">&quot;gds.Data&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>), </span>
<span id="cb40-2"><a href="genome-wide-association-study-gwas-analysis.html#cb40-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">model =</span> diabetes_diagnosed_doctor <span class="sc">~</span> sex <span class="sc">+</span> hdl_cholesterol)[[<span class="dv">1</span>]]</span>
<span id="cb40-3"><a href="genome-wide-association-study-gwas-analysis.html#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="genome-wide-association-study-gwas-analysis.html#cb40-4" aria-hidden="true" tabindex="-1"></a>results_single</span></code></pre></div>
<pre><code># A tibble: 865,240 x 10
   rs      chr       pos p.value n.obs  freq    Est Est.SE ref_allele alt_allele
 * &lt;chr&gt;   &lt;chr&gt;   &lt;int&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;     
 1 rs4648~ 1      2.89e6 7.42e-6  1292 0.512 -0.323 0.0721 C          T         
 2 rs4659~ 1      1.20e8 8.74e-6  1296 0.587  0.321 0.0723 T          C         
 3 rs1274~ 1      2.90e6 9.69e-6  1294 0.548 -0.321 0.0725 C          T         
 4 rs3137~ 1      8.62e7 1.08e-5  1293 0.277 -0.375 0.0852 C          T         
 5 rs1132~ 1      1.17e8 1.08e-5  1292 0.859 -0.459 0.104  T          C         
 6 rs1736~ 1      2.82e7 2.26e-5  1293 0.864 -0.466 0.110  C          T         
 7 rs2920~ 1      2.30e7 2.64e-5  1293 0.343  0.327 0.0779 G          A         
 8 rs7411~ 1      1.47e8 3.16e-5  1294 0.932  0.608 0.146  A          G         
 9 rs5800~ 1      5.30e7 3.69e-5  1293 0.929  0.590 0.143  C          A         
10 rs7290~ 1      5.30e7 4.18e-5  1294 0.929  0.586 0.143  G          A         
# ... with 865,230 more rows</code></pre>
<p>We can display the results of the GWAS using a Manhattan plot.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="genome-wide-association-study-gwas-analysis.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(results_single)</span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="OmicSHIELD_bookdown_files/figure-html/singleCohortManhattan_plot-1.png" alt="Manhattan plot of the single-cohort results." width="672" />
<p class="caption">
(#fig:singleCohortManhattan_plot)Manhattan plot of the single-cohort results.
</p>
</div>
<p>Moreover, we can visualize the region with the lowest p-value in detail using the function <code>LocusZoom</code>. There are multiple arguments to control the gene annotation, display a different region of the GWAS, display a different range of positions and more. Make sure to check <code>?LocusZoom</code> for all the details.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="genome-wide-association-study-gwas-analysis.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">LocusZoom</span>(results_single, <span class="at">use_biomaRt =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:locuszoom"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/locuszoom-1.png" alt="LocusZoom plot of the surrounding region of the top hit SNP." width="672" />
<p class="caption">
Figure 5.2: LocusZoom plot of the surrounding region of the top hit SNP.
</p>
</div>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="genome-wide-association-study-gwas-analysis.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datashield.logout</span>(conns)</span></code></pre></div>
</div>
</div>
<div id="multi-cohorts" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Multi cohorts</h2>
<table>
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">
<p>
⚠️ RESOURCES USED ALONG THIS SECTION
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">
<p>
From <a href="https://opal-demo.obiba.org/">https://opal-demo.obiba.org/</a> :
</p>
</td>
</tr>
<tr class="even">
<td style="text-align: left;">
<table>
<thead>
<tr class="header">
<th>
<p>
STUDY
</p>
</th>
<th>
<p>
TABLE
</p>
</th>
<th>
<p>
PROFILE
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.chr{1..21}A
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort2
</p>
</td>
<td>
<p>
GWAS.chr{1..21}B
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort3
</p>
</td>
<td>
<p>
GWAS.chr{1..21}C
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_1
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort2
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_2
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort3
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_3
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr 
</tbody></table></td></tr>
</tbody>
</table>
<p>Now all the individuals used for the single cohort use case will be divided into three different synthetic cohorts. The results from each cohort will then be meta-analyzed to be compared to the results obtained by studying all the individuals together. As with the single-cohort methodology illustrated on the prior section, this method guarantees GDPR data confidentiality. The structure for a three cohort study is illustrated on the following figure (note this can be extrapolated for cohorts with a bigger (or lower) partner count).</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:multiCohortImage"></span>
<img src="fig/multi_cohort.png" alt="Proposed infrastructure to perform multi-cohort GWAS studies." width="736" />
<p class="caption">
Figure 5.3: Proposed infrastructure to perform multi-cohort GWAS studies.
</p>
</div>
<p>The data analyst corresponds to the “RStudio” session, which through DataSHIELD Interface (DSI) connects with the Opal server located at the cohort network. This Opal server contains an array of resources that correspond to the different VCF files (sliced by chromosome)<a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a> and a resource that corresponds to the phenotypes table of the studied individuals.</p>
<p>This use case will portray a complete study methodology, more precisely we will follow this flowchart:</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:multiCohortFlowchart"></span>
<img src="fig/multi_cohort_flowchart.png" alt="Flowchart to perform multi-cohort GWAS studies. Adapted from @winkler2014quality" width="553" />
<p class="caption">
Figure 5.4: Flowchart to perform multi-cohort GWAS studies. Adapted from <span class="citation"><a href="#ref-winkler2014quality" role="doc-biblioref">Winkler et al.</a> (<a href="#ref-winkler2014quality" role="doc-biblioref">2014</a>)</span>
</p>
</div>
<p>Brief description of each step:</p>
<ul>
<li><strong>Step 1:</strong> Logging into the Opal servers and load all the VCF files to the remote R sessions. Corresponds to the sections <a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2">Connection to the Opal server</a>, <a href="genome-wide-association-study-gwas-analysis.html#multi-vcf">Assign the VCF resources</a>, <a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-1">Assign the phenotypes</a> and <a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information">Merge VCF (genotype) and phenotype information</a>.</li>
<li><strong>Step 2:</strong> Performing a GWAS analysis on each cohort and collect all the aggregated non-disclosive statistics. Corresponds to the sections <a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1">Performing the GWAS</a> and <a href="genome-wide-association-study-gwas-analysis.html#gwas-plots">GWAS Plots</a>.</li>
<li><strong>Step 3:</strong> Apply different QC control methodologies to assess possible problems on the data used on each cohort to perform the analysis. Corresponds to the section <a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control">Meta-level quality control</a>.</li>
<li><strong>Step 4:</strong> Perform the meta-analysis of the aggregated statistics. Corresponds to the section <a href="#meta-analysis-2">Meta-analysis</a>.</li>
<li><strong>Step 5:</strong> Perform a pooled ultra fast GWAS and compare the results to the ones of the meta-analysis. Corresponds to the section <a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas">Ultra fast GWAS</a>.</li>
<li><strong>Step 6:</strong> Perform post-omics analysis: Enrichment analysis. Corresponds to the section <a href="genome-wide-association-study-gwas-analysis.html#post-omics">Post-omic analysis</a>.</li>
</ul>
<div id="connection-to-the-opal-server-2" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Connection to the Opal server</h3>
<p>We have to create an Opal connection object to the different cohorts server. We do that using the following functions.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="genome-wide-association-study-gwas-analysis.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSI&#39;</span>)</span>
<span id="cb45-2"><a href="genome-wide-association-study-gwas-analysis.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSOpal&#39;</span>)</span>
<span id="cb45-3"><a href="genome-wide-association-study-gwas-analysis.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsBaseClient&#39;</span>)</span>
<span id="cb45-4"><a href="genome-wide-association-study-gwas-analysis.html#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsOmicsClient&#39;</span>)</span>
<span id="cb45-5"><a href="genome-wide-association-study-gwas-analysis.html#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="genome-wide-association-study-gwas-analysis.html#cb45-6" aria-hidden="true" tabindex="-1"></a>builder <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">newDSLoginBuilder</span>()</span>
<span id="cb45-7"><a href="genome-wide-association-study-gwas-analysis.html#cb45-7" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort1&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb45-8"><a href="genome-wide-association-study-gwas-analysis.html#cb45-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb45-9"><a href="genome-wide-association-study-gwas-analysis.html#cb45-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb45-10"><a href="genome-wide-association-study-gwas-analysis.html#cb45-10" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort2&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb45-11"><a href="genome-wide-association-study-gwas-analysis.html#cb45-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb45-12"><a href="genome-wide-association-study-gwas-analysis.html#cb45-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb45-13"><a href="genome-wide-association-study-gwas-analysis.html#cb45-13" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort3&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb45-14"><a href="genome-wide-association-study-gwas-analysis.html#cb45-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb45-15"><a href="genome-wide-association-study-gwas-analysis.html#cb45-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb45-16"><a href="genome-wide-association-study-gwas-analysis.html#cb45-16" aria-hidden="true" tabindex="-1"></a>logindata <span class="ot">&lt;-</span> builder<span class="sc">$</span><span class="fu">build</span>()</span>
<span id="cb45-17"><a href="genome-wide-association-study-gwas-analysis.html#cb45-17" aria-hidden="true" tabindex="-1"></a>conns <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">datashield.login</span>(<span class="at">logins =</span> logindata)</span></code></pre></div>
<p>It is important to note that in this use case, we are only using one server that contains all the resources (<a href="https://opal-demo.obiba.org/" class="uri">https://opal-demo.obiba.org/</a>), on this server there are all the resources that correspond to the different cohorts. On a more real scenario each one of the <code>builder$append</code> instructions would be connecting to a different server.</p>
</div>
<div id="multi-vcf" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Assign the VCF resources</h3>
<p>Now that we have created a connection object to the different Opals, we have started three R session, our analysis will take place on those remote sessions, so we have to load the data into them.</p>
<p>In this use case we will use 63 different resources from the <code>GWAS</code> project hosted on the <a href="https://opal-demo.obiba.org/">demo Opal server</a>. This resources correspond to VCF files with information on individual chromosomes (21 chromosomes per three cohorts). The names of the resources are <code>chrXXY</code> (where XX is the chromosome number and Y is the cohort A/B/C). Following the Opal syntax, we will refer to them using the string <code>GWAS.chrXXY</code>.</p>
<p>We have to refer specifically to each different server by using <code>conns[X]</code>, this allows us to communicate with the server of interest to indicate to it the resources that it has to load.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="genome-wide-association-study-gwas-analysis.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 1 resources</span></span>
<span id="cb46-2"><a href="genome-wide-association-study-gwas-analysis.html#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb46-3"><a href="genome-wide-association-study-gwas-analysis.html#cb46-3" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">1</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;A&quot;</span>))</span>
<span id="cb46-4"><a href="genome-wide-association-study-gwas-analysis.html#cb46-4" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb46-5"><a href="genome-wide-association-study-gwas-analysis.html#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="genome-wide-association-study-gwas-analysis.html#cb46-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 2 resources</span></span>
<span id="cb46-7"><a href="genome-wide-association-study-gwas-analysis.html#cb46-7" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb46-8"><a href="genome-wide-association-study-gwas-analysis.html#cb46-8" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">2</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;B&quot;</span>))</span>
<span id="cb46-9"><a href="genome-wide-association-study-gwas-analysis.html#cb46-9" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb46-10"><a href="genome-wide-association-study-gwas-analysis.html#cb46-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-11"><a href="genome-wide-association-study-gwas-analysis.html#cb46-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 3 resources</span></span>
<span id="cb46-12"><a href="genome-wide-association-study-gwas-analysis.html#cb46-12" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb46-13"><a href="genome-wide-association-study-gwas-analysis.html#cb46-13" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">3</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;C&quot;</span>))</span>
<span id="cb46-14"><a href="genome-wide-association-study-gwas-analysis.html#cb46-14" aria-hidden="true" tabindex="-1"></a>  })</span></code></pre></div>
<p>Now we have assigned all the resources named <code>GWAS.chrXXY</code> into our remote R session. We have assigned them to the variables called <code>chrXX</code>. To verify this step has been performed correctly, we could use the <code>ds.class</code> function to check for their class and that they exist on the remote sessions.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="genome-wide-association-study-gwas-analysis.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;chr1&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;GDSFileResourceClient&quot; &quot;FileResourceClient&quot;    &quot;ResourceClient&quot;       
[4] &quot;R6&quot;                   

$cohort2
[1] &quot;GDSFileResourceClient&quot; &quot;FileResourceClient&quot;    &quot;ResourceClient&quot;       
[4] &quot;R6&quot;                   

$cohort3
[1] &quot;GDSFileResourceClient&quot; &quot;FileResourceClient&quot;    &quot;ResourceClient&quot;       
[4] &quot;R6&quot;                   </code></pre>
<p>We can see that the object <code>chr1</code> exists in all the three servers.</p>
<p>Finally the resources are resolved to retrieve the data in the remote sessions.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="genome-wide-association-study-gwas-analysis.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb49-2"><a href="genome-wide-association-study-gwas-analysis.html#cb49-2" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.expr</span>(<span class="at">conns =</span> conns, <span class="at">symbol =</span> <span class="fu">paste0</span>(<span class="st">&quot;gds&quot;</span>, x, <span class="st">&quot;_object&quot;</span>),</span>
<span id="cb49-3"><a href="genome-wide-association-study-gwas-analysis.html#cb49-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">expr =</span> <span class="fu">as.symbol</span>(<span class="fu">paste0</span>(<span class="st">&quot;as.resource.object(chr&quot;</span>, x, <span class="st">&quot;)&quot;</span>)))</span>
<span id="cb49-4"><a href="genome-wide-association-study-gwas-analysis.html#cb49-4" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>Now we have resolved the resources named <code>chrXX</code> into our remote R sessions. The objects retrieved have been assigned into variables named <code>gdsXX_object</code>. We can check the process was successful as we did before.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="genome-wide-association-study-gwas-analysis.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;gds1_object&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;GdsGenotypeReader&quot;
attr(,&quot;package&quot;)
[1] &quot;GWASTools&quot;

$cohort2
[1] &quot;GdsGenotypeReader&quot;
attr(,&quot;package&quot;)
[1] &quot;GWASTools&quot;

$cohort3
[1] &quot;GdsGenotypeReader&quot;
attr(,&quot;package&quot;)
[1] &quot;GWASTools&quot;</code></pre>
</div>
<div id="assign-the-phenotypes-1" class="section level3" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> Assign the phenotypes</h3>
<p>The objects we have loaded into our remote sessions are VCF files that contain genomic information of the individuals. To perform a GWAS this information has to be related to some phenotypes to extract relationships. Therefore, we need to load the phenotypes into the remote sessions. The phenotypes information is a table that contains the individuals as rows and phenotypes as columns. In this use case, we will use a resource (as with the VCF files) to load the phenotypes table into the remote sessions.</p>
<p>As with the VCF resources, here we load a different resource to each cohort, as it is expected that each cohort has only the phenotypes information regarding their individuals, therefore different tables. Note that we are assigning the resource to the same variable at each server (<code>pheno</code>), so we only need one function call to resolve all the resources together.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="genome-wide-association-study-gwas-analysis.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 1 phenotypes table</span></span>
<span id="cb52-2"><a href="genome-wide-association-study-gwas-analysis.html#cb52-2" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">1</span>], <span class="st">&quot;pheno&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_1&quot;</span>)</span>
<span id="cb52-3"><a href="genome-wide-association-study-gwas-analysis.html#cb52-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-4"><a href="genome-wide-association-study-gwas-analysis.html#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 2 phenotypes table</span></span>
<span id="cb52-5"><a href="genome-wide-association-study-gwas-analysis.html#cb52-5" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">2</span>], <span class="st">&quot;pheno&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_2&quot;</span>)</span>
<span id="cb52-6"><a href="genome-wide-association-study-gwas-analysis.html#cb52-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-7"><a href="genome-wide-association-study-gwas-analysis.html#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 3 phenotypes table</span></span>
<span id="cb52-8"><a href="genome-wide-association-study-gwas-analysis.html#cb52-8" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">3</span>], <span class="st">&quot;pheno&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_3&quot;</span>)</span>
<span id="cb52-9"><a href="genome-wide-association-study-gwas-analysis.html#cb52-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-10"><a href="genome-wide-association-study-gwas-analysis.html#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Resolve phenotypes table</span></span>
<span id="cb52-11"><a href="genome-wide-association-study-gwas-analysis.html#cb52-11" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.expr</span>(<span class="at">conns =</span> conns, <span class="at">symbol =</span> <span class="st">&quot;pheno_object&quot;</span>,</span>
<span id="cb52-12"><a href="genome-wide-association-study-gwas-analysis.html#cb52-12" aria-hidden="true" tabindex="-1"></a>                            <span class="at">expr =</span> <span class="fu">quote</span>(<span class="fu">as.resource.data.frame</span>(pheno)))</span></code></pre></div>
<p>We can follow the same analogy as before to know that we have assigned the phenotypes table to a variable called <code>pheno_object</code> on the remote R session.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="genome-wide-association-study-gwas-analysis.html#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.class</span>(<span class="st">&quot;pheno_object&quot;</span>)</span></code></pre></div>
<pre><code>$cohort1
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; 

$cohort2
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; 

$cohort3
[1] &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot; </code></pre>
<p>We can also check the column names to see which information is present on the table.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="genome-wide-association-study-gwas-analysis.html#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.colnames</span>(<span class="st">&quot;pheno_object&quot;</span>)[[<span class="dv">1</span>]]</span></code></pre></div>
<pre><code> [1] &quot;age_cancer_diagnosis&quot;                &quot;age_recruitment&quot;                    
 [3] &quot;alcohol_day&quot;                         &quot;basophill_count&quot;                    
 [5] &quot;behaviour_tumour&quot;                    &quot;bmi&quot;                                
 [7] &quot;c_reactive_protein_reportability&quot;    &quot;record_origin&quot;                      
 [9] &quot;report_format&quot;                       &quot;cholesterol&quot;                        
[11] &quot;birth_country&quot;                       &quot;actual_tobacco_smoker&quot;              
[13] &quot;date_diagnosis&quot;                      &quot;diabetes_diagnosed_doctor&quot;          
[15] &quot;diastolic_blood_pressure&quot;            &quot;duration_moderate_activity&quot;         
[17] &quot;duration_vigorous_activity&quot;          &quot;energy&quot;                             
[19] &quot;eosinophill_count&quot;                   &quot;ethnicity&quot;                          
[21] &quot;ever_smoked&quot;                         &quot;fasting_time&quot;                       
[23] &quot;alcohol_frequency&quot;                   &quot;hdl_cholesterol&quot;                    
[25] &quot;hip_circumference&quot;                   &quot;histology_tumour&quot;                   
[27] &quot;home_e_coordinate&quot;                   &quot;home_n_coordinate&quot;                  
[29] &quot;e_house_score&quot;                       &quot;s_house_score&quot;                      
[31] &quot;w_house_score&quot;                       &quot;e_income_score&quot;                     
[33] &quot;w_income_score&quot;                      &quot;ldl_direct&quot;                         
[35] &quot;lymphocyte_count&quot;                    &quot;monocyte_count&quot;                     
[37] &quot;neutrophill_count&quot;                   &quot;n_accidental_death_close_gen_family&quot;
[39] &quot;household_count&quot;                     &quot;children_count&quot;                     
[41] &quot;operations_count&quot;                    &quot;operative_procedures&quot;               
[43] &quot;past_tobacco&quot;                        &quot;platelet_count&quot;                     
[45] &quot;pulse_rate&quot;                          &quot;qualification&quot;                      
[47] &quot;cancer_occurrences&quot;                  &quot;reticulocuyte_count&quot;                
[49] &quot;sleep_duration&quot;                      &quot;height&quot;                             
[51] &quot;subject_id&quot;                          &quot;triglycerides&quot;                      
[53] &quot;cancer_type_idc10&quot;                   &quot;cancer_type_idc9&quot;                   
[55] &quot;weight&quot;                              &quot;leukocyte_count&quot;                    
[57] &quot;sex&quot;                                 &quot;phenotype&quot;                          
[59] &quot;age_death&quot;                           &quot;age_diagnosis_diabetes&quot;             
[61] &quot;date_k85_report&quot;                     &quot;date_k86_report&quot;                    
[63] &quot;date_death&quot;                          &quot;enm_diseases&quot;                       
[65] &quot;source_k85_report&quot;                   &quot;source_k86_report&quot;                  
[67] &quot;age_hbp_diagnosis&quot;                   &quot;mental_disorders&quot;                   
[69] &quot;respiratory_disorders&quot;               &quot;cancer_year&quot;                        
[71] &quot;circulatory_disorders&quot;               &quot;digestive_disorders&quot;                
[73] &quot;nervous_disorders&quot;                   &quot;immune_disorders&quot;                   </code></pre>
</div>
<div id="merge-vcf-genotype-and-phenotype-information" class="section level3" number="5.2.4">
<h3><span class="header-section-number">5.2.4</span> Merge VCF (genotype) and phenotype information</h3>
<p>Arrived at this point, we have 21 VCF objects at each cohort R session (each one corresponds to a chromosome) and a phenotypes table. The next step is merging each of the VCF objects with the phenotypes table. The same procedure as the <a href="genome-wide-association-study-gwas-analysis.html#vcf_merge">single cohort</a> can be applied to extract the arguments required.</p>
<p>With all this information we can now merge the phenotypes and VCF objects into a type of object named GenotypeData. We will use the <code>ds.GenotypeData</code> function.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="genome-wide-association-study-gwas-analysis.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb57-2"><a href="genome-wide-association-study-gwas-analysis.html#cb57-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ds.GenotypeData</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&#39;gds&#39;</span>, x,<span class="st">&#39;_object&#39;</span>), <span class="at">covars =</span> <span class="st">&#39;pheno_object&#39;</span>, </span>
<span id="cb57-3"><a href="genome-wide-association-study-gwas-analysis.html#cb57-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">columnId =</span> <span class="st">&quot;subject_id&quot;</span>, <span class="at">sexId =</span> <span class="st">&quot;sex&quot;</span>,</span>
<span id="cb57-4"><a href="genome-wide-association-study-gwas-analysis.html#cb57-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">male_encoding =</span> <span class="st">&quot;male&quot;</span>, <span class="at">female_encoding =</span> <span class="st">&quot;female&quot;</span>,</span>
<span id="cb57-5"><a href="genome-wide-association-study-gwas-analysis.html#cb57-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">case_control_column =</span> <span class="st">&quot;diabetes_diagnosed_doctor&quot;</span>, </span>
<span id="cb57-6"><a href="genome-wide-association-study-gwas-analysis.html#cb57-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">case =</span> <span class="st">&quot;Yes&quot;</span>, <span class="at">control =</span> <span class="st">&quot;No&quot;</span>,</span>
<span id="cb57-7"><a href="genome-wide-association-study-gwas-analysis.html#cb57-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">newobj.name =</span> <span class="fu">paste0</span>(<span class="st">&#39;gds.Data&#39;</span>, x), <span class="at">datasources =</span> conns)</span>
<span id="cb57-8"><a href="genome-wide-association-study-gwas-analysis.html#cb57-8" aria-hidden="true" tabindex="-1"></a>})</span></code></pre></div>
<p>The objects we just created are named <code>gds.DataXX</code> on the remote session. Now we are ready to perform a GWAS analysis. We will use the same formula as the single cohort use case to be able to compare the results later:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="genome-wide-association-study-gwas-analysis.html#cb58-1" aria-hidden="true" tabindex="-1"></a>diabetes_diagnosed_doctor <span class="sc">~</span> sex <span class="sc">+</span> hdl_cholesterol</span></code></pre></div>
</div>
<div id="performing-the-gwas-1" class="section level3" number="5.2.5">
<h3><span class="header-section-number">5.2.5</span> Performing a meta - GWAS</h3>
<p>We now proceed to perform the GWAS analysis.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="genome-wide-association-study-gwas-analysis.html#cb59-1" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">ds.metaGWAS</span>(<span class="at">genoData =</span> <span class="fu">paste0</span>(<span class="st">&quot;gds.Data&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>), </span>
<span id="cb59-2"><a href="genome-wide-association-study-gwas-analysis.html#cb59-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">model =</span> diabetes_diagnosed_doctor <span class="sc">~</span> sex <span class="sc">+</span> hdl_cholesterol)</span></code></pre></div>
<p>And we can see the results for each cohort.</p>
<div id="cohort-results" class="tab">
<button class="tablinks" id="gwas_results_1" onclick="unrolltab(event, &#39;cohort-results&#39;, &#39;Cohort1&#39;)">
Cohort 1
</button>
<button class="tablinks" onclick="unrolltab(event, &#39;cohort-results&#39;, &#39;Cohort2&#39;)">
Cohort 2
</button>
<button class="tablinks" onclick="unrolltab(event, &#39;cohort-results&#39;, &#39;Cohort3&#39;)">
Cohort 3
</button>
<div id="Cohort1" class="tabcontent">
<div id="section" class="section level4 unnumbered">
<h4>⠀</h4>
<pre><code># A tibble: 865,162 x 10
   rs      chr      pos  p.value n.obs  freq    Est Est.SE ref_allele alt_allele
 * &lt;chr&gt;   &lt;chr&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;     
 1 rs5795~ 1     3.90e7  8.49e-7   430 0.736  0.709 0.144  T          C         
 2 rs1132~ 1     1.17e8  1.64e-9   428 0.874 -1.12  0.186  T          C         
 3 rs4970~ 1     3.90e7  1.30e-6   350 0.747  0.817 0.169  A          G         
 4 rs9726~ 1     2.81e7  1.15e-5   430 0.787  0.766 0.175  C          T         
 5 rs2906~ 1     4.43e7  5.31e-9   428 0.559 -0.554 0.0949 A          G         
 6 rs1211~ 1     1.87e8  2.15e-5   428 0.391 -0.603 0.142  T          A         
 7 rs1626~ 1     2.06e8  1.38e-8   428 0.216 -0.849 0.150  C          A         
 8 rs1090~ 1     2.81e7  1.64e-8   430 0.798  0.682 0.121  G          A         
 9 rs1938~ 1     1.87e8  1.61e-9   429 0.597  0.576 0.0955 A          T         
10 rs1572~ 1     1.99e7  5.38e-3   428 0.206 -0.539 0.194  T          C         
# ... with 865,152 more rows</code></pre>
</div>
</div>
<div id="Cohort2" class="tabcontent">
<div id="section-1" class="section level4 unnumbered">
<h4>⠀</h4>
<pre><code># A tibble: 865,190 x 10
   rs      chr      pos  p.value n.obs  freq    Est Est.SE ref_allele alt_allele
 * &lt;chr&gt;   &lt;chr&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;     
 1 rs9426~ 1     2.98e7  2.52e-7   549 0.757 -0.751  0.146 C          T         
 2 rs1240~ 1     8.57e7  6.22e-5   549 0.929  0.940  0.235 A          G         
 3 rs2762~ 1     1.45e8  8.91e-5   550 0.519  1.21   0.309 T          C         
 4 rs1116~ 1     8.58e7  1.36e-3   549 0.783  0.468  0.146 T          C         
 5 rs4657~ 1     1.67e8  1.02e-3   552 0.342 -0.373  0.114 G          T         
 6 rs2295~ 1     1.60e8  7.14e-4   549 0.896  0.669  0.198 C          A         
 7 rs1891~ 1     1.16e8  6.15e-5   550 0.724  0.550  0.137 T          C         
 8 rs7768~ 1     2.21e8  2.96e-4   551 0.929 -0.853  0.236 C          G         
 9 rs1209~ 1     3.13e7  6.69e-4   551 0.826 -0.523  0.154 C          T         
10 rs6670~ 1     1.16e8  2.28e-4   552 0.722  0.496  0.135 T          C         
# ... with 865,180 more rows</code></pre>
</div>
</div>
<div id="Cohort3" class="tabcontent">
<div id="section-2" class="section level4 unnumbered">
<h4>⠀</h4>
<pre><code># A tibble: 849,560 x 10
   rs    chr      pos  p.value n.obs  freq    Est   Est.SE ref_allele alt_allele
 * &lt;chr&gt; &lt;chr&gt;  &lt;int&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;     
 1 rs49~ 1     6.22e7 2.80e- 5   285 0.632 -0.877  0.209   G          A         
 2 rs75~ 1     6.22e7 0.         285 0.547 -0.630 -0.00451 T          C         
 3 rs13~ 1     1.02e8 2.34e- 3   286 0.938  0.780  0.256   A          G         
 4 rs12~ 1     1.72e8 1.90e- 2   285 0.898  0.622  0.265   T          G         
 5 rs75~ 1     1.02e8 8.25e-14   285 0.866  1.40   0.188   G          A         
 6 rs13~ 1     1.02e8 2.21e-13   285 0.890  2.34   0.320   A          G         
 7 rs60~ 1     2.45e8 2.37e- 4   286 0.857  0.620  0.169   T          A         
 8 rs19~ 1     1.87e8 1.57e-32   285 0.755 -1.90   0.160   C          G         
 9 rs12~ 1     1.72e8 4.63e- 5   285 0.894  1.29   0.317   G          T         
10 rs76~ 1     1.72e8 7.62e- 1   285 0.911  0.114  0.375   G          C         
# ... with 849,550 more rows</code></pre>
</div>
</div>
</div>
</div>
<div id="gwas-plots" class="section level3" number="5.2.6">
<h3><span class="header-section-number">5.2.6</span> GWAS Plots</h3>
<p>We can display the results of the GWAS using a Manhattan plot for each cohort.</p>
<div id="gwas-results" class="tab">
<button class="tablinks" id="gwas_plot_1" onclick="unrolltab(event, &#39;gwas-results&#39;, &#39;Cohort1plot&#39;)">
Cohort 1
</button>
<button class="tablinks" onclick="unrolltab(event, &#39;gwas-results&#39;, &#39;Cohort2plot&#39;)">
Cohort 2
</button>
<button class="tablinks" onclick="unrolltab(event, &#39;gwas-results&#39;, &#39;Cohort3plot&#39;)">
Cohort 3
</button>
<div id="Cohort1plot" class="tabcontent">
<div id="section-3" class="section level4 unnumbered">
<h4>⠀</h4>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="genome-wide-association-study-gwas-analysis.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(results[[<span class="dv">1</span>]])</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:multiCohortManhattanPlot"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/multiCohortManhattanPlot-1.png" alt="Manhattan plot of the Cohort 1." width="672" />
<p class="caption">
Figure 5.5: Manhattan plot of the Cohort 1.
</p>
</div>
</div>
</div>
<div id="Cohort2plot" class="tabcontent">
<div id="section-4" class="section level4 unnumbered">
<h4>⠀</h4>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="genome-wide-association-study-gwas-analysis.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(results[[<span class="dv">2</span>]])</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:multiCohortManhattanPlot2"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/multiCohortManhattanPlot2-1.png" alt="Manhattan plot of the Cohort 1." width="672" />
<p class="caption">
Figure 5.6: Manhattan plot of the Cohort 1.
</p>
</div>
</div>
</div>
<div id="Cohort3plot" class="tabcontent">
<div id="section-5" class="section level4 unnumbered">
<h4>⠀</h4>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="genome-wide-association-study-gwas-analysis.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(results[[<span class="dv">3</span>]])</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:multiCohortManhattanPlot3"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/multiCohortManhattanPlot3-1.png" alt="Manhattan plot of the Cohort 1." width="672" />
<p class="caption">
Figure 5.7: Manhattan plot of the Cohort 1.
</p>
</div>
</div>
</div>
</div>
</div>
<div id="meta-level-quality-control" class="section level3" number="5.2.7">
<h3><span class="header-section-number">5.2.7</span> Meta-level quality control</h3>
<p>Up to this point we have collected the aggregated statistics from each cohort and we have visualized them. Before meta-analyzing this statistics we have to perform some quality control to make sure there are no major issues. There are three different methods bundled in <code>dsOmicsClient</code> to perform such task:</p>
<ul>
<li><strong>EAF Plot:</strong> Plotting the reported EAFs against a reference set, such as from the HapMap or 1000 Genomes projects, or from one specific study, can help visualize patterns that pinpoint strand issues, allele miscoding or the inclusion of individuals whose self-reported ancestry did not match their genetic ancestry.</li>
<li><strong>P-Z Plot:</strong> Plot to reveal issues with beta estimates, standard errors and P values. The plots compare P values reported in the association GWAS with P values calculated from Z-statistics (P.ztest) derived from the reported beta and standard error. A study with no issues should show perfect concordance. It will generate a plot per study, where concordance of all the study SNPs will be displayed.</li>
<li><strong>SE-N Plot:</strong> Plot to reveal issues with trait transformations. A collection of studies with no issues will fall on a line.</li>
</ul>
<p>If any of these QC raises some inconsistencies, please contact the correspondent study analyst.</p>
<div id="eaf-plot" class="section level4" number="5.2.7.1">
<h4><span class="header-section-number">5.2.7.1</span> EAF Plot</h4>
<p>For this plot we are comparing the estimated allele frequencies to a reference set, so the first step is to get ourselves a reference set. For this example we will be using the <code>ALL.wgs.phase1_release_v3.20101123</code> reference set. We will download it and load it into the client R session.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="genome-wide-association-study-gwas-analysis.html#cb66-1" aria-hidden="true" tabindex="-1"></a>reference <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(<span class="st">&quot;https://homepages.uni-regensburg.de/~wit59712/easyqc/exomechip/ALL.wgs.phase1_release_v3.20101123.snps_indels_sv.sites.EUR_AF.txt.exm.tab.sexChr.DI.txt.GZ&quot;</span>)</span>
<span id="cb66-2"><a href="genome-wide-association-study-gwas-analysis.html#cb66-2" aria-hidden="true" tabindex="-1"></a>reference</span></code></pre></div>
<pre><code># A tibble: 75,292 x 8
   ExmName    chrChr_Pos    Chr    Pos Rs          Ref   Alt     Freq
   &lt;chr&gt;      &lt;chr&gt;       &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt;
 1 exm2268640 chr1:762320     1 762320 rs75333668  C     T     0.0013
 2 exm47      chr1:865628     1 865628 rs41285790  G     A     0.01  
 3 exm55      chr1:865694     1 865694 rs9988179   C     T     0.0026
 4 exm67      chr1:871159     1 871159 .           G     A     0.0013
 5 exm106     chr1:874762     1 874762 rs139437968 C     T     0.0013
 6 exm158     chr1:878744     1 878744 .           G     C     0.0026
 7 exm241     chr1:880502     1 880502 rs74047418  C     T     0.0013
 8 exm2253575 chr1:881627     1 881627 rs2272757   G     A     0.63  
 9 exm269     chr1:881918     1 881918 rs35471880  G     A     0.06  
10 exm340     chr1:888659     1 888659 rs3748597   T     C     0.95  
# ... with 75,282 more rows</code></pre>
<p>Now we just use the function <code>eafPlot</code> and we pass to it the GWAS results and the reference set.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="genome-wide-association-study-gwas-analysis.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">eafPlot</span>(<span class="at">x =</span> results, <span class="at">reference =</span> reference, <span class="at">rs_reference =</span> <span class="st">&quot;Rs&quot;</span>, <span class="at">freq_reference =</span> <span class="st">&quot;Freq&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:eafPlot"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/eafPlot-1.png" alt="EAF Plot." width="672" />
<p class="caption">
Figure 5.8: EAF Plot.
</p>
</div>
<p>The obtained results are the expected, which is a linear correlation of slope one. This indicates the alleles are correctly encoded. For further explanation on how to assess the results produced by this plot, please refer to the Figure 4 of <span class="citation"><a href="#ref-winkler2014quality" role="doc-biblioref">Winkler et al.</a> (<a href="#ref-winkler2014quality" role="doc-biblioref">2014</a>)</span>.</p>
</div>
<div id="p-z-plot" class="section level4" number="5.2.7.2">
<h4><span class="header-section-number">5.2.7.2</span> P-Z Plot</h4>
<p>This plot will assess the issues with beta estimates, standard errors and P values. Here we will only need the results from the <code>ds.GWAS</code> function and pass them to the function <code>pzPlot</code>.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="genome-wide-association-study-gwas-analysis.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pzPlot</span>(results)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:pzplot"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/pzplot-1.png" alt="P-Z Plot." width="672" />
<p class="caption">
Figure 5.9: P-Z Plot.
</p>
</div>
<p>Please note that all the SNPs are plotted, for that reason there may be some performance issues if there are many SNPs to be plotted.</p>
<p>A dataset with no issues will show perfect concordance. For further details please refer to the Figure 3 of <span class="citation"><a href="#ref-winkler2014quality" role="doc-biblioref">Winkler et al.</a> (<a href="#ref-winkler2014quality" role="doc-biblioref">2014</a>)</span>.</p>
</div>
<div id="se-n-plot" class="section level4" number="5.2.7.3">
<h4><span class="header-section-number">5.2.7.3</span> SE-N Plot</h4>
<p>Finally, we will assess issues with trait transformations. For this we will only need the results from the <code>ds.GWAS</code> function and pass them to the function <code>seNPlot</code>.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="genome-wide-association-study-gwas-analysis.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seNPlot</span>(results)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:senplot"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/senplot-1.png" alt="SE-N Plot." width="672" />
<p class="caption">
Figure 5.10: SE-N Plot.
</p>
</div>
<p>If all the study points fall on a line, no issues are present. For further detail please refer to the Figure 2 of <span class="citation"><a href="#ref-winkler2014quality" role="doc-biblioref">Winkler et al.</a> (<a href="#ref-winkler2014quality" role="doc-biblioref">2014</a>)</span>.</p>
</div>
</div>
<div id="meta-analysis" class="section level3" number="5.2.8">
<h3><span class="header-section-number">5.2.8</span> Meta-analysis</h3>
<p>Given no problems have been detected on the meta-level QC, we will proceed. Up to this point, we have obtained association results for each cohort on the study. The next step is to combine this information using meta-analysis methods to derive a pooled estimate. Each researcher might have an already built pipeline to do so, or a preferred method; nevertheless, we included a couple methods inside <code>dsOmicsClient</code>. They are the following:</p>
<ul>
<li>Meta-analysis of p-values: Using the sum of logs method (Fisher’s method).</li>
<li>Meta-analysis of beta values: Using a fixed-effects model. Methodology extracted and adapted from <span class="citation"><a href="#ref-betameta" role="doc-biblioref">Pirinen</a> (<a href="#ref-betameta" role="doc-biblioref">2020</a>)</span>.</li>
</ul>
<div id="assessment-of-the-meta-analysis-methods" class="section level4" number="5.2.8.1">
<h4><span class="header-section-number">5.2.8.1</span> Assessment of the meta-analysis methods</h4>
<p>Previously, we have calculated the same GWAS analysis both using all the individuals and a simulated three-cohort scenario. This gives us the ability to assess how good or bad are the different meta-analysis methods; to do so, we will compare both meta-analysis to the complete data, the evaluation method will be counting the number of top hits that match, more specifically we will search for matches on the top twenty hits.</p>
<p>We apply both methods to the multi-cohort use case results:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="genome-wide-association-study-gwas-analysis.html#cb71-1" aria-hidden="true" tabindex="-1"></a>meta_pvalues <span class="ot">&lt;-</span> <span class="fu">metaPvalues</span>(results)</span>
<span id="cb71-2"><a href="genome-wide-association-study-gwas-analysis.html#cb71-2" aria-hidden="true" tabindex="-1"></a>meta_bvalues <span class="ot">&lt;-</span> <span class="fu">metaBetaValues</span>(results)</span></code></pre></div>
<p>We can now compare the 20 top hits from the complete data (previously stored on the variable <code>results_single</code>) to the 20 top hits yielded by both meta-analysis methodologies.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="genome-wide-association-study-gwas-analysis.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract 20 top hits for all three cases</span></span>
<span id="cb72-2"><a href="genome-wide-association-study-gwas-analysis.html#cb72-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># All individuals</span></span>
<span id="cb72-3"><a href="genome-wide-association-study-gwas-analysis.html#cb72-3" aria-hidden="true" tabindex="-1"></a>top_20_complete <span class="ot">&lt;-</span> results_single <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p.value) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>)</span>
<span id="cb72-4"><a href="genome-wide-association-study-gwas-analysis.html#cb72-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Pvalue meta-analysis</span></span>
<span id="cb72-5"><a href="genome-wide-association-study-gwas-analysis.html#cb72-5" aria-hidden="true" tabindex="-1"></a>top_20_pval <span class="ot">&lt;-</span> meta_pvalues <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p.meta) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>)</span>
<span id="cb72-6"><a href="genome-wide-association-study-gwas-analysis.html#cb72-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Beta meta-analysis</span></span>
<span id="cb72-7"><a href="genome-wide-association-study-gwas-analysis.html#cb72-7" aria-hidden="true" tabindex="-1"></a>top_20_beta <span class="ot">&lt;-</span> meta_bvalues <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pval) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>)</span>
<span id="cb72-8"><a href="genome-wide-association-study-gwas-analysis.html#cb72-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-9"><a href="genome-wide-association-study-gwas-analysis.html#cb72-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of hits shared by all individuals case and beta meta-analysis</span></span>
<span id="cb72-10"><a href="genome-wide-association-study-gwas-analysis.html#cb72-10" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(top_20_complete<span class="sc">$</span>rs <span class="sc">%in%</span> top_20_beta<span class="sc">$</span>rs)</span></code></pre></div>
<pre><code>[1] 0</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="genome-wide-association-study-gwas-analysis.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of hits shared by all individuals case and pvalue meta-analysis</span></span>
<span id="cb74-2"><a href="genome-wide-association-study-gwas-analysis.html#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(top_20_complete<span class="sc">$</span>rs <span class="sc">%in%</span> top_20_pval<span class="sc">$</span>rs)</span></code></pre></div>
<pre><code>[1] 0</code></pre>
<p>The meta-analysis using beta values recovers a higher amount of top SNPs, therefore it can be considerate a more powerful method for GWAS meta-analyses.</p>
</div>
<div id="visualizing-the-results-of-the-meta-analysis" class="section level4" number="5.2.8.2">
<h4><span class="header-section-number">5.2.8.2</span> Visualizing the results of the meta-analysis</h4>
<p>We can use the manhattan plot and locus zoom to visualize the results of the meta-analysis as we visualized the single cohorts before. To help assess the meta-analysis behaviour, we will plot together the meta-analysis results and the complete data results.</p>
<div id="manhatan-plot" class="section level5" number="5.2.8.2.1">
<h5><span class="header-section-number">5.2.8.2.1</span> Manhatan plot</h5>
<div id="manhattan-plots" class="tab">
<button class="tablinks" id="manhattan-plots-tab" onclick="unrolltab(event, &#39;manhattan-plots&#39;, &#39;meta-analysis-gwas&#39;)">
Meta-analysis
</button>
<button class="tablinks" onclick="unrolltab(event, &#39;manhattan-plots&#39;, &#39;common-truth&#39;)">
Complete data
</button>
<div id="meta-analysis-gwas" class="tabcontent">
<div id="section-6" class="section level6 unnumbered">
<h6>⠀</h6>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="genome-wide-association-study-gwas-analysis.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(meta_bvalues)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-11"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/unnamed-chunk-11-1.png" alt="Manhattan plot of the meta-analysis results." width="672" />
<p class="caption">
Figure 5.11: Manhattan plot of the meta-analysis results.
</p>
</div>
</div>
</div>
<div id="common-truth" class="tabcontent">
<div id="section-7" class="section level6 unnumbered">
<h6>⠀</h6>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="genome-wide-association-study-gwas-analysis.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(results_single)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-12"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/unnamed-chunk-12-1.png" alt="Manhattan plot of the complete data." width="672" />
<p class="caption">
Figure 5.12: Manhattan plot of the complete data.
</p>
</div>
</div>
</div>
</div>
</div>
<div id="locuszoom" class="section level5" number="5.2.8.2.2">
<h5><span class="header-section-number">5.2.8.2.2</span> LocusZoom</h5>
<div id="locuszoom-plots" class="tab">
<button class="tablinks" id="locuszoom-plots-tabs" onclick="unrolltab(event, &#39;locuszoom-plots&#39;, &#39;meta-analysis-gwas-locus&#39;)">
Meta-analysis
</button>
<button class="tablinks" onclick="unrolltab(event, &#39;locuszoom-plots&#39;, &#39;common-truth-locus&#39;)">
Complete data
</button>
<div id="meta-analysis-gwas-locus" class="tabcontent">
<div id="section-8" class="section level6 unnumbered">
<h6>⠀</h6>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="genome-wide-association-study-gwas-analysis.html#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">LocusZoom</span>(meta_bvalues, <span class="at">pvalue =</span> <span class="st">&quot;p.F&quot;</span>, <span class="at">use_biomaRt =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-13"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/unnamed-chunk-13-1.png" alt="LocusZoom plot of the meta-analysis results." width="672" />
<p class="caption">
Figure 5.13: LocusZoom plot of the meta-analysis results.
</p>
</div>
</div>
</div>
<div id="common-truth-locus" class="tabcontent">
<div id="section-9" class="section level6 unnumbered">
<h6>⠀</h6>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="genome-wide-association-study-gwas-analysis.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">LocusZoom</span>(results_single, <span class="at">use_biomaRt =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-14"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/unnamed-chunk-14-1.png" alt="LocusZoom plot of the complete data." width="672" />
<p class="caption">
Figure 5.14: LocusZoom plot of the complete data.
</p>
</div>
</div>
</div>
</div>
</div>
</div>
<div id="meta-analysis-versus-pooled-linear-models" class="section level4" number="5.2.8.3">
<h4><span class="header-section-number">5.2.8.3</span> Meta-analysis versus pooled linear models</h4>
<p>The <code>dsOmicsClient</code> package also bundles a function to fit pooled generalized linear models (GLM) on single (or multiple) SNPs. On this section we will compare the results of the meta-analysis to the pooled GLM models to see which method is more accurate. The meta-analysis method will be the beta values (fixed-effects model).</p>
<p>The study methodology will be similar to the one we just performed. The top 20 hits yielded by the meta-analysis will be selected, those SNPs will be studied using the pooled GLM function. Both results will be compared to the complete data.</p>
<p>We aim to replicate a real scenario with this study, typically there is no access to the complete data (if so there is no point on performing meta-analysis on synthetic multi-cohorts), therefore we simulate the discovery of SNPs of interest via meta-analysis, then we want to assess if studying individually those SNPs using pooled GLM will put us closer to the results yielded by having the complete data.</p>
<p>We have already calculated the top 20 SNPs discovered by the beta meta-analysis.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="genome-wide-association-study-gwas-analysis.html#cb80-1" aria-hidden="true" tabindex="-1"></a>top_20_beta</span></code></pre></div>
<pre><code># A tibble: 20 x 10
   rs          chr         pos    b.F     se.F   p.F     OR    low     up  pval
   &lt;chr&gt;       &lt;chr&gt;     &lt;int&gt;  &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
 1 rs55713135  1      53499225 -0.810 0.00107      0  0.445  0.444  0.446     0
 2 rs6679530   1     209619557 -0.496 0.00677      0  0.609  0.601  0.617     0
 3 rs6670490   1     205861433 -1.89  0.0359       0  0.151  0.141  0.162     0
 4 rs4660989   1      40246741  0.270 0.00624      0  1.31   1.29   1.33      0
 5 rs4949365   1      31690861  2.49  0.000936     0 12.1   12.1   12.1       0
 6 rs367936905 1     220040256  1.09  0.0223       0  2.97   2.84   3.10      0
 7 rs17537624  1     223173619 -1.00  0.0156       0  0.368  0.357  0.379     0
 8 rs73093734  1     226640064 -1.71  0.00294      0  0.181  0.179  0.182     0
 9 rs538977    1      76670242  0.409 0.00107      0  1.51   1.50   1.51      0
10 rs260913    1     163968414 -1.79  0.0107       0  0.167  0.164  0.171     0
11 rs1323097   1      78702112  2.65  0.0144       0 14.2   13.8   14.6       0
12 rs1336584   1      88656688 -1.08  0.0182       0  0.340  0.329  0.353     0
13 rs6682847   1     214639253 -0.480 0.00505      0  0.618  0.612  0.625     0
14 rs35166549  1     214583778  0.461 0.0110       0  1.59   1.55   1.62      0
15 rs2152556   1     240603314  0.766 0.0165       0  2.15   2.08   2.22      0
16 rs2712270   1     119329378  0.491 0.00421      0  1.63   1.62   1.65      0
17 rs11210076  1      73323631  1.26  0.00190      0  3.53   3.52   3.54      0
18 rs1776023   1     219915393 -1.87  0.0220       0  0.154  0.148  0.161     0
19 rs3890207   1      35504782 -1.19  0.00792      0  0.305  0.300  0.309     0
20 rs1416342   1      58680391 -1.27  0.00450      0  0.280  0.277  0.282     0</code></pre>
<p>Next, we have to fit pooled GLM using the <code>ds.glmSNP</code> function, obviously we have to fit the same model that we did on the GWAS (<code>Diabetes.diagnosed.by.doctor ~ sex + HDL.cholesterol</code>), otherwise the results will not be consistent.</p>
<p>The <code>ds.glmSNP</code> function is an extension of the <code>dsBaseClient::ds.glm</code> function, therefore it offers pooled analysis capabilities and disclosure controls.</p>
<p>Remember that we have the genotype data separated by chromosome, following that we will construct the GLM calls as follows:</p>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="genome-wide-association-study-gwas-analysis.html#cb82-1" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(top_20_beta), <span class="cf">function</span>(x){</span>
<span id="cb82-2"><a href="genome-wide-association-study-gwas-analysis.html#cb82-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tryCatch</span>({</span>
<span id="cb82-3"><a href="genome-wide-association-study-gwas-analysis.html#cb82-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ds.glmSNP</span>(<span class="at">snps.fit =</span> top_20_beta[x, <span class="dv">1</span>], </span>
<span id="cb82-4"><a href="genome-wide-association-study-gwas-analysis.html#cb82-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">model =</span> diabetes_diagnosed_doctor <span class="sc">~</span> sex <span class="sc">+</span> hdl_cholesterol, </span>
<span id="cb82-5"><a href="genome-wide-association-study-gwas-analysis.html#cb82-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">genoData =</span> <span class="fu">paste0</span>(<span class="st">&quot;gds.Data&quot;</span>, top_20_beta[x, <span class="dv">2</span>]))</span>
<span id="cb82-6"><a href="genome-wide-association-study-gwas-analysis.html#cb82-6" aria-hidden="true" tabindex="-1"></a>  }, <span class="at">error =</span> <span class="cf">function</span>(w){<span class="cn">NULL</span>})</span>
<span id="cb82-7"><a href="genome-wide-association-study-gwas-analysis.html#cb82-7" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb82-8"><a href="genome-wide-association-study-gwas-analysis.html#cb82-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Simple data cleaning and column name changes</span></span>
<span id="cb82-9"><a href="genome-wide-association-study-gwas-analysis.html#cb82-9" aria-hidden="true" tabindex="-1"></a>ss <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">do.call</span>(rbind, s))</span>
<span id="cb82-10"><a href="genome-wide-association-study-gwas-analysis.html#cb82-10" aria-hidden="true" tabindex="-1"></a>ss <span class="ot">&lt;-</span> ss <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;rs&quot;</span>)</span>
<span id="cb82-11"><a href="genome-wide-association-study-gwas-analysis.html#cb82-11" aria-hidden="true" tabindex="-1"></a>pooled <span class="ot">&lt;-</span> ss <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Est.pooled =</span> Estimate, </span>
<span id="cb82-12"><a href="genome-wide-association-study-gwas-analysis.html#cb82-12" aria-hidden="true" tabindex="-1"></a>                               <span class="at">SE.pooled =</span> Std..Error, </span>
<span id="cb82-13"><a href="genome-wide-association-study-gwas-analysis.html#cb82-13" aria-hidden="true" tabindex="-1"></a>                               <span class="at">pval.pooled =</span> p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb82-14"><a href="genome-wide-association-study-gwas-analysis.html#cb82-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(n, p.adj))</span></code></pre></div>
<!-- ```{r} -->
<!-- # ds.glmSNP(snps.fit = unlist(top_20_beta[3:5, 1]), model = diabetes_diagnosed_doctor ~ sex + hdl_cholesterol, genoData = paste0("gds.Data", unlist(top_20_beta[3:5, 2])), mc.cores = 3) -->
<!-- ``` -->
<p>We now extract the results of interest from the meta-analysis and the complete data.</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="genome-wide-association-study-gwas-analysis.html#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get data from all individuals study</span></span>
<span id="cb83-2"><a href="genome-wide-association-study-gwas-analysis.html#cb83-2" aria-hidden="true" tabindex="-1"></a>original <span class="ot">&lt;-</span> results_single <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pval.original =</span> p.value, </span>
<span id="cb83-3"><a href="genome-wide-association-study-gwas-analysis.html#cb83-3" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">Est.original =</span> Est, </span>
<span id="cb83-4"><a href="genome-wide-association-study-gwas-analysis.html#cb83-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">SE.original =</span> Est.SE) <span class="sc">%&gt;%</span></span>
<span id="cb83-5"><a href="genome-wide-association-study-gwas-analysis.html#cb83-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rs, Est.original, SE.original, pval.original))</span>
<span id="cb83-6"><a href="genome-wide-association-study-gwas-analysis.html#cb83-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Arrange dadta from beta meta-analysis</span></span>
<span id="cb83-7"><a href="genome-wide-association-study-gwas-analysis.html#cb83-7" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">&lt;-</span> top_20_beta <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pval.beta =</span> pval, </span>
<span id="cb83-8"><a href="genome-wide-association-study-gwas-analysis.html#cb83-8" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">Est.beta =</span> b.F, </span>
<span id="cb83-9"><a href="genome-wide-association-study-gwas-analysis.html#cb83-9" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">SE.beta =</span> se.F) <span class="sc">%&gt;%</span></span>
<span id="cb83-10"><a href="genome-wide-association-study-gwas-analysis.html#cb83-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rs, Est.beta, SE.beta, pval.beta))</span>
<span id="cb83-11"><a href="genome-wide-association-study-gwas-analysis.html#cb83-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge all data on a single table</span></span>
<span id="cb83-12"><a href="genome-wide-association-study-gwas-analysis.html#cb83-12" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(<span class="cf">function</span>(x,y){<span class="fu">inner_join</span>(x,y, <span class="at">by =</span> <span class="st">&quot;rs&quot;</span>)}, </span>
<span id="cb83-13"><a href="genome-wide-association-study-gwas-analysis.html#cb83-13" aria-hidden="true" tabindex="-1"></a>                <span class="fu">list</span>(original, pooled, beta))</span></code></pre></div>
<p>With all the information gathered we can visualize the results on the Table <a href="genome-wide-association-study-gwas-analysis.html#tab:tab4">4</a>.</p>
<caption>
<p><span id="tab:tab4" style="font-size:8pt">Table 4: Beta values, standard errors and p-values yielded by the three methods: GWAS of all individuals, meta-analysis of synthetic cohorts, pooled analysis of synthetic cohorts.</span></p>
</caption>
<p>Finally, to conclude this study we can compute the bias and mean squared error (code not shown). The results are shown on the Table <a href="genome-wide-association-study-gwas-analysis.html#tab:tab5">5</a>.</p>
<caption>
<p><span id="tab:tab5" style="font-size:8pt">Table 5: Mean squared error and bias for the betas from the meta-analysis and pooled analysis compared to the complete data (all individuals).</span></p>
</caption>
<div id="jkxlowiqsz" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#jkxlowiqsz .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 10pt;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jkxlowiqsz .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jkxlowiqsz .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jkxlowiqsz .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jkxlowiqsz .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jkxlowiqsz .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jkxlowiqsz .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: bold;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jkxlowiqsz .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: bold;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jkxlowiqsz .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jkxlowiqsz .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jkxlowiqsz .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jkxlowiqsz .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#jkxlowiqsz .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jkxlowiqsz .gt_from_md > :first-child {
  margin-top: 0;
}

#jkxlowiqsz .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jkxlowiqsz .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jkxlowiqsz .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#jkxlowiqsz .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jkxlowiqsz .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#jkxlowiqsz .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jkxlowiqsz .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jkxlowiqsz .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jkxlowiqsz .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jkxlowiqsz .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jkxlowiqsz .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#jkxlowiqsz .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jkxlowiqsz .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#jkxlowiqsz .gt_left {
  text-align: left;
}

#jkxlowiqsz .gt_center {
  text-align: center;
}

#jkxlowiqsz .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jkxlowiqsz .gt_font_normal {
  font-weight: normal;
}

#jkxlowiqsz .gt_font_bold {
  font-weight: bold;
}

#jkxlowiqsz .gt_font_italic {
  font-style: italic;
}

#jkxlowiqsz .gt_super {
  font-size: 65%;
}

#jkxlowiqsz .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 65%;
}
</style>
<table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">MSE</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">Bias</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr class="gt_group_heading_row">
      <td colspan="3" class="gt_group_heading">Beta</td>
    </tr>
    <tr><td class="gt_row gt_left gt_stub">Pooled</td>
<td class="gt_row gt_right">1.9 &times; 10<sup class='gt_super'>&minus;4</sup></td>
<td class="gt_row gt_right">&minus;1.2 &times; 10<sup class='gt_super'>&minus;3</sup></td></tr>
    <tr><td class="gt_row gt_left gt_stub">Meta</td>
<td class="gt_row gt_right">2.0</td>
<td class="gt_row gt_right">1.3 &times; 10<sup class='gt_super'>&minus;1</sup></td></tr>
  </tbody>
  
  
</table>
</div>
<p>With this results we can conclude that refining the statistics for the SNPs of interest using pooled GLM is advisable. Therefore the recommended flowchart when performing a GWAS using <code>dsOmicsClient+DataSHIELD</code> is the following:</p>
<p>REVISAR ESTOS RESULTADOS</p>
<ol style="list-style-type: decimal">
<li>Perform GWAS with <code>dsOmicsClient::dsGWAS</code></li>
<li>Perform meta-analysis on the results with <code>dsOmicsClient::metaBetaValues</code></li>
<li>Extract top hits from the meta-analysis.</li>
<li>Perform pooled GLM on top hits with <code>dsOmicsClient::ds.glmSNP</code></li>
</ol>
</div>
</div>
<div id="ultra-fast-gwas" class="section level3" number="5.2.9">
<h3><span class="header-section-number">5.2.9</span> Ultra fast pooled GWAS analysis</h3>
<p>An alternate methodology to performing a meta analysis is to use the fast GWAS methodology. This method is named after the performance it achieves, however, it does provide further advantages other than computational speed. The main advantage is that it does not compute the results for each cohort to then be meta-analyzed, it actually computes them using a pooled technique that achieves the <em>exact</em> same results when all the individuals are together than when they are separated by cohorts. This is of great importance as there is no need to worry about cohort-imbalance, which is usually a problem in multi-cohort meta-analysis studies. The only downside of this method is that it does not yield the same results of a traditional GWAS analysis pipeline, the inacuraccy of the results however is really low, detailed information about that can be found on the <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-166">paper that proposes the method</a>.</p>
<p>The performance of the function can be summarized to be ~45 seconds per 1M of SNPs (statistic extracted using 2.5k individuals divided into three study servers). From our test this performance has hold linear up to 10M SNPs, we did not test any further. The graphic for &lt; 1M SNPs can be found on the following figure.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:fastgwasperformance"></span>
<img src="fig/gwas_benchmark.png" alt="Benhmark of `ds.fastGWAS` function." width="80%" />
<p class="caption">
Figure 5.15: Benhmark of <code>ds.fastGWAS</code> function.
</p>
</div>
<p>The usage is similar to the meta-GWAS with a couple of added options, feel free to check the documentation with <code>?ds.fastGWAS</code> to see information about them.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="genome-wide-association-study-gwas-analysis.html#cb84-1" aria-hidden="true" tabindex="-1"></a>results_fast <span class="ot">&lt;-</span> <span class="fu">ds.fastGWAS</span>(<span class="at">genoData =</span> <span class="fu">paste0</span>(<span class="st">&quot;gds.Data&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>), </span>
<span id="cb84-2"><a href="genome-wide-association-study-gwas-analysis.html#cb84-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">formula =</span> <span class="st">&quot;diabetes_diagnosed_doctor ~ sex + hdl_cholesterol&quot;</span>, </span>
<span id="cb84-3"><a href="genome-wide-association-study-gwas-analysis.html#cb84-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">family =</span> <span class="st">&quot;binomial&quot;</span>, <span class="at">snpBlock =</span> <span class="dv">5000</span>)</span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="genome-wide-association-study-gwas-analysis.html#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datashield.logout</span>(conns)</span></code></pre></div>
<div id="gwas-plot" class="section level4" number="5.2.9.1">
<h4><span class="header-section-number">5.2.9.1</span> GWAS Plot</h4>
<p>We can display the results of the fast GWAS using a Manhattan plot.</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="genome-wide-association-study-gwas-analysis.html#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">manhattan</span>(results_fast, <span class="at">pvalCol =</span> <span class="dv">4</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:fastManhattanPlot"></span>
<img src="OmicSHIELD_bookdown_files/figure-html/fastManhattanPlot-1.png" alt="Manhattan plot of the Cohort 1." width="672" />
<p class="caption">
Figure 5.16: Manhattan plot of the Cohort 1.
</p>
</div>
</div>
<div id="meta-analysis-versus-fast-gwas" class="section level4" number="5.2.9.2">
<h4><span class="header-section-number">5.2.9.2</span> Meta-analysis versus fast GWAS</h4>
<p>To finish the GWAS section we can evaluate the results of the beta meta-analysis and the results of the fast GWAS. Both methods will be compared to the ground truth.</p>
<p>As before we can start by seeing the amount of SNPs that both methodologies recover as top hits.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="genome-wide-association-study-gwas-analysis.html#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract 20 top hits for all three cases</span></span>
<span id="cb87-2"><a href="genome-wide-association-study-gwas-analysis.html#cb87-2" aria-hidden="true" tabindex="-1"></a>  <span class="co"># All individuals</span></span>
<span id="cb87-3"><a href="genome-wide-association-study-gwas-analysis.html#cb87-3" aria-hidden="true" tabindex="-1"></a>top_20_complete <span class="ot">&lt;-</span> results_single <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p.value) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>)</span>
<span id="cb87-4"><a href="genome-wide-association-study-gwas-analysis.html#cb87-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Fast GWAS</span></span>
<span id="cb87-5"><a href="genome-wide-association-study-gwas-analysis.html#cb87-5" aria-hidden="true" tabindex="-1"></a>top_20_fast <span class="ot">&lt;-</span> results_fast <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(p.value) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>)</span>
<span id="cb87-6"><a href="genome-wide-association-study-gwas-analysis.html#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Beta meta-analysis</span></span>
<span id="cb87-7"><a href="genome-wide-association-study-gwas-analysis.html#cb87-7" aria-hidden="true" tabindex="-1"></a>top_20_beta <span class="ot">&lt;-</span> meta_bvalues <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(pval) <span class="sc">%&gt;%</span> <span class="fu">head</span>(<span class="dv">20</span>)</span>
<span id="cb87-8"><a href="genome-wide-association-study-gwas-analysis.html#cb87-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-9"><a href="genome-wide-association-study-gwas-analysis.html#cb87-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of hits shared by all individuals case and beta meta-analysis</span></span>
<span id="cb87-10"><a href="genome-wide-association-study-gwas-analysis.html#cb87-10" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(top_20_complete<span class="sc">$</span>rs <span class="sc">%in%</span> top_20_beta<span class="sc">$</span>rs)</span></code></pre></div>
<pre><code>[1] 0</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="genome-wide-association-study-gwas-analysis.html#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of hits shared by all individuals case and fast GWAS</span></span>
<span id="cb89-2"><a href="genome-wide-association-study-gwas-analysis.html#cb89-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(top_20_complete<span class="sc">$</span>rs <span class="sc">%in%</span> top_20_fast<span class="sc">$</span>rs)</span></code></pre></div>
<pre><code>[1] 0</code></pre>
<p>On that regard it seems the meta-analysis is performing better.</p>
<p>We now extract the results of interest from the meta-analysis and the complete data.</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="genome-wide-association-study-gwas-analysis.html#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get data from all individuals study</span></span>
<span id="cb91-2"><a href="genome-wide-association-study-gwas-analysis.html#cb91-2" aria-hidden="true" tabindex="-1"></a>original <span class="ot">&lt;-</span> results_single <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pval.original =</span> p.value, </span>
<span id="cb91-3"><a href="genome-wide-association-study-gwas-analysis.html#cb91-3" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">Est.original =</span> Est, </span>
<span id="cb91-4"><a href="genome-wide-association-study-gwas-analysis.html#cb91-4" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">SE.original =</span> Est.SE) <span class="sc">%&gt;%</span></span>
<span id="cb91-5"><a href="genome-wide-association-study-gwas-analysis.html#cb91-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rs, Est.original, SE.original, pval.original))</span>
<span id="cb91-6"><a href="genome-wide-association-study-gwas-analysis.html#cb91-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Arrange dadta from beta meta-analysis</span></span>
<span id="cb91-7"><a href="genome-wide-association-study-gwas-analysis.html#cb91-7" aria-hidden="true" tabindex="-1"></a>beta <span class="ot">&lt;-</span> top_20_beta <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pval.beta =</span> pval, </span>
<span id="cb91-8"><a href="genome-wide-association-study-gwas-analysis.html#cb91-8" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">Est.beta =</span> b.F, </span>
<span id="cb91-9"><a href="genome-wide-association-study-gwas-analysis.html#cb91-9" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">SE.beta =</span> se.F) <span class="sc">%&gt;%</span></span>
<span id="cb91-10"><a href="genome-wide-association-study-gwas-analysis.html#cb91-10" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rs, Est.beta, SE.beta, pval.beta))</span>
<span id="cb91-11"><a href="genome-wide-association-study-gwas-analysis.html#cb91-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Arrange data from fast GWAS</span></span>
<span id="cb91-12"><a href="genome-wide-association-study-gwas-analysis.html#cb91-12" aria-hidden="true" tabindex="-1"></a>fast <span class="ot">&lt;-</span> top_20_fast <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">pval.fast =</span> p.value, </span>
<span id="cb91-13"><a href="genome-wide-association-study-gwas-analysis.html#cb91-13" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">Est.fast =</span> Est, </span>
<span id="cb91-14"><a href="genome-wide-association-study-gwas-analysis.html#cb91-14" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">SE.fast =</span> Est.SE) <span class="sc">%&gt;%</span></span>
<span id="cb91-15"><a href="genome-wide-association-study-gwas-analysis.html#cb91-15" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(rs, Est.fast, SE.fast, pval.fast))</span>
<span id="cb91-16"><a href="genome-wide-association-study-gwas-analysis.html#cb91-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge all data on a single table</span></span>
<span id="cb91-17"><a href="genome-wide-association-study-gwas-analysis.html#cb91-17" aria-hidden="true" tabindex="-1"></a>total <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(<span class="cf">function</span>(x,y){<span class="fu">inner_join</span>(x,y, <span class="at">by =</span> <span class="st">&quot;rs&quot;</span>)}, </span>
<span id="cb91-18"><a href="genome-wide-association-study-gwas-analysis.html#cb91-18" aria-hidden="true" tabindex="-1"></a>                <span class="fu">list</span>(original, fast, beta))</span></code></pre></div>
<p>With all the information gathered we can visualize the results on the Table <a href="genome-wide-association-study-gwas-analysis.html#tab:tab6">6</a>.</p>
<caption>
<p><span id="tab:tab6" style="font-size:8pt">Table 6: Beta values, standard errors and p-values yielded by the three methods: GWAS of all individuals, meta-analysis of synthetic cohorts, fast GWAS.</span></p>
</caption>
<p>Finally, to conclude this study we can compute the bias and mean squared error (code not shown). The results are shown on the Table <a href="#tab:tab7">7</a>.</p>
<caption>
<p><span id="tab:tab5" style="font-size:8pt">Table 7: Mean squared error and bias for the betas from the meta-analysis and fast GWAS compared to the complete data (all individuals).</span></p>
</caption>
<div id="xiigbegkxf" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#xiigbegkxf .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 10pt;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#xiigbegkxf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xiigbegkxf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#xiigbegkxf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#xiigbegkxf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xiigbegkxf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#xiigbegkxf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: bold;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#xiigbegkxf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: bold;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#xiigbegkxf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#xiigbegkxf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#xiigbegkxf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#xiigbegkxf .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#xiigbegkxf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#xiigbegkxf .gt_from_md > :first-child {
  margin-top: 0;
}

#xiigbegkxf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#xiigbegkxf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#xiigbegkxf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#xiigbegkxf .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xiigbegkxf .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#xiigbegkxf .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#xiigbegkxf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#xiigbegkxf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#xiigbegkxf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#xiigbegkxf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xiigbegkxf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#xiigbegkxf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#xiigbegkxf .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#xiigbegkxf .gt_left {
  text-align: left;
}

#xiigbegkxf .gt_center {
  text-align: center;
}

#xiigbegkxf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#xiigbegkxf .gt_font_normal {
  font-weight: normal;
}

#xiigbegkxf .gt_font_bold {
  font-weight: bold;
}

#xiigbegkxf .gt_font_italic {
  font-style: italic;
}

#xiigbegkxf .gt_super {
  font-size: 65%;
}

#xiigbegkxf .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 65%;
}
</style>
<table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1"></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">MSE</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">Bias</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr class="gt_group_heading_row">
      <td colspan="3" class="gt_group_heading">Beta</td>
    </tr>
    <tr><td class="gt_row gt_left gt_stub">Fast</td>
<td class="gt_row gt_right">NaN</td>
<td class="gt_row gt_right">NaN</td></tr>
    <tr><td class="gt_row gt_left gt_stub">Meta</td>
<td class="gt_row gt_right">NaN</td>
<td class="gt_row gt_right">NaN</td></tr>
  </tbody>
  
  
</table>
</div>
<p>On this brief comparison we’ve seen that the fast GWAS retrieves sensibly less SNPs from the ground truth but it has better performance of bias and MSE. It must be noted that the data used for this GWAS examples does not suffer from any cohort imbalance, so the meta-analysis is expected to yield proper results, on a different environment the fast GWAS is expected to exceed the performance of the meta-analysis in a more remarkable manner.</p>
</div>
</div>
</div>
<div id="post-omics" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Post-omic analysis</h2>
<p>Up to this point we have analyzed the genotype and phenotype data by performing a genome wide association study. We can further study the yielded results by performing an enrichment analysis.</p>
<div id="enrichment-analysis" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Enrichment analysis</h3>
<p>The GWAS results are a collection of SNPs that displayed a relationship with a given phenotype. Those SNPs may fall inside gene regions, and those genes can be functionally related. Studying this can help researchers understand the underlying biological processes. Those analysis are carried by querying the selected gene to a database, for this example we will be using the <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4397996/">DisGeNET database</a>.</p>
<div id="annotation-of-genes" class="section level4" number="5.3.1.1">
<h4><span class="header-section-number">5.3.1.1</span> Annotation of genes</h4>
<p>The first step of the enrichment analysis is to annotate the genes that contain the SNPs of interest yielded by the GWAS. There are many ways to perform this step, we propose to use the <code>biomaRt</code> package and retrieve the annotation from the <code>ensembl</code> genome browser.</p>
<p>First we select the SNPs of interest. We have established a <code>p.value</code> threshold to do so.</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="genome-wide-association-study-gwas-analysis.html#cb92-1" aria-hidden="true" tabindex="-1"></a>snps_of_interest <span class="ot">&lt;-</span> results_fast <span class="sc">%&gt;%</span> </span>
<span id="cb92-2"><a href="genome-wide-association-study-gwas-analysis.html#cb92-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb92-3"><a href="genome-wide-association-study-gwas-analysis.html#cb92-3" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(chr, pos, rs, p.value) <span class="sc">%&gt;%</span> </span>
<span id="cb92-4"><a href="genome-wide-association-study-gwas-analysis.html#cb92-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(p.value <span class="sc">&lt;</span> <span class="fl">0.00005</span>) </span></code></pre></div>
<p>And with this list of SNPs we can move to the annotation.</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="genome-wide-association-study-gwas-analysis.html#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb93-2"><a href="genome-wide-association-study-gwas-analysis.html#cb93-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform ensembl connection</span></span>
<span id="cb93-3"><a href="genome-wide-association-study-gwas-analysis.html#cb93-3" aria-hidden="true" tabindex="-1"></a>gene.ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">&quot;ensembl&quot;</span>, </span>
<span id="cb93-4"><a href="genome-wide-association-study-gwas-analysis.html#cb93-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>, <span class="at">GRCh =</span> <span class="dv">37</span>)</span>
<span id="cb93-5"><a href="genome-wide-association-study-gwas-analysis.html#cb93-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Query SNP by SNP</span></span>
<span id="cb93-6"><a href="genome-wide-association-study-gwas-analysis.html#cb93-6" aria-hidden="true" tabindex="-1"></a>genes_interest <span class="ot">&lt;-</span> <span class="fu">Reduce</span>(rbind, <span class="fu">apply</span>(snps_of_interest, <span class="dv">1</span>, <span class="cf">function</span>(x){</span>
<span id="cb93-7"><a href="genome-wide-association-study-gwas-analysis.html#cb93-7" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb93-8"><a href="genome-wide-association-study-gwas-analysis.html#cb93-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&#39;entrezgene_id&#39;</span>, <span class="st">&#39;chromosome_name&#39;</span>, </span>
<span id="cb93-9"><a href="genome-wide-association-study-gwas-analysis.html#cb93-9" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&#39;ensembl_gene_id&#39;</span>, <span class="st">&#39;start_position&#39;</span>, </span>
<span id="cb93-10"><a href="genome-wide-association-study-gwas-analysis.html#cb93-10" aria-hidden="true" tabindex="-1"></a>                   <span class="st">&quot;end_position&quot;</span>), </span>
<span id="cb93-11"><a href="genome-wide-association-study-gwas-analysis.html#cb93-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">filters =</span> <span class="fu">c</span>(<span class="st">&#39;start&#39;</span>, <span class="st">&#39;end&#39;</span>, <span class="st">&#39;chromosome_name&#39;</span>), </span>
<span id="cb93-12"><a href="genome-wide-association-study-gwas-analysis.html#cb93-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">list</span>(<span class="at">start =</span> x[<span class="dv">2</span>], <span class="at">end =</span> x[<span class="dv">2</span>], </span>
<span id="cb93-13"><a href="genome-wide-association-study-gwas-analysis.html#cb93-13" aria-hidden="true" tabindex="-1"></a>                  <span class="at">chromosome_name =</span> x[<span class="dv">1</span>]), </span>
<span id="cb93-14"><a href="genome-wide-association-study-gwas-analysis.html#cb93-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">mart =</span> gene.ensembl)</span>
<span id="cb93-15"><a href="genome-wide-association-study-gwas-analysis.html#cb93-15" aria-hidden="true" tabindex="-1"></a>}))</span></code></pre></div>
</div>
<div id="enrichment-analysis-query" class="section level4" number="5.3.1.2">
<h4><span class="header-section-number">5.3.1.2</span> Enrichment analysis query</h4>
<p>Having the genes of interest annotated, the final step is to query them to DisGeNET. We will use the function <code>enrichDGN</code> from the package <code>DOSE</code> for it. This package is part of a large <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/figures/clusterProfiler-diagram.png">ecosystem</a> of packages dedicated to enrichment analysis, this allows for easy representation of the query results.</p>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="genome-wide-association-study-gwas-analysis.html#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb94-2"><a href="genome-wide-association-study-gwas-analysis.html#cb94-2" aria-hidden="true" tabindex="-1"></a>enrichment <span class="ot">&lt;-</span> <span class="fu">enrichDGN</span>(<span class="fu">unlist</span>(genes_interest<span class="sc">$</span>entrezgene_id), </span>
<span id="cb94-3"><a href="genome-wide-association-study-gwas-analysis.html#cb94-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">pvalueCutoff =</span> <span class="fl">0.2</span>)</span></code></pre></div>
</div>
<div id="enrichment-analysis-plots" class="section level4" number="5.3.1.3">
<h4><span class="header-section-number">5.3.1.3</span> Enrichment analysis plots</h4>
<p>Here we will portray some of the visualizations that can be performed with the query results. For more information and further visualization options please read the <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/index.html">official vignette</a>.</p>
<div id="barplot" class="section level5" number="5.3.1.3.1">
<h5><span class="header-section-number">5.3.1.3.1</span> Barplot</h5>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="genome-wide-association-study-gwas-analysis.html#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb95-2"><a href="genome-wide-association-study-gwas-analysis.html#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(enrichment)</span></code></pre></div>
<p><img src="OmicSHIELD_bookdown_files/figure-html/barplot_enrichment-1.png" width="672" /></p>
</div>
<div id="dotplot" class="section level5" number="5.3.1.3.2">
<h5><span class="header-section-number">5.3.1.3.2</span> Dotplot</h5>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="genome-wide-association-study-gwas-analysis.html#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dotplot</span>(enrichment)</span></code></pre></div>
<p><img src="OmicSHIELD_bookdown_files/figure-html/dotplot_enrichment-1.png" width="672" /></p>
</div>
<div id="cnet-plot" class="section level5" number="5.3.1.3.3">
<h5><span class="header-section-number">5.3.1.3.3</span> CNET plot</h5>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="genome-wide-association-study-gwas-analysis.html#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggnewscale)</span>
<span id="cb97-2"><a href="genome-wide-association-study-gwas-analysis.html#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="co"># convert gene ID to Symbol</span></span>
<span id="cb97-3"><a href="genome-wide-association-study-gwas-analysis.html#cb97-3" aria-hidden="true" tabindex="-1"></a>edox <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(enrichment, <span class="st">&#39;org.Hs.eg.db&#39;</span>, <span class="st">&#39;ENTREZID&#39;</span>)</span>
<span id="cb97-4"><a href="genome-wide-association-study-gwas-analysis.html#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cnetplot</span>(edox, <span class="at">foldChange=</span>genes_interest<span class="sc">$</span>entrezgene_id)</span></code></pre></div>
<p><img src="OmicSHIELD_bookdown_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
</div>
</div>
</div>
</div>
</div>
<h3> References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-betameta" class="csl-entry">
Pirinen, Matti. 2020. <span>“GWAS 9: Meta-Analysis and Summary Statistics.”</span> <a href="https://www.mv.helsinki.fi/home/mjxpirin/GWAS_course/material/GWAS9.html">https://www.mv.helsinki.fi/home/mjxpirin/GWAS_course/material/GWAS9.html</a>.
</div>
<div id="ref-winkler2014quality" class="csl-entry">
Winkler, Thomas W, Felix R Day, Damien C Croteau-Chonka, Andrew R Wood, Adam E Locke, Reedik Mägi, Teresa Ferreira, et al. 2014. <span>“Quality Control and Conduct of Genome-Wide Association Meta-Analyses.”</span> <em>Nature Protocols</em> 9 (5): 1192–212.
</div>
</div>
<div class="footnotes">
<hr />
<ol start="2">
<li id="fn2"><p>The same methodology and code can be used with unitary VCF resources that contain the variant information of all chromosomes.<a href="genome-wide-association-study-gwas-analysis.html#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>The same methodology and code can be used with unitary VCF resources that contain the variant information of all chromosomes.<a href="genome-wide-association-study-gwas-analysis.html#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>It’s important to note that all the other encodings present on the case-control column that are not the case or control, will be turned into missings. As an example, if we have <code>"cancer"</code>/<code>"no cancer"</code>/<code>"no information"</code>/<code>"prefer not to answer"</code> we can specifcy <code>case="cancer"</code>, <code>control="no cancer"</code> and all the individuals with <code>"no information"</code>/<code>"prefer not to answer"</code> will be interpreted as missings.<a href="genome-wide-association-study-gwas-analysis.html#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>The same methodology and code can be used with unitary VCF resources that contain the variant information of all chromosomes.<a href="genome-wide-association-study-gwas-analysis.html#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="DatasetsDescription.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="polygenic-risk-scores.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OmicSHIELD_bookdown.pdf", "OmicSHIELD_bookdown.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
