<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 SNPTEST | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</title>
  <meta name="description" content="8 SNPTEST | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta name="generator" content="bookdown 0.29 and GitBook 2.6.7" />

  <meta property="og:title" content="8 SNPTEST | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 SNPTEST | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  
  
  



<meta name="date" content="2023-06-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="plink.html"/>
<link rel="next" href="differential-gene-expression-dge-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script>
  function unrolltab(evt, divName, tabName) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementById(divName).getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
      tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementById(divName).getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
      tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(tabName).style.display = "block";
    evt.currentTarget.className += " active";
  }
</script>

<script>
  document.addEventListener("DOMContentLoaded", function(event) { 
    document.getElementById("gwas_results_1").click();
    document.getElementById("gwas_plot_1").click();
    document.getElementById("manhattan-plots-tab").click();
    document.getElementById("locuszoom-plots-tabs").click();
 });
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="" target="blank">OmicSHIELD</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#materials-to-read-beforehand"><i class="fa fa-check"></i><b>1.1</b> Materials to read beforehand</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#what-are-resources-a-very-simple-explanation-without-any-technicalities"><i class="fa fa-check"></i><b>1.2</b> What are “resources”: A very simple explanation without any technicalities</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#capabilities-of-omicshield"><i class="fa fa-check"></i><b>1.3</b> Capabilities of OmicSHIELD</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#opal-servers"><i class="fa fa-check"></i><b>1.4</b> Opal servers</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>2</b> Omic data analysis: types of implemented analyses</a></li>
<li class="chapter" data-level="3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html"><i class="fa fa-check"></i><b>3</b> Datasets description</a>
<ul>
<li class="chapter" data-level="3.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAdescription"><i class="fa fa-check"></i><b>3.1</b> CINECA project</a></li>
<li class="chapter" data-level="3.2" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#CINECAopal"><i class="fa fa-check"></i><b>3.2</b> Data organization in the Opal server</a></li>
<li class="chapter" data-level="3.3" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#TCGALiverdatadescription"><i class="fa fa-check"></i><b>3.3</b> TCGA Liver project</a></li>
<li class="chapter" data-level="3.4" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#GSE66351description"><i class="fa fa-check"></i><b>3.4</b> Epigenetic data from the GSE66351 project</a></li>
<li class="chapter" data-level="3.5" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXdescription"><i class="fa fa-check"></i><b>3.5</b> HELIX project</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="DatasetsDescription.html"><a href="DatasetsDescription.html#HELIXopal"><i class="fa fa-check"></i><b>3.5.1</b> Data organization in the Opal server</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html"><i class="fa fa-check"></i><b>4</b> Exploration of Genotype data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#connection-to-the-opal-server"><i class="fa fa-check"></i><b>4.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="4.2" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#multi-vcf"><i class="fa fa-check"></i><b>4.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="4.3" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>4.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="4.4" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#merge-vcf-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>4.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="4.5" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-info"><i class="fa fa-check"></i><b>4.5</b> Basic information of the Genotype data</a></li>
<li class="chapter" data-level="4.6" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#basic-statistics-of-the-genotype-data"><i class="fa fa-check"></i><b>4.6</b> Basic statistics of the genotype data</a></li>
<li class="chapter" data-level="4.7" data-path="exploration-of-genotype-data.html"><a href="exploration-of-genotype-data.html#principal-component-analysis"><i class="fa fa-check"></i><b>4.7</b> Principal component analysis</a></li>
</ul></li>
<li class="part"><span><b>II Workflows</b></span></li>
<li class="chapter" data-level="5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html"><i class="fa fa-check"></i><b>5</b> Genome wide association study (GWAS) analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#single-cohort"><i class="fa fa-check"></i><b>5.1</b> Single cohort</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-1"><i class="fa fa-check"></i><b>5.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.1.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-vcf-resources"><i class="fa fa-check"></i><b>5.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.1.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes"><i class="fa fa-check"></i><b>5.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.1.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#vcf_merge"><i class="fa fa-check"></i><b>5.1.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.1.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas"><i class="fa fa-check"></i><b>5.1.5</b> Performing the GWAS</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-cohorts"><i class="fa fa-check"></i><b>5.2</b> Multi cohorts</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>5.2.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.2.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-vcf2"><i class="fa fa-check"></i><b>5.2.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="5.2.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-10"><i class="fa fa-check"></i><b>5.2.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="5.2.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information-1"><i class="fa fa-check"></i><b>5.2.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="5.2.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1"><i class="fa fa-check"></i><b>5.2.5</b> Performing a meta - GWAS</a></li>
<li class="chapter" data-level="5.2.6" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#gwas-plots"><i class="fa fa-check"></i><b>5.2.6</b> GWAS Plots</a></li>
<li class="chapter" data-level="5.2.7" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control"><i class="fa fa-check"></i><b>5.2.7</b> Meta-level quality control</a></li>
<li class="chapter" data-level="5.2.8" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-analysis"><i class="fa fa-check"></i><b>5.2.8</b> Meta-analysis</a></li>
<li class="chapter" data-level="5.2.9" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas"><i class="fa fa-check"></i><b>5.2.9</b> Ultra fast pooled GWAS analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#post-omics"><i class="fa fa-check"></i><b>5.3</b> Post-omic analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#enrichment-analysis"><i class="fa fa-check"></i><b>5.3.1</b> Enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html"><i class="fa fa-check"></i><b>6</b> Polygenic risk scores</a>
<ul>
<li class="chapter" data-level="6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#definition"><i class="fa fa-check"></i><b>6.1</b> Definition</a></li>
<li class="chapter" data-level="6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-configuration"><i class="fa fa-check"></i><b>6.2</b> Study configuration</a></li>
<li class="chapter" data-level="6.3" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#source-of-polygenic-scores"><i class="fa fa-check"></i><b>6.3</b> Source of polygenic scores</a></li>
<li class="chapter" data-level="6.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>6.4</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="6.5" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#assign-the-vcf-resources-1"><i class="fa fa-check"></i><b>6.5</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="6.6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#calculate-the-prs"><i class="fa fa-check"></i><b>6.6</b> Calculate the PRS</a>
<ul>
<li class="chapter" data-level="6.6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#pgs-catalog-example"><i class="fa fa-check"></i><b>6.6.1</b> PGS Catalog Example</a></li>
<li class="chapter" data-level="6.6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#custom-prs_table-example"><i class="fa fa-check"></i><b>6.6.2</b> Custom <code>prs_table</code> example</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-of-the-prs-results"><i class="fa fa-check"></i><b>6.7</b> Study of the PRS results</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plink.html"><a href="plink.html"><i class="fa fa-check"></i><b>7</b> PLINK</a>
<ul>
<li class="chapter" data-level="7.1" data-path="plink.html"><a href="plink.html#connection-to-the-opal-server-3"><i class="fa fa-check"></i><b>7.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="7.2" data-path="plink.html"><a href="plink.html#build-the-plink-call"><i class="fa fa-check"></i><b>7.2</b> Build the PLINK call</a></li>
<li class="chapter" data-level="7.3" data-path="plink.html"><a href="plink.html#call-the-analysis"><i class="fa fa-check"></i><b>7.3</b> Call the analysis</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="snptest.html"><a href="snptest.html"><i class="fa fa-check"></i><b>8</b> SNPTEST</a>
<ul>
<li class="chapter" data-level="8.1" data-path="snptest.html"><a href="snptest.html#connection-to-the-opal-server-4"><i class="fa fa-check"></i><b>8.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="8.2" data-path="snptest.html"><a href="snptest.html#build-the-snptest-call"><i class="fa fa-check"></i><b>8.2</b> Build the SNPTEST call</a></li>
<li class="chapter" data-level="8.3" data-path="snptest.html"><a href="snptest.html#call-the-analysis-1"><i class="fa fa-check"></i><b>8.3</b> Call the analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>9</b> Differential gene expression (DGE) analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#connection-to-the-opal-server-5"><i class="fa fa-check"></i><b>9.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="9.2" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#assign-the-rse-resource"><i class="fa fa-check"></i><b>9.2</b> Assign the RSE resource</a></li>
<li class="chapter" data-level="9.3" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#inspect-the-rse"><i class="fa fa-check"></i><b>9.3</b> Inspect the RSE</a></li>
<li class="chapter" data-level="9.4" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#pre-processing-for-rnaseq-data"><i class="fa fa-check"></i><b>9.4</b> Pre-processing for RNAseq data</a></li>
<li class="chapter" data-level="9.5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#dge-analysis"><i class="fa fa-check"></i><b>9.5</b> DGE analysis</a></li>
<li class="chapter" data-level="9.6" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>9.6</b> Surrogate variable analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>10</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="10.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#initial-steps-for-all-use-cases"><i class="fa fa-check"></i><b>10.1</b> Initial steps for all use cases</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#connection-to-the-opal-server-6"><i class="fa fa-check"></i><b>10.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="10.1.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#assign-the-expressionset-resource"><i class="fa fa-check"></i><b>10.1.2</b> Assign the <code>ExpressionSet</code> resource</a></li>
<li class="chapter" data-level="10.1.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#ewas-inspect"><i class="fa fa-check"></i><b>10.1.3</b> Inspect the <code>ExpressionSet</code></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-pooled-analysis"><i class="fa fa-check"></i><b>10.2</b> Single CpG pooled analysis</a></li>
<li class="chapter" data-level="10.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpgs-pooled-analysis"><i class="fa fa-check"></i><b>10.3</b> Multiple CpGs pooled analysis</a></li>
<li class="chapter" data-level="10.4" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#full-genome-meta-analysis"><i class="fa fa-check"></i><b>10.4</b> Full genome meta-analysis</a></li>
<li class="chapter" data-level="10.5" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>10.5</b> Adjusting for surrogate variables</a></li>
</ul></li>
<li class="part"><span><b>III Showcase on a real dataset</b></span></li>
<li class="chapter" data-level="11" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html"><i class="fa fa-check"></i><b>11</b> Transcriptomic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="11.1" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXone"><i class="fa fa-check"></i><b>11.1</b> Getting started.</a></li>
<li class="chapter" data-level="11.2" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXdatamanip"><i class="fa fa-check"></i><b>11.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="11.3" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXDGE"><i class="fa fa-check"></i><b>11.3</b> Differential gene expression analysis.</a></li>
<li class="chapter" data-level="11.4" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXsurrogateAnnot"><i class="fa fa-check"></i><b>11.4</b> Surrogate variable analysis.</a></li>
<li class="chapter" data-level="11.5" data-path="TranscriptHELIX.html"><a href="TranscriptHELIX.html#TranscriptHELIXenrichment"><i class="fa fa-check"></i><b>11.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html"><i class="fa fa-check"></i><b>12</b> Epigenetic analysis in the HELIX cohort.</a>
<ul>
<li class="chapter" data-level="12.1" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_gettingstarted"><i class="fa fa-check"></i><b>12.1</b> Getting started.</a></li>
<li class="chapter" data-level="12.2" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_load"><i class="fa fa-check"></i><b>12.2</b> Data formatting and manipulation in DataSHIELD.</a></li>
<li class="chapter" data-level="12.3" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis3"><i class="fa fa-check"></i><b>12.3</b> Full genome meta-analysis.</a></li>
<li class="chapter" data-level="12.4" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIX_analysis4"><i class="fa fa-check"></i><b>12.4</b> Adjusting for surrogate variables.</a></li>
<li class="chapter" data-level="12.5" data-path="DNAmHELIX.html"><a href="DNAmHELIX.html#DNAmHELIXenrichment"><i class="fa fa-check"></i><b>12.5</b> Enrichment analysis of functional annotations.</a></li>
</ul></li>
<li class="part"><span><b>IV Technical details</b></span></li>
<li class="chapter" data-level="13" data-path="fast-gwas.html"><a href="fast-gwas.html"><i class="fa fa-check"></i><b>13</b> Fast GWAS</a></li>
<li class="chapter" data-level="14" data-path="pca.html"><a href="pca.html"><i class="fa fa-check"></i><b>14</b> PCA</a></li>
<li class="chapter" data-level="15" data-path="differential-privacy.html"><a href="differential-privacy.html"><i class="fa fa-check"></i><b>15</b> Differential privacy</a></li>
<li class="chapter" data-level="16" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html"><i class="fa fa-check"></i><b>16</b> Internal structure of the PRS functionality</a>
<ul>
<li class="chapter" data-level="16.1" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#client"><i class="fa fa-check"></i><b>16.1</b> Client</a></li>
<li class="chapter" data-level="16.2" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-resolver"><i class="fa fa-check"></i><b>16.2</b> Server resolver</a></li>
<li class="chapter" data-level="16.3" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-analysis"><i class="fa fa-check"></i><b>16.3</b> Server analysis</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="compress.html"><a href="compress.html"><i class="fa fa-check"></i><b>17</b> Compression of GDS files vs. performance</a></li>
<li class="chapter" data-level="18" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html"><i class="fa fa-check"></i><b>18</b> Recommended R server specs</a>
<ul>
<li class="chapter" data-level="18.1" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#limma-voom"><i class="fa fa-check"></i><b>18.1</b> Limma + voom</a></li>
<li class="chapter" data-level="18.2" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#gwas-pca"><i class="fa fa-check"></i><b>18.2</b> GWAS + PCA</a></li>
<li class="chapter" data-level="18.3" data-path="recommended-r-server-specs.html"><a href="recommended-r-server-specs.html#general-remarks"><i class="fa fa-check"></i><b>18.3</b> General remarks</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="disclosure-filter-configurations.html"><a href="disclosure-filter-configurations.html"><i class="fa fa-check"></i><b>19</b> Disclosure filter configurations</a></li>
<li class="part"><span><b>V Appendix</b></span></li>
<li class="chapter" data-level="20" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>20</b> Session info</a></li>
<li class="chapter" data-level="21" data-path="contributors.html"><a href="contributors.html"><i class="fa fa-check"></i><b>21</b> Contributors</a></li>
<li class="part"><span><b>VI References</b></span></li>
<li class="chapter" data-level="22" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>22</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="snptest" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">8</span> SNPTEST<a href="snptest.html#snptest" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>A simple association test use case will be illustrated in this section to portrait the usage of SNPTEST in OmicSHIELD.</p>
<table>
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">
<p>
⚠️ RESOURCES USED ALONG THIS SECTION
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">
<p>
From <a href="https://opal-demo.obiba.org/">https://opal-demo.obiba.org/</a> :
</p>
</td>
</tr>
<tr class="even">
<td style="text-align: left;">
<table>
<thead>
<tr class="header">
<th>
<p>
STUDY
</p>
</th>
<th>
<p>
TABLE
</p>
</th>
<th>
<p>
PROFILE
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
RSRC.brge_snptest
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr
</tbody></table></td></tr>
</tbody>
</table>
<p>The structure followed is illustrated on the following figure.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:snptestImage"></span>
<img src="fig/snptest.png" alt="Proposed infrastructure for SNPTEST analysis." width="730" />
<p class="caption">
Figure 8.1: Proposed infrastructure for SNPTEST analysis.
</p>
</div>
<p>The data analyst corresponds to the “RStudio” session, which through DataSHIELD Interface (DSI) connects with the Opal server located at the cohort network. This Opal server contains a resource with the URL where the computational resource is located, also the access credentials are kept secret inside the resource and only the Opal administrator has access to them. A connection to this computational resource is created on the Opal server and the aggregated results are passed to the analyst.</p>
<div id="connection-to-the-opal-server-4" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Connection to the Opal server<a href="snptest.html#connection-to-the-opal-server-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We have to create an Opal connection object to the cohort server. We do that using the following functions.</p>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="snptest.html#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DSOpal)</span>
<span id="cb133-2"><a href="snptest.html#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dsBaseClient)</span>
<span id="cb133-3"><a href="snptest.html#cb133-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dsOmicsClient)</span>
<span id="cb133-4"><a href="snptest.html#cb133-4" aria-hidden="true" tabindex="-1"></a>builder <span class="ot">&lt;-</span> <span class="fu">newDSLoginBuilder</span>()</span>
<span id="cb133-5"><a href="snptest.html#cb133-5" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;study1&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org&quot;</span>,</span>
<span id="cb133-6"><a href="snptest.html#cb133-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span> <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb133-7"><a href="snptest.html#cb133-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">resource =</span> <span class="st">&quot;RSRC.brge_snptest&quot;</span>,</span>
<span id="cb133-8"><a href="snptest.html#cb133-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb133-9"><a href="snptest.html#cb133-9" aria-hidden="true" tabindex="-1"></a>logindata <span class="ot">&lt;-</span> builder<span class="sc">$</span><span class="fu">build</span>()</span>
<span id="cb133-10"><a href="snptest.html#cb133-10" aria-hidden="true" tabindex="-1"></a>conns <span class="ot">&lt;-</span> <span class="fu">datashield.login</span>(<span class="at">logins =</span> logindata, <span class="at">assign =</span> <span class="cn">TRUE</span>,</span>
<span id="cb133-11"><a href="snptest.html#cb133-11" aria-hidden="true" tabindex="-1"></a>                          <span class="at">symbol =</span> <span class="st">&quot;client&quot;</span>)</span></code></pre></div>
</div>
<div id="build-the-snptest-call" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Build the SNPTEST call<a href="snptest.html#build-the-snptest-call" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now, we are ready to run any SNPTEST command from the client site. Notice that in this case we want to assess association between the genotype data in gen format and use as phenotype the variable ‘bin1’ that is in the file ‘sample’, which is a binomial variable (case/control). The sentence in a SNPTEST command would be (NOTE: we avoid -o to indicate the output file since the file will be available in R as a tibble). The arguments must be encapsulated in a single character without the command ‘snptest’.</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="snptest.html#cb134-1" aria-hidden="true" tabindex="-1"></a>snptest.arguments <span class="ot">&lt;-</span> <span class="st">&quot;-frequentist 1 -method score -pheno bin1 -data cohort1.gen cohort1.sample cohort2.gen cohort2.sample&quot;</span></span></code></pre></div>
</div>
<div id="call-the-analysis-1" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Call the analysis<a href="snptest.html#call-the-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Then, the analyses are performed by:</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="snptest.html#cb135-1" aria-hidden="true" tabindex="-1"></a>ans.snptest <span class="ot">&lt;-</span> <span class="fu">ds.snptest</span>(<span class="st">&quot;client&quot;</span>, snptest.arguments)</span></code></pre></div>
<p>The object <code>ans.snptest</code> contains the SNPTEST results at each server as well as the outuput provided by the command</p>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="snptest.html#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(ans.snptest, names)</span></code></pre></div>
<pre><code>$study1
[1] &quot;results&quot;     &quot;snptest.out&quot;</code></pre>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="snptest.html#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ans.snptest<span class="sc">$</span>study1<span class="sc">$</span>results)</span></code></pre></div>
<pre><code># A tibble: 6 x 50
  alternate_ids rsid  chromosome position alleleA alleleB index average_maximum~
  &lt;chr&gt;         &lt;chr&gt; &lt;lgl&gt;         &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt;            &lt;dbl&gt;
1 INSERTION_1   RSID~ NA                1 A       AGTGCTA     1            0.992
2 DELETION_1    RSID~ NA                2 A       -           2            0.946
3 SNPID_3       RSID~ NA                3 A       G           3            0.986
4 SNPID_4       RSID~ NA                4 A       G           4            0.991
5 SNPID_5       RSID~ NA                5 A       G           5            0.984
6 SNPID_6       RSID~ NA                6 A       G           6            0.992
# ... with 42 more variables: info &lt;dbl&gt;, cohort_1_AA &lt;dbl&gt;, cohort_1_AB &lt;dbl&gt;,
#   cohort_1_BB &lt;dbl&gt;, cohort_1_NULL &lt;dbl&gt;, cohort_2_AA &lt;dbl&gt;,
#   cohort_2_AB &lt;dbl&gt;, cohort_2_BB &lt;dbl&gt;, cohort_2_NULL &lt;dbl&gt;, all_AA &lt;dbl&gt;,
#   all_AB &lt;dbl&gt;, all_BB &lt;dbl&gt;, all_NULL &lt;dbl&gt;, all_total &lt;dbl&gt;,
#   cases_AA &lt;dbl&gt;, cases_AB &lt;dbl&gt;, cases_BB &lt;dbl&gt;, cases_NULL &lt;dbl&gt;,
#   cases_total &lt;dbl&gt;, controls_AA &lt;dbl&gt;, controls_AB &lt;dbl&gt;, controls_BB &lt;dbl&gt;,
#   controls_NULL &lt;dbl&gt;, controls_total &lt;dbl&gt;, all_maf &lt;dbl&gt;, cases_maf &lt;dbl&gt;,
#   controls_maf &lt;dbl&gt;, missing_data_proportion &lt;dbl&gt;, het_OR &lt;lgl&gt;,
#   het_OR_lower &lt;lgl&gt;, het_OR_upper &lt;lgl&gt;, hom_OR &lt;lgl&gt;, hom_OR_lower &lt;lgl&gt;,
#   hom_OR_upper &lt;lgl&gt;, all_OR &lt;lgl&gt;, all_OR_lower &lt;lgl&gt;, all_OR_upper &lt;lgl&gt;,
#   frequentist_add_pvalue &lt;dbl&gt;, frequentist_add_info &lt;dbl&gt;,
#   frequentist_add_beta_1 &lt;dbl&gt;, frequentist_add_se_1 &lt;dbl&gt;, comment &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="snptest.html#cb140-1" aria-hidden="true" tabindex="-1"></a>ans.snptest<span class="sc">$</span>study<span class="sc">$</span>snptest.out</span></code></pre></div>
<pre><code>$status
[1] 0

$output
 [1] &quot;Welcome to SNPTEST v2.5.2 (revision 2a8e744975fbd46cc3c020721719dd415da0cb89)&quot;
 [2] &quot;© University of Oxford 2008-2015&quot;                                             
 [3] &quot;https://mathgen.stats.ox.ac.uk/genetics_software/snptest/snptest.html&quot;        
 [4] &quot;Read LICENCE file for conditions of use.&quot;                                     
 [5] &quot;&quot;                                                                             
 [6] &quot;==============&quot;                                                               
 [7] &quot;&quot;                                                                             
 [8] &quot;Data Files : &quot;                                                                
 [9] &quot; -gen files : cohort1.gen cohort2.gen &quot;                                       
[10] &quot; -sample files : cohort1.sample cohort2.sample &quot;                              
[11] &quot;&quot;                                                                             
[12] &quot;Tests : &quot;                                                                     
[13] &quot; -frequentist : 1&quot;                                                            
[14] &quot; -method score&quot;                                                               
[15] &quot;&quot;                                                                             
[16] &quot;reading sample exclusion lists&quot;                                               
[17] &quot;&quot;                                                                             
[18] &quot;Inspecting data (this may take some time)...&quot;                                 
[19] &quot;Sample and exclusions summary :&quot;                                              
[20] &quot; - Number of individuals in : (cohort 1)   (cohort 2) &quot;                       
[21] &quot;                              500          500        &quot;                       
[22] &quot;&quot;                                                                             
[23] &quot;&quot;                                                                             
[24] &quot;Reading sample files :&quot;                                                       
[25] &quot;Summary of covariates and phenotypes&quot;                                         
[26] &quot; # discrete variables : 3&quot;                                                    
[27] &quot;  cov1 : type = D (Discrete covariate)&quot;                                       
[28] &quot;  cov2 : type = D (Discrete covariate)&quot;                                       
[29] &quot;  sex : type = D (Discrete covariate)&quot;                                        
[30] &quot; # continuous variables : 2&quot;                                                  
[31] &quot;  cov3 : type = C (Continuous covariate)&quot;                                     
[32] &quot;  cov4 : type = C (Continuous covariate)&quot;                                     
[33] &quot; # phenotypes : 4&quot;                                                            
[34] &quot;  pheno1 : type = P (Continuous phenotype)&quot;                                   
[35] &quot;  pheno2 : type = P (Continuous phenotype)&quot;                                   
[36] &quot;  bin1 : type = B (Binary phenotype)&quot;                                         
[37] &quot;  bin2 : type = B (Binary phenotype)&quot;                                         
[38] &quot;Covariate summary :&quot;                                                          
[39] &quot;  (no covariates in use.)&quot;                                                    
[40] &quot;Phenotype summary :&quot;                                                          
[41] &quot;  bin1    : missing  levels&quot;                                                  
[42] &quot;            2        1(998)&quot;                                                  
[43] &quot;&quot;                                                                             
[44] &quot;You have specified the following model:&quot;                                      
[45] &quot;  bin1 ~ (baseline) + (genotype)&quot;                                             
[46] &quot;&quot;                                                                             
[47] &quot;Phenotype being used : bin1&quot;                                                  
[48] &quot;&quot;                                                                             
[49] &quot;Data Summaries : &quot;                                                            
[50] &quot; -number of SNPs = (unknown)&quot;                                                 
[51] &quot;&quot;                                                                             
[52] &quot;Data with missing genotype data threshold and exclusion list applied :&quot;       
[53] &quot; cohort1.gen : 500&quot;                                                           
[54] &quot; cohort2.gen : 500&quot;                                                           
[55] &quot;&quot;                                                                             
[56] &quot;--------------------------------------------------------------------------&quot;   
[57] &quot;&quot;                                                                             
[58] &quot;SinglePhenotypeTest&quot;                                                          
[59] &quot;--------------------------------------------------------------------------&quot;   
[60] &quot;&quot;                                                                             
[61] &quot;Analyzing Data :&quot;                                                             
[62] &quot; scanning... read chunk [1 of (unknown)]... done.&quot;                            
[63] &quot; scanning... read chunk [2 of (unknown)]... done.&quot;                            
[64] &quot; scanning... no more data.&quot;                                                   
[65] &quot;&quot;                                                                             
[66] &quot;finito&quot;                                                                       

$error
character(0)

$command
[1] &quot;cd /home/master/data &amp;&amp; snptest -frequentist 1 -method score -pheno bin1 -data cohort1.gen cohort1.sample cohort2.gen cohort2.sample -o /tmp/ssh-8008/ex.out&quot;

attr(,&quot;class&quot;)
[1] &quot;resource.exec&quot;</code></pre>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="snptest.html#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datashield.logout</span>(conns)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="plink.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-gene-expression-dge-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OmicSHIELD_bookdown.pdf", "OmicSHIELD_bookdown.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
